%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{tgtermes}
\usepackage{tgheros}
\renewcommand{\ttdefault}{txtt}



\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=auto}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\usepackage{sphinxmessages}
\setcounter{tocdepth}{1}

\global\renewcommand{\AA}{\text{\r{A}}}

\title{GOMC Documentation}
\date{Jun 24, 2022}
\release{2.75}
\author{GOMC Development Group}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}



\chapter{Overview}
\label{\detokenize{overview:overview}}\label{\detokenize{overview::doc}}
\sphinxAtStartPar
This document will instruct a new user how to download, compile, prepare the input files, and run the GOMC molecular simulation code. A basic understanding of statistical physics is recommended to complete this tutorial.

\sphinxAtStartPar
To demonstrate the capabilities of the code, the user is guided through the process of downloading, compiling a GOMC executable, and preparing input files such as PDB, PSF, Parameter, and Configuration file. Executable is then used to calculate the saturated vapor and liquid equilibria (VLE) using Gibbs Ensemble Monte Carlo on systems of pure isobutane (R600a), a branched alkane that whose application as a refrigerant/propellant is increasing. The Transferable Potentials for Phase Equilibria (TraPPE) united atom (UA) force field is used to describe the molecular geometry constraints and the intermolecular interactions.

\sphinxAtStartPar
\sphinxurl{http://en.wikipedia.org/wiki/Isobutane}


\chapter{Release 2.75 Notes}
\label{\detokenize{release_notes:release-2-75-notes}}\label{\detokenize{release_notes::doc}}
\sphinxAtStartPar
The GPL 3.0 License has been replaced by the MIT License. Users should upgrade only if they are comfortable with running code under this new license.

\sphinxAtStartPar
Certain changes have been made which differ from previous GOMC behavior.  New features have been added to assist users and developers in compiling, running, and analyzing, improve reproducability, increase the capacity of GOMC to simulate biological molecules, perform Hybrid Monte\sphinxhyphen{}Carlo/Molecular Dynamics simulations, and increase performance.  A non\sphinxhyphen{}comprehensive list is provided below.

\sphinxAtStartPar
Differing behavior:
\begin{quote}

\sphinxAtStartPar
Previous GOMC versions used REMARKS in the PDB header to save box dimensions and random number generator state.  While this is still currently partially supported, it is in the process of being deprecated and checkpointing should be used.  Secondly, restarting GOMC should no longer be performed using the merged files containing both boxes (*.BOX\_0.pdb, *.merged.psf) which are produced solely for visualization.  Furthermore, The user should now use the box\sphinxhyphen{}specific restart files (pdb, psf, xsc, coor, chk) as input.  Finally, the pdb trajectory files (*.BOX\_0.pdb) are in the process of being deprecated and replaced by binary trajectory files (*.BOX\_0.dcd), though both are currently available.
\end{quote}

\sphinxAtStartPar
Updated Manual Sections:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{(}{)}%
\item {} 
\sphinxAtStartPar
Introduction

\sphinxAtStartPar
GOMC supported Monte Carlo moves
\begin{quote}

\sphinxAtStartPar
Force\sphinxhyphen{}biased Multiparticle move (Rigid\sphinxhyphen{}body displacement or rotation of all molecules)

\sphinxAtStartPar
Brownian Motion Multiparticle move (Rigid\sphinxhyphen{}body displacement or rotation of all molecule
\end{quote}

\sphinxAtStartPar
GOMC supported molecules
\begin{quote}

\sphinxAtStartPar
Biological molecules which consist of multiple residues are now supported.  Care should be taken when generating the molecules such that all bonds, angles, and dihedrals are included in the PSF file.  Support for these molecules is experimental.
\end{quote}

\item {} 
\sphinxAtStartPar
Recommended Software Tools

\sphinxAtStartPar
Molecular Simulation Design Framework (MoSDeF)

\item {} 
\sphinxAtStartPar
Compiling GOMC

\sphinxAtStartPar
./metamake.sh {[}OPTIONS{]} {[}ARGUMENTS{]}

\sphinxAtStartPar
OPTIONS
\begin{quote}
\begin{optionlist}{3cm}
\item [\sphinxhyphen{}d]  
\sphinxAtStartPar
Compile in debug mode.
\item [\sphinxhyphen{}g]  
\sphinxAtStartPar
Compile with gcc.
\item [\sphinxhyphen{}m]  
\sphinxAtStartPar
Compile with MPI enabled.
\item [\sphinxhyphen{}p]  
\sphinxAtStartPar
Compile with NVTX profiling for CUDA
\item [\sphinxhyphen{}t]  
\sphinxAtStartPar
Compile Google tests.
\end{optionlist}
\end{quote}

\item {} 
\sphinxAtStartPar
GPU\sphinxhyphen{}accelerated GOMC

\sphinxAtStartPar
A section describing the GPU\sphinxhyphen{}accelerated regions of GOMC code is included.

\item {} 
\sphinxAtStartPar
Input File Formats

\sphinxAtStartPar
Support for binary coordinates, trajectories, box dimensions, velocities, and checkpoint files are included.  Checkpoint files guaruntee trajectory files can be concatenated, along with ensuring no deviation from a single simulation’s results in a simulation which was interrupted and restarted.
\begin{quote}

\sphinxAtStartPar
(5a) Restart a simulation from a checkpoint

\sphinxAtStartPar
(5b) Restart from binary coordinates and box dimensions

\sphinxAtStartPar
(5c) Target insertions to subvolumes

\sphinxAtStartPar
(5d) Overwrite start step
\end{quote}

\item {} 
\sphinxAtStartPar
Hybrid Monte Carlo\sphinxhyphen{}Molecular Dynamics (MCMD)

\sphinxAtStartPar
Instructions on running an alternating Hybrid MCMD algorithm using GOMC and NAMD are included.

\end{enumerate}


\chapter{Introduction}
\label{\detokenize{introduction:introduction}}\label{\detokenize{introduction::doc}}
\sphinxAtStartPar
GPU Optimized Monte Carlo (GOMC) is open\sphinxhyphen{}source software for simulating many\sphinxhyphen{}body molecular systems using the Metropolis Monte Carlo algorithm. GOMC is written in object oriented C++, which was chosen since it offers a good balance between code development time, interoperability with existing software elements, and code performance. The software may be compiled as a single\sphinxhyphen{}threaded application, a multi\sphinxhyphen{}threaded application using OpenMP, or to use many\sphinxhyphen{}core heterogeneous CPU\sphinxhyphen{}GPU architectures using OpenMP and CUDA. GOMC officially supports Windows 7 or newer and most modern distribution of GNU/Linux. This software has the ability to compile on recent versions of macOS (x64 \& ARM); however, such a platform is not officially supported.

\sphinxAtStartPar
GOMC employs widely\sphinxhyphen{}used simulation file types (PDB, PSF, CHARMM\sphinxhyphen{}style parameter file) and supports polar and non\sphinxhyphen{}polar linear and branched molecules. GOMC can be used to study vapor\sphinxhyphen{}liquid and liquid\sphinxhyphen{}liquid equilibria, adsorption in porous materials, surfactant self\sphinxhyphen{}assembly, and condensed phase structure for complex molecules.

\sphinxAtStartPar
To cite GOMC software, please refer to \sphinxhref{https://www.sciencedirect.com/science/article/pii/S2352711018301171}{GOMC paper}.


\section{GOMC supported ensembles:}
\label{\detokenize{introduction:gomc-supported-ensembles}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Canonical (NVT)

\item {} 
\sphinxAtStartPar
Isobaric\sphinxhyphen{}isothermal (NPT)

\item {} 
\sphinxAtStartPar
Grand canonical (\(\mu\) VT)

\item {} 
\sphinxAtStartPar
Constant volume Gibbs (NVT\sphinxhyphen{}Gibbs)

\item {} 
\sphinxAtStartPar
Constant pressure Gibbs (NPT\sphinxhyphen{}Gibbs)

\end{itemize}


\section{GOMC supported Monte Carlo moves:}
\label{\detokenize{introduction:gomc-supported-monte-carlo-moves}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Rigid\sphinxhyphen{}body displacement

\item {} 
\sphinxAtStartPar
Rigid\sphinxhyphen{}body rotation

\item {} 
\sphinxAtStartPar
\sphinxhref{https://www.tandfonline.com/doi/abs/10.1080/08927022.2013.804183?journalCode=gmos20}{Force\sphinxhyphen{}biased Multiparticle} move (Rigid\sphinxhyphen{}body displacement or rotation of all molecules)

\item {} 
\sphinxAtStartPar
\sphinxhref{https://www.tandfonline.com/doi/abs/10.1080/08927022.2013.804183?journalCode=gmos20}{Brownian Motion Multiparticle} move (Rigid\sphinxhyphen{}body displacement or rotation of all molecules)

\item {} 
\sphinxAtStartPar
Regrowth using \sphinxhref{https://pubs.acs.org/doi/abs/10.1021/jp984742e}{coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias}

\item {} 
\sphinxAtStartPar
Crankshaft using combination of \sphinxhref{https://aip.scitation.org/doi/abs/10.1063/1.438608}{crankshaft} and \sphinxhref{https://pubs.acs.org/doi/abs/10.1021/jp984742e}{coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias}

\item {} 
\sphinxAtStartPar
Intra\sphinxhyphen{}box swap using \sphinxhref{https://pubs.acs.org/doi/abs/10.1021/jp984742e}{coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias}

\item {} 
\sphinxAtStartPar
Intra\sphinxhyphen{}box \sphinxhref{https://aip.scitation.org/doi/abs/10.1063/1.5025184}{molecular exchange Monte Carlo}

\item {} 
\sphinxAtStartPar
Intra\sphinxhyphen{}box targeted swap using \sphinxhref{https://pubs.acs.org/doi/abs/10.1021/jp984742e}{coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias}

\item {} 
\sphinxAtStartPar
Inter\sphinxhyphen{}box swap using \sphinxhref{https://pubs.acs.org/doi/abs/10.1021/jp984742e}{coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias}

\item {} 
\sphinxAtStartPar
Inter\sphinxhyphen{}box targeted swap using \sphinxhref{https://pubs.acs.org/doi/abs/10.1021/jp984742e}{coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias}

\item {} 
\sphinxAtStartPar
Inter\sphinxhyphen{}box \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0378381218305351}{molecular exchange monter carlo}

\item {} 
\sphinxAtStartPar
Volume exchange (both isotropic and anisotropic)

\end{itemize}


\section{GOMC supported force fields:}
\label{\detokenize{introduction:gomc-supported-force-fields}}\begin{itemize}
\item {} 
\sphinxAtStartPar
OPLS

\item {} 
\sphinxAtStartPar
CHARMM

\item {} 
\sphinxAtStartPar
TraPPE

\item {} 
\sphinxAtStartPar
Mie

\item {} 
\sphinxAtStartPar
Martini

\end{itemize}


\section{GOMC supported molecules:}
\label{\detokenize{introduction:gomc-supported-molecules}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Polar molecules (using Ewald summation)

\item {} 
\sphinxAtStartPar
Non\sphinxhyphen{}polar molecules (standard LJ and Mie potential)

\item {} 
\sphinxAtStartPar
Linear molecules (using \sphinxhref{https://pubs.acs.org/doi/abs/10.1021/jp984742e}{coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias})

\item {} 
\sphinxAtStartPar
Branched molecules (using \sphinxhref{https://pubs.acs.org/doi/abs/10.1021/jp984742e}{coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias})

\item {} 
\sphinxAtStartPar
Cyclic molecules (using combination of \sphinxhref{https://pubs.acs.org/doi/abs/10.1021/jp984742e}{coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias} and \sphinxhref{https://aip.scitation.org/doi/abs/10.1063/1.3644939}{crankshaft} to sample intramolecular degrees of freedom of cyclic molecules)

\item {} 
\sphinxAtStartPar
Large biomolecules can be loaded into GOMC (although current sampling is limited to \sphinxhref{https://aip.scitation.org/doi/abs/10.1063/1.3644939}{crankshaft} to sample intramolecular degrees of freedom)

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
Biomolecules often have defined secondary structure which is maintained through improper terms, CMAP, and missing angles and dihedrals.

\item {} 
\sphinxAtStartPar
These complexities make sampling incorrect (improper, CMAP) or impossible (missing angles and dihedrals) in GOMC and these molecules should be held fixed.

\end{itemize}
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
It is important to start the simulation with correct molecular geometry such as correct bond length, angles, and dihedral.

\item {} 
\sphinxAtStartPar
In GOMC if the defined bond length in \sphinxcode{\sphinxupquote{Parameter}} file is different from calculated bond length in \sphinxcode{\sphinxupquote{PDB}} files by more than 0.02 \(Å\), you will receive a warning message with detailed information (box, residue id, specified bond length, and calculated bond length)

\end{itemize}
\end{sphinxadmonition}

\begin{sphinxadmonition}{important}{Important:}\begin{itemize}
\item {} 
\sphinxAtStartPar
Molecular geometry of \sphinxcode{\sphinxupquote{Linear}} and \sphinxcode{\sphinxupquote{Branched}} molecules will be corrected during the simulation by using the Monte Carlo moves that uses coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias method, such as \sphinxcode{\sphinxupquote{Regrowth}}, \sphinxcode{\sphinxupquote{Intra\sphinxhyphen{}box swap}}, and \sphinxcode{\sphinxupquote{Inter\sphinxhyphen{}box swap}}.

\end{itemize}
\end{sphinxadmonition}

\begin{sphinxadmonition}{warning}{Warning:}\begin{itemize}
\item {} 
\sphinxAtStartPar
Bond length of the \sphinxcode{\sphinxupquote{Cyclic}} molecules that belong to the body of rings will never be changed. Incorrect bond length may result in incorrect simulation results.

\item {} 
\sphinxAtStartPar
To sample the angles and dihedrals of a \sphinxcode{\sphinxupquote{Cyclic}} molecule that belongs to the body of the ring, \sphinxcode{\sphinxupquote{Regrowth}} or \sphinxcode{\sphinxupquote{Crankshaft}} Monte Carlo move must be used.

\item {} 
\sphinxAtStartPar
Any atom or group attached to the body of the ring, will uses coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias to sample the molecular geometry.

\item {} 
\sphinxAtStartPar
Flexible \sphinxcode{\sphinxupquote{Cyclic}} molecules with multiple rings (3 or more) that share edges (e.g. tricyclic), are not supported in GOMC. This is due the fact that no \sphinxcode{\sphinxupquote{Crankshaft}} move can alter the angle or dihedral of this atom, without changing the bond length.

\end{itemize}
\end{sphinxadmonition}


\chapter{Software Requirements}
\label{\detokenize{software_requirements:software-requirements}}\label{\detokenize{software_requirements::doc}}

\section{C++11 Compliant Compiler}
\label{\detokenize{software_requirements:c-11-compliant-compiler}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Linux/macOS
\begin{itemize}
\item {} 
\sphinxAtStartPar
icpc (Intel C++ Compiler)

\sphinxAtStartPar
In Linux, the Intel compiler will generally produce the fastest CPU executables (when running on Intel Core processors). Type the following command in a terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} icpc \PYGZhy{}\PYGZhy{}version
\end{sphinxVerbatim}

\sphinxAtStartPar
If gives a version number 16.0.3 (2016 Initial version) or later, you’re all set. Otherwise, we recommend upgrading.

\item {} 
\sphinxAtStartPar
g++

\sphinxAtStartPar
Type the following command in a terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} g++ \PYGZhy{}\PYGZhy{}version
\end{sphinxVerbatim}

\sphinxAtStartPar
If gives a version number 4.4 or later, you’re all set. Otherwise, we recommend upgrading.

\end{itemize}

\item {} 
\sphinxAtStartPar
Windows

\sphinxAtStartPar
Visual Studio Microsoft’s Visual Studio 2010 or later is recommended.

\sphinxAtStartPar
To check the version:

\sphinxAtStartPar
\sphinxstyleemphasis{Help} (top tab) \sphinxhyphen{}\textgreater{} \sphinxstyleemphasis{About Microsoft Visual Studio}

\noindent\sphinxincludegraphics{{vshelp}.png}

\noindent\sphinxincludegraphics{{vsabout}.png}

\end{itemize}


\section{CMake}
\label{\detokenize{software_requirements:cmake}}
\sphinxAtStartPar
To check if cmake is installed:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} which cmake
\end{sphinxVerbatim}

\sphinxAtStartPar
To check the version number:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} cmake \PYGZhy{}\PYGZhy{}version
\end{sphinxVerbatim}

\sphinxAtStartPar
The minimum required version is 2.8. However, we recommend to use version 3.2 or later.


\section{CUDA Toolkit}
\label{\detokenize{software_requirements:cuda-toolkit}}
\sphinxAtStartPar
CUDA is required to compile the GPU executable in both Windows and Linux. However, is not required to compile the CPU code. To download and install CUDA visit NVIDIA’s webpage:

\sphinxAtStartPar
\sphinxurl{https://developer.nvidia.com/cuda-downloads}

\sphinxAtStartPar
\sphinxurl{https://developer.nvidia.com/cuda}

\sphinxAtStartPar
Please refer to CUDA Developer webpages to select an appropriate version for the desired platform. To install CUDA in Linux root/sudo, privileges are generally required. In Windows, administrative access is required.

\sphinxAtStartPar
To check if nvcc is installed:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} which nvcc
\end{sphinxVerbatim}

\sphinxAtStartPar
To check the version number:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} nvcc \PYGZhy{}\PYGZhy{}version
\end{sphinxVerbatim}

\sphinxAtStartPar
The GPU builds of the code requires NVIDIA’s CUDA 8.0 or newer.


\section{MPI (Optional for Standard; Required for MultiSim)}
\label{\detokenize{software_requirements:mpi-optional-for-standard-required-for-multisim}}
\sphinxAtStartPar
An MPI Library is required to compile the MPI version of GOMC in both Windows and Linux.  However, it is not required to compile standard GOMC.  There are a couple of options to install an MPI library.
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{)}%
\item {} 
\sphinxAtStartPar
We recommend the Intel MPI Library:

\end{enumerate}
\begin{quote}

\sphinxAtStartPar
\sphinxurl{https://software.intel.com/en-us/mpi-library}
\end{quote}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{)}%
\setcounter{enumi}{1}
\item {} 
\sphinxAtStartPar
The alternative we recommend to Intel MPI is MPICH. MPICH binary packages are available in many UNIX distributions and for Windows. For example, you can search for it using “yum” (on Fedora), “apt” (Debian/Ubuntu), “pkg\_add” (FreeBSD) or “port”/”brew” (Mac OS).

\end{enumerate}
\begin{quote}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} sudo apt\PYGZhy{}get install mpich
\end{sphinxVerbatim}
\end{quote}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{)}%
\setcounter{enumi}{2}
\item {} 
\sphinxAtStartPar
Another option is the OpenMPI library.

\end{enumerate}
\begin{quote}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} sudo apt\PYGZhy{}get install openmpi\PYGZhy{}bin openmpi\PYGZhy{}common openssh\PYGZhy{}client openssh\PYGZhy{}server libopenmpi2 libopenmpi\PYGZhy{}dev
\end{sphinxVerbatim}
\end{quote}


\chapter{Recommended Software Tools}
\label{\detokenize{software_tools:recommended-software-tools}}\label{\detokenize{software_tools::doc}}
\sphinxAtStartPar
The listed programs are used in this manual and are generally considered necessary.


\section{Packmol}
\label{\detokenize{software_tools:packmol}}
\sphinxAtStartPar
Packmol is a free molecule packing tool (written in Fortran), created by José Mario Martínez, a professor of mathematics at the State University of Campinas, Brazil. Packmol allows a specified number of molecules to be packed at defined separating distances within a certain region of space. More information regarding downloading and installing Packmol is available on their homepage:

\sphinxAtStartPar
\sphinxurl{http://www.ime.unicamp.br/~martinez/packmol}

\begin{sphinxadmonition}{warning}{Warning:}
\sphinxAtStartPar
One of Packmol’s limitations is that it is unaware of topology; it treats each molecule or group of molecules as a rigid set of points. It is highly suggested to used the optimized structure of the molecule as the input file to packmol.
\end{sphinxadmonition}

\begin{sphinxadmonition}{warning}{Warning:}
\sphinxAtStartPar
Another more serious limitation is that it is not aware of periodic boundary conditions (PBC). As a result, when using Packmol to pack PDBs for GOMC, it is recommended to pack to a box 1\sphinxhyphen{}2 Angstroms smaller than the simulation box size. This prevents hard overlaps over the periodic boundary.
\end{sphinxadmonition}


\section{VMD}
\label{\detokenize{software_tools:vmd}}
\sphinxAtStartPar
VMD (Visual Molecular Dynamics) is a 3\sphinxhyphen{}D visualization and manipulation engine for molecular systems written in C\sphinxhyphen{}language. VMD is distributed and maintained by the University of Illinois at Urbana\sphinxhyphen{}Champaign. Its sources and binaries are free to download. It comes with a robust scripting engine, which is capable of running python and tcl scripts. More info can be found here:

\sphinxAtStartPar
\sphinxurl{http://www.ks.uiuc.edu/Research/vmd/}

\sphinxAtStartPar
Although GOMC uses the same fundamental file types, PDB (coordinates) and PSF (topology) as VMD, it uses some special tricks to obey certain rules of those file formats. One useful purpose of VMD is visualization and analyze your systems.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{vmd}.png}
\caption{A system of united atom isobutane molecules}\label{\detokenize{software_tools:id1}}\end{figure}

\sphinxAtStartPar
Nonetheless, the most critical part of VMD is a tool called PSFGen. PSFGen uses a tcl or python script to generate a PDB and PSF file for a system of one or more molecules. It is, perhaps, the most convenient way to generate a compliant PSF file.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{psfgen}.png}
\caption{An overview of the PSFGen file generation process and its relationship to VMD/NAMD}\label{\detokenize{software_tools:id2}}\end{figure}

\begin{sphinxadmonition}{tip}{Tip:}
\sphinxAtStartPar
To read more about PSFGen, reference:

\sphinxAtStartPar
\sphinxhref{http://www.ks.uiuc.edu/Research/vmd/plugins/psfgen}{Plugin homepage @ UIUC}

\sphinxAtStartPar
\sphinxhref{http://www.ks.uiuc.edu/Training/Tutorials/namd/namd-tutorial-html/node6.html}{Generating a Protein Structure File (PSF), part of the NAMD Tutorial from UIUC}

\sphinxAtStartPar
\sphinxhref{http://www.ks.uiuc.edu/Research/vmd/plugins/psfgen/ug.pdf}{In\sphinxhyphen{}Depth Overview {[}PDF{]}}
\end{sphinxadmonition}


\section{Molecular Simulation Design Framework (MoSDeF)}
\label{\detokenize{software_tools:molecular-simulation-design-framework-mosdef}}
\sphinxAtStartPar
In this section, the MosDef python interface for creating customized GOMC simulations are discussed.

\sphinxAtStartPar
Link to documentation: \sphinxurl{https://mbuild.mosdef.org/en/stable/getting\_started/writers/GOMC\_file\_writers.html}

\sphinxAtStartPar
Link to MosDef Examples Repository: \sphinxurl{https://github.com/GOMC-WSU/GOMC-MoSDeF}

\sphinxAtStartPar
Link to Youtube tutorials : \sphinxurl{https://www.youtube.com/playlist?list=PLdxD0z6HRx8Y9VhwcODxAHNQBBJDRvxMf}

\sphinxAtStartPar
Link to signac documentation: \sphinxurl{https://signac.io/}

\sphinxAtStartPar
Link to MosDef documentation: \sphinxurl{https://mosdef.org/}

\sphinxAtStartPar
The Molecular Simulation Design Framework (MosDeF) is a GOMC\sphinxhyphen{}compatible software that allows these simulations to be transparent and reproducible and permits the easy generation of all the required files to run a GOMC simulation (the forcefield, coordinate, topology and GOMC control files).  This mosdef\sphinxhyphen{}gomc package is available via conda.  The MoSDeF software also lowers the entry barrier for new users, minimizes the expert knowledge traditionally required to set up a simulation, and streamlines this process for more experienced users.  MoSDeF is comprised of several conda packages (mBuild, foyer, and gmso), which are stand\sphinxhyphen{}alone packages; however, they are designed to all work seamlessly together, which is the case with MoSDeF\sphinxhyphen{}GOMC.

\sphinxAtStartPar
In general, molecules imported or built using mBuild,  packed into a simulation box(s) and passed into the charmm writer function with additional parameters as arguments.  The charmm writer then atom\sphinxhyphen{}types the simulation box’s molecules using foyer, obtaining the molecular force field parameters.  The next step utilizes the charmm writer to output the forcefield files, PDB/PSF, and GOMC control files, all the files needed to run a GOMC simulation.  This MoSDeF\sphinxhyphen{}GOMC software is fully scriptable and compatible with Signac, allowing a fully automated and reproducible workflow.


\chapter{How to get the software}
\label{\detokenize{download:how-to-get-the-software}}\label{\detokenize{download::doc}}
\sphinxAtStartPar
The CPU and GPU code are merged together under GOMC project. Currently, version control is handled through the GitHub repository. The latest GOMC release, Example files, and User Manual can be downloaded from GOMC website or GitHub repository.


\section{GitHub}
\label{\detokenize{download:github}}
\sphinxAtStartPar
The posted builds in Master branch are “frozen” versions of the code that have been validated for a number of systems and ensembles. Other branches are created as a means of implementing new features. The latest updated code builds, manual, example files, and other resources can be obtained via the following GitHub repository:

\sphinxAtStartPar
\sphinxhref{https://github.com/GOMC-WSU}{GOMC GitHub Repository}

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics[width=1.000\linewidth]{{github}.png}
\end{figure}

\sphinxAtStartPar
GOMC and Examples repository can be found under the main page. Under GOMC repository, the code and manual can be found. Each repository can be downloaded by clicking on the Clone or download tab.

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics[width=1.000\linewidth]{{clone}.png}
\end{figure}

\sphinxAtStartPar
To clone the GOMC using git, execute the following command in your terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git clone https://github.com/GOMC\PYGZhy{}WSU/GOMC.git
\end{sphinxVerbatim}

\sphinxAtStartPar
To clone the GOMC Example files using git, execute the following command in your terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git clone https://github.com/GOMC\PYGZhy{}WSU/GOMC\PYGZus{}Examples.git
\end{sphinxVerbatim}


\section{Website}
\label{\detokenize{download:website}}
\sphinxAtStartPar
To access the GOMC website, please click on the following link:
\sphinxhref{http://gomc.eng.wayne.edu/}{GOMC Website}

\sphinxAtStartPar
The code can be found under the download tab, below and to the right of the logo. When new betas (or release builds) are announced, they will replace the prior code under the downloads tab. An announcement will be posted on the front page to notify users.

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics[width=1.000\linewidth]{{website}.jpg}
\end{figure}

\sphinxAtStartPar
GOMC is distributed as a compressed folder, containing the source and build system. To compile the code after downloading it, the first step is to extract the compressed build folder.

\sphinxAtStartPar
In Linux, the GPU and CPU codes are compressed using gzip and tar (*.tar.gz). To extract, simply move to the desire folder and type in the command line:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} tar \PYGZhy{}xzvf \PYGZlt{}file name\PYGZgt{}.tar.gz
\end{sphinxVerbatim}


\chapter{Compiling GOMC}
\label{\detokenize{compiling:compiling-gomc}}\label{\detokenize{compiling::doc}}
\sphinxAtStartPar
GOMC generates four executable files for CPU code; \sphinxcode{\sphinxupquote{GOMC\_CPU\_GEMC}} (Gibbs ensemble), \sphinxcode{\sphinxupquote{GOMC\_CPU\_NVT}} (NVT ensemble), \sphinxcode{\sphinxupquote{GOMC\_CPU\_NPT}} (isobaric\sphinxhyphen{}isothermal ensemble), and \sphinxcode{\sphinxupquote{GOMC\_CPU\_GCMC}} (Grand canonical ensemble). In case of installing CUDA Toolkit, GOMC will generate additional four executable files for GPU code; \sphinxcode{\sphinxupquote{GOMC\_GPU\_GEMC}}, \sphinxcode{\sphinxupquote{GOMC\_GPU\_NVT}}, \sphinxcode{\sphinxupquote{GOMC\_GPU NPT}}, and \sphinxcode{\sphinxupquote{GOMC\_GPU\_GCMC}}.

\sphinxAtStartPar
Each ensemble has a respective unit test executable \sphinxcode{\sphinxupquote{GOMC\_GEMC\_Test}} (Gibbs ensemble), \sphinxcode{\sphinxupquote{GOMC\_NVT\_Test}} (NVT ensemble), \sphinxcode{\sphinxupquote{GOMC\_NPT\_Test}} (isobaric\sphinxhyphen{}isothermal ensemble), and \sphinxcode{\sphinxupquote{GOMC\_GCMC\_Test}} (Grand canonical ensemble).  In case of installing CUDA Toolkit, GOMC will generate additional four unit test executables for GPU code; \sphinxcode{\sphinxupquote{GOMC\_GPU\_GEMC\_Test}}, \sphinxcode{\sphinxupquote{GOMC\_GPU\_NVT\_Test}}, \sphinxcode{\sphinxupquote{GOMC\_GPU NPT\textasciigrave{}\_Test\textasciigrave{}, and \textasciigrave{}\textasciigrave{}GOMC\_GPU\_GCMC\_Test}}.

\sphinxAtStartPar
This section guides users to compile GOMC in Linux or Windows.


\section{Linux}
\label{\detokenize{compiling:linux}}
\sphinxAtStartPar
First, navigate your command line to the GOMC base directory. To compile GOMC on Linux, give permission to “metamake.sh” by running the following command:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} chmod u+x metamake.sh
\end{sphinxVerbatim}

\sphinxAtStartPar
Metamake is the build script which creates a “bin” directory, configures and runs cmake file, and compiles the code as well. All executable files will be generated in the “bin” directory.  By default, GOMC compiles all ensembles with the Intel compiler (icc), if available.  Changes to this configuration can be made with options and arguments.

\sphinxAtStartPar
./metamake.sh {[}OPTIONS{]} {[}ARGUMENTS{]}

\sphinxAtStartPar
OPTIONS
\begin{quote}
\begin{optionlist}{3cm}
\item [\sphinxhyphen{}d]  
\sphinxAtStartPar
Compile in debug mode.
\item [\sphinxhyphen{}g]  
\sphinxAtStartPar
Compile with gcc.
\item [\sphinxhyphen{}m]  
\sphinxAtStartPar
Compile with MPI enabled.
\item [\sphinxhyphen{}p]  
\sphinxAtStartPar
Compile with NVTX profiling for CUDA
\item [\sphinxhyphen{}t]  
\sphinxAtStartPar
Compile Google tests.
\end{optionlist}
\end{quote}

\sphinxAtStartPar
ARGUMENTS
\begin{quote}

\sphinxAtStartPar
NVT
NPT
GCMC
GEMC

\sphinxAtStartPar
If CUDA Toolkit found:
\begin{quote}

\sphinxAtStartPar
GPU\_NVT
GPU\_NPT
GPU\_GCMC
GPU\_GEMC
\end{quote}

\sphinxAtStartPar
If testing enabled:
\begin{quote}

\sphinxAtStartPar
GOMC\_NVT\_Test
GOMC\_NPT\_Test
GOMC\_GCMC\_Test
GOMC\_GEMC\_Test

\sphinxAtStartPar
GOMC\_GPU\_NVT\_Test
GOMC\_GPU\_NPT\_Test
GOMC\_GPU\_GCMC\_Test
GOMC\_GPU\_GEMC\_Test
\end{quote}
\end{quote}


\section{Windows}
\label{\detokenize{compiling:windows}}
\sphinxAtStartPar
To compile GOMC on in Windows, follow these steps:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Open the Windows\sphinxhyphen{}compatible CMake GUI.

\item {} 
\sphinxAtStartPar
Set the Source Folder to the GOMC root folder.

\item {} 
\sphinxAtStartPar
Set the build Folder to your Build Folder.

\item {} 
\sphinxAtStartPar
Click configure, select your compiler/environment

\item {} 
\sphinxAtStartPar
Wait for CMake to finish the configuration.

\item {} 
\sphinxAtStartPar
Click configure again and click generate.

\item {} 
\sphinxAtStartPar
Open the CMake\sphinxhyphen{}generated project/solution etc. to the desired IDE (e.g Visual Studio).

\item {} 
\sphinxAtStartPar
Using the solution in the IDE of choice build GOMC per the IDE’s standard release compilation/exe\sphinxhyphen{} cutable generation methods.

\end{enumerate}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
You can also use CMake from the Windows command line if its directory is added to the PATH environment variable.
\end{sphinxadmonition}


\section{Configuring CMake}
\label{\detokenize{compiling:configuring-cmake}}
\sphinxAtStartPar
GOMC uses CMAKE to generate multi\sphinxhyphen{}platform intermediate files to compile the project. In this section, you can find all the information needed to configure CMake.
We recommend using a different directory for the CMake output than the home directory of the project as CMake tend to generate lots of files.

\sphinxAtStartPar
We recommend configuring CMake through metamake.sh OPTIONS, but CMake has a ridiculously expansive set of options which are not all configurable through metamake.  This document will only reproduce the most obviously relevant ones.  When possible, options should be passed into CMake via command line options rather than the CMakeCached.txt file:
\begin{description}
\item[{CMAKE\_BUILD\_TYPE}] \leavevmode
\sphinxAtStartPar
To get the best performance you should build the project in release mode. In CMake GUI you can set the value of “CMAKE\_BUILD\_TYPE” to “Release” and in CMake command line you can add the following to the CMake:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZhy{}DCMAKE\PYGZus{}BUILD\PYGZus{}TYPE\PYG{o}{=}Release
\end{sphinxVerbatim}

\sphinxAtStartPar
To compile the GOMC in debug mode, in CMake GUI, change the value of “CMAKE\_BUILD\_TYPE” to “Debug” and in CMake command line you can add the following to the CMake:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZhy{}DCMAKE\PYGZus{}BUILD\PYGZus{}TYPE\PYG{o}{=}Debug
\end{sphinxVerbatim}

\sphinxAtStartPar
Other options are “\textless{}None | ReleaseWithDebInfo | MinSizeRel\textgreater{}”.

\item[{CMAKE\_CXX\_COMPILER}] \leavevmode
\sphinxAtStartPar
This option will set the compiler. It is recommended to use the Intel Compiler and linking tools, if possible (icc/icpc/etc.). They significantly outperform the default GNU and Visual Studio compiler tools and are available for free for academic use with registration.

\item[{CMAKE\_CXX\_FLAGS\_RELEASE:STRING}] \leavevmode
\sphinxAtStartPar
To run the parallel version of CPU code, it needs to be compiled with openmp library. Open the file “CMakeCache.txt”, while still in the “bin” folder, and change the value from “\sphinxhyphen{}O3 \sphinxhyphen{}DNDEBUG” to “\sphinxhyphen{}O3 \sphinxhyphen{}qopenmp \sphinxhyphen{}DNDEBUG”. Recompile the GOMC by typing the command:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} make
\end{sphinxVerbatim}

\item[{ENSEMBLE\_NVT}] \leavevmode
\sphinxAtStartPar
You can turn the compilation of CPU version of NVT ensemble on or off using this option.
\sphinxhyphen{}DENSEMBLE\_NVT=\textless{}On | Off\textgreater{}

\item[{ENSEMBLE\_NPT}] \leavevmode
\sphinxAtStartPar
You can turn the compilation of CPU version of NPT ensemble on or off using this option.
\sphinxhyphen{}DENSEMBLE\_NPT=\textless{}On | Off\textgreater{}

\item[{ENSEMBLE\_GCMC}] \leavevmode
\sphinxAtStartPar
You can turn the compilation of CPU version of GCMC ensemble on or off using this option.
\sphinxhyphen{}DENSEMBLE\_GCMC=\textless{}On | Off\textgreater{}

\item[{ENSEMBLE\_GEMC}] \leavevmode
\sphinxAtStartPar
You can turn the compilation of CPU version of GEMC ensemble on or off using this option.
\sphinxhyphen{}DENSEMBLE\_GEMC=\textless{}On | Off\textgreater{}

\item[{ENSEMBLE\_GPU\_NVT}] \leavevmode
\sphinxAtStartPar
You can turn the compilation of GPU version of NVT ensemble on or off using this option.
\sphinxhyphen{}DENSEMBLE\_NVT=\textless{}On | Off\textgreater{}

\item[{ENSEMBLE\_GPU\_NPT}] \leavevmode
\sphinxAtStartPar
You can turn the compilation of GPU version of NPT ensemble on or off using this option.
\sphinxhyphen{}DENSEMBLE\_NPT=\textless{}On | Off\textgreater{}

\item[{ENSEMBLE\_GPU\_GCMC}] \leavevmode
\sphinxAtStartPar
You can turn the compilation of GPU version of GCMC ensemble on or off using this option.
\sphinxhyphen{}DENSEMBLE\_GCMC=\textless{}On | Off\textgreater{}

\item[{ENSEMBLE\_GPU\_GEMC}] \leavevmode
\sphinxAtStartPar
You can turn the compilation of GPU version of GEMC ensemble on or off using this option.
\sphinxhyphen{}DENSEMBLE\_GEMC=\textless{}On | Off\textgreater{}

\end{description}


\chapter{GPU\sphinxhyphen{}Accelerated GOMC}
\label{\detokenize{gpu_acceleration:gpu-accelerated-gomc}}\label{\detokenize{gpu_acceleration::doc}}\begin{description}
\item[{All moves use the following general GPU\sphinxhyphen{}Accelerated kernels:}] \leavevmode\begin{itemize}
\item {} 
\sphinxAtStartPar
All\sphinxhyphen{}molecule Intermolecular Lennard Jones and Coulombic Energy

\item {} 
\sphinxAtStartPar
All\sphinxhyphen{}molecule Intermolecular Reciprocal Space Energy

\item {} 
\sphinxAtStartPar
Image calculation for Ewald Summation

\item {} 
\sphinxAtStartPar
Minimum Image Calculation

\end{itemize}

\item[{GOMC currently supports several move\sphinxhyphen{}specific GPU\sphinxhyphen{}Accelerated kernels:}] \leavevmode
\end{description}


\chapter{Input File Formats}
\label{\detokenize{input_file:input-file-formats}}\label{\detokenize{input_file::doc}}
\sphinxAtStartPar
In order to run simulation in GOMC, the following files need to be provided:
\begin{itemize}
\item {} 
\sphinxAtStartPar
GOMC executable

\item {} 
\sphinxAtStartPar
PDB file(s)

\item {} 
\sphinxAtStartPar
PSF file(s)

\item {} 
\sphinxAtStartPar
Parameter file

\item {} 
\sphinxAtStartPar
Input file “NAME.conf” (proprietary control file)

\end{itemize}

\sphinxAtStartPar
In order to restart a simulation in GOMC from exactly where it left off, the following files also need to be provided:
\begin{itemize}
\item {} 
\sphinxAtStartPar
XSC file(s)

\item {} 
\sphinxAtStartPar
COOR file(s)

\item {} 
\sphinxAtStartPar
CHK file

\end{itemize}

\sphinxAtStartPar
In order to run a hybrid MCMD simulation, the following files also need to be provided:
\begin{itemize}
\item {} 
\sphinxAtStartPar
XSC file(s)

\item {} 
\sphinxAtStartPar
COOR file(s)

\item {} 
\sphinxAtStartPar
VEL files(s)

\item {} 
\sphinxAtStartPar
CHK file

\end{itemize}


\section{PDB File}
\label{\detokenize{input_file:pdb-file}}
\sphinxAtStartPar
GOMC requires only one PDB file for NVT and NPT ensembles. However, GOMC requires two PDB files for GEMC and GCMC ensembles.


\subsection{What is PDB file}
\label{\detokenize{input_file:what-is-pdb-file}}
\sphinxAtStartPar
The term PDB can refer to the Protein Data Bank (\sphinxurl{http://www.rcsb.org/pdb/}), to a data file provided there, or to any file following the PDB format.
Files in the PDB include various information such as the name of the compound, the ATOM and HETATM records containing the coordinates of the molecules, and etc.
PDB widely used by NAMD, GROMACS, CHARMM, ACEMD, and Amber. GOMC ignore everything in a PDB file except for the REMARK, CRYST1, ATOM, and END records.
An overview of the PDB standard can be found here:

\sphinxAtStartPar
\sphinxurl{http://www.wwpdb.org/documentation/file-format-content/format33/sect2.html\#HEADER}

\sphinxAtStartPar
\sphinxurl{http://www.wwpdb.org/documentation/file-format-content/format33/sect8.html\#CRYST1}

\sphinxAtStartPar
\sphinxurl{http://www.wwpdb.org/documentation/file-format-content/format33/sect9.html\#ATOM}

\sphinxAtStartPar
PDB contains four major parts; \sphinxcode{\sphinxupquote{REMARK}}, \sphinxcode{\sphinxupquote{CRYST1}}, \sphinxcode{\sphinxupquote{ATOM}}, and \sphinxcode{\sphinxupquote{END}}. Here is the definition of each field and how GOMC is using them to get the information it requires.
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{REMARK}}:
This header records present experimental  details, annotations, comments, and information not included in other records (for more information,
\sphinxhref{http://www.wwpdb.org/documentation/file-format-content/format33/sect2.html\#HEADER}{click here}).

\sphinxAtStartPar
However, GOMC uses this header to print simulation informations.
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Max Displacement} (Å)

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Max Rotation} (Degree)

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Max volume exchange} (\(\AA^3\))

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Monte Carlo Steps} (MC)

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{CRYST1}}:
This header records the unit \sphinxhref{http://www.wwpdb.org/documentation/file-format-content/format33/sect8.html\#CRYST1}{cell dimension parameters}.
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Lattice constant}: a,b,c (Å)

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Lattice angles}: \(\alpha, \beta, \gamma\) (Degree)

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{ATOM}}:
The \sphinxhref{http://www.wwpdb.org/documentation/file-format-content/format33/sect9.html\#ATOM}{ATOM} records present the atomic coordinates for standard amino acids
and nucleotides. They also present the occupancy and temperature factor for each atom.
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{ATOM}: Record name

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{serial}: Atom serial number.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{name}: Atom name.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{resName}: Residue name.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{chainID}: Chain identifier.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{resSeq}: Residue sequence number.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{x}: Coordinates for X (Å).

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{y}: Coordinates for Y (Å).

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{z}: Coordinates for Z (Å).

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{occupancy}: GOMC uses to define which atoms belong to which box.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{beta}: Beta or Temperature factor. GOMC uses this value to define the mobility of the atoms. element: Element symbol.

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{END}}:
A frame in the PDB file is terminated with the keyword.

\end{itemize}

\sphinxAtStartPar
Here are the PDB output of GOMC for the first molecule of isobutane:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
REMARK    GOMC   122.790    3.14159     3439.817     1000000
CRYST1  35.245    35.245    35.245    90.00   90.00   90.00
ATOM    1     C1    ISB     1     0.911    \PYGZhy{}0.313    0.000    0.00    0.00    C
ATOM    2     C1    ISB     1     1.424    \PYGZhy{}1.765    0.000    0.00    0.00    C
ATOM    3     C1    ISB     1    \PYGZhy{}0.629    \PYGZhy{}0.313    0.000    0.00    0.00    C
ATOM    4     C1    ISB     1     1.424     0.413   \PYGZhy{}1.257    0.00    0.00    C
END
\end{sphinxVerbatim}

\sphinxAtStartPar
The fields seen here in order from left to right are the record type, atom ID, atom name, residue name, residue ID, x, y, and z coordinates, occupancy, temperature factor (called beta), and segment name.

\sphinxAtStartPar
The atom name is “C1” and residue name is “ISB”. The PSF file (next section) contains a lookup table of atoms. These contain the atom name from the PDB and
the name of the atom kind in the parameter file it corresponds to. As multiple different atom names will all correspond to the same parameter,
these can be viewed “atom aliases” of sorts. The chain letter (in this case ‘A’) is sometimes used when packing a number of PDBs into a single PDB file.

\begin{sphinxadmonition}{important}{Important:}\begin{itemize}
\item {} 
\sphinxAtStartPar
VMD requires a constant number of ATOMs in a multi\sphinxhyphen{}frame PDB (multiple records terminated by “END” in a single file). To compensate for this, all atoms
from all boxes in the system are written to the output PDBs of this code.

\item {} 
\sphinxAtStartPar
For atoms not currently in a box, the coordinates are set to \sphinxcode{\sphinxupquote{\textless{} 0.00, 0.00, 0.00 \textgreater{}}}. The occupancy is commonly just set to “1.00” and is left unused by
many codes. We recycle this legacy parameter by using it to denote, in our output PDBs, the box a molecule is in (box 0 occupancy=0.00 ; box 1 occupancy=1.00)

\item {} 
\sphinxAtStartPar
The beta value in GOMC code is used to define the mobility of the molecule.
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{Beta = 0.00}}: molecule can move and transfer within and between boxes.

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{Beta = 1.00}}: molecule is fixed in its position.

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{Beta = 2.00}}: molecule can move within the box but cannot be transferred between boxes.

\end{itemize}

\end{itemize}
\end{sphinxadmonition}


\subsection{Generating PDB file}
\label{\detokenize{input_file:generating-pdb-file}}
\sphinxAtStartPar
With that overview of the format in mind, the following steps describe how a PDB file is typically built.
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
A single molecule PDB is obtained. In this example, the GaussView was used to draw the molecule, which was then edited by hand to adhere
to the PDB spec properly. There are many open\sphinxhyphen{}source software that can build a molecule for you, such as \sphinxhref{https://avogadro.cc/docs/getting-started/drawing-molecules/}{Avagadro} ,
\sphinxhref{http://www.ks.uiuc.edu/Research/vmd/plugins/molefacture/}{molefacture} in VMD and more. The end result is a PDB for a single molecule:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
REMARK   1 File created by GaussView 5.0.8
ATOM     1  C1   ISB  1   0.911  \PYGZhy{}0.313    0.000  C
ATOM     2  C1   ISB  1   1.424  \PYGZhy{}1.765    0.000  C
ATOM     3  C1   ISB  1  \PYGZhy{}0.629  \PYGZhy{}0.313    0.000  C
ATOM     4  C1   ISB  1   1.424   0.413   \PYGZhy{}1.257  C
END
\end{sphinxVerbatim}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{1}
\item {} 
\sphinxAtStartPar
Next, packings are calculated to place the simulation in a region of vapor\sphinxhyphen{}liquid coexistence. There are a couple of ways to do this in Gibbs ensemble:

\end{enumerate}
\begin{itemize}
\item {} 
\sphinxAtStartPar
Pack both boxes to a single middle density, which is an average of the liquid and vapor densities.

\item {} 
\sphinxAtStartPar
Same as previous method, but add a modest amount to axis of one box (e.g. 10\sphinxhyphen{}30 A). This technique can be handy in the constant pressure Gibbs ensemble.

\item {} 
\sphinxAtStartPar
Pack one box to the predicted liquid density and the other to the vapor density.

\sphinxAtStartPar
A good reference for getting the information needed to estimate packing is the NIST Web Book database of pure compounds:

\sphinxAtStartPar
\sphinxurl{http://webbook.nist.gov/chemistry/}

\end{itemize}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{2}
\item {} 
\sphinxAtStartPar
After packing is determined, a basic pack can be performed with a Packmol script. Here is the example of packing 1000 isobutane in 70 A cubic box:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
tolerance   3.0
filetype    pdb
output      STEP2\PYGZus{}ISB\PYGZus{}packed\PYGZus{}BOX 0.pdb
structure   isobutane.pdb
number      1000
inside cube 0.1   0.1   0.1   70.20
end     structure
\end{sphinxVerbatim}

\sphinxAtStartPar
Copy the above text into “pack\_isobutane.inp” file, save it and run the script by typing the following line into the terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./packmol \PYGZlt{} pack\PYGZus{}isobutane.inp
\end{sphinxVerbatim}


\section{PSF File}
\label{\detokenize{input_file:psf-file}}
\sphinxAtStartPar
GOMC requires only one PSF file for NVT and NPT ensembles. However, GOMC requires two PSF files for GEMC and GCMC ensembles.


\subsection{What is PSF file}
\label{\detokenize{input_file:what-is-psf-file}}
\sphinxAtStartPar
Protein structure file (PSF), contains all of the molecule\sphinxhyphen{}specific information needed to apply a particular force field to a molecular system.
The CHARMM force field is divided into a topology file, which is needed to generate the PSF file, and a parameter file, which supplies specific numerical
values for the generic CHARMM potential function. The topology file defines the atom types used in the force field; the atom names, types, bonds, and partial
charges of each residue type; and any patches necessary to link or otherwise mutate these basic residues. The parameter file provides a mapping between bonded
and nonbonded interactions involving the various combinations of atom types found in the topology file and specific spring constants and similar parameters for
all of the bond, angle, dihedral, improper, and van der Waals terms in the CHARMM potential function. PSF file widely used by by NAMD, CHARMM, and X\sphinxhyphen{}PLOR.

\sphinxAtStartPar
The PSF file contains six main sections: \sphinxcode{\sphinxupquote{remarks}}, \sphinxcode{\sphinxupquote{atoms}}, \sphinxcode{\sphinxupquote{bonds}}, \sphinxcode{\sphinxupquote{angles}}, \sphinxcode{\sphinxupquote{dihedrals}}, and \sphinxcode{\sphinxupquote{impropers}} (dihedral force terms used to maintain
planarity). Each section starts with a specific header described bellow:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NTITLE}}: remarks on the file.
The following is taken from a PSF file for isobutane:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
PSF
      3  !NTITLE
REMARKS  original generated structure x\PYGZhy{}plor psf file
REMARKS  topology ./Top\PYGZus{}Branched\PYGZus{}Alkanes.inp
REMARKS  segment ISB \PYGZob{} first NONE; last NONE; auto angles dihedrals \PYGZcb{}
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NATOM}}: Defines the atom names, types, and partial charges of each residue type.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
atom    ID
segment name
residue ID
residue name
atom    name
atom    type
atom    charge
atom    mass
\end{sphinxVerbatim}

\sphinxAtStartPar
The following is taken from a PSF file for isobutane:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
4000 !NATOM
1    ISB  1  ISB    C1    CH1    0.000000   13.0190  0
2    ISB  1  ISB    C2    CH3    0.000000   15.0350  0
3    ISB  1  ISB    C3    CH3    0.000000   15.0350  0
4    ISB  1  ISB    C4    CH3    0.000000   15.0350  0
5    ISB  2  ISB    C1    CH1    0.000000   13.0190  0
6    ISB  2  ISB    C2    CH3    0.000000   15.0350  0
7    ISB  2  ISB    C3    CH3    0.000000   15.0350  0
8    ISB  2  ISB    C4    CH3    0.000000   15.0350  0
\end{sphinxVerbatim}

\sphinxAtStartPar
The fields in the atom section, from left to right are atom ID, segment name, residue ID, residue name, atom name, atom type, charge, mass, and an unused 0.

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NBOND}}: The covalent bond section lists four pairs of atoms per line. The following is taken from a PSF file for isobutane:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
3000   !BOND:     bonds
1   2   1   3   1   4   5   6
5   7   5   8
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NTHETA}}: The angle section lists three triples of atoms per line. The following is taken from a PSF file for isobutane:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
3000   !NTHETA:   angles
2   1   4   2   1   3   3   1   4
6   5   8   6   5   7   7   5   8
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NPHI}}: The dihedral sections list two quadruples of atoms per line.

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NIMPHI}}: The improper sections list two quadruples of atoms per line. GOMC currently does not support improper. For the molecules without dihedral or improper, PDF file look like the following:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
0   !NPHI: dihedrals
0   !NIMPHI: impropers
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
(other sections such as cross terms)

\end{itemize}

\begin{sphinxadmonition}{important}{Important:}\begin{itemize}
\item {} 
\sphinxAtStartPar
The PSF file format is a highly redundant file format. It repeats identical topology of thousands of molecules of a common kind in some cases. GOMC follows the same approach as NAMD, allowing this excess information externally and compiling it in the code.

\item {} 
\sphinxAtStartPar
Other sections (e.g. cross terms) contain unsupported or legacy parameters and are ignored.

\item {} 
\sphinxAtStartPar
Following the restriction of VMD, the order of the atoms in PSF file must match the order of the atoms in the PDB file.

\item {} 
\sphinxAtStartPar
Improper entries are read and stored, but are not currently used. Support will eventually be added for this.

\end{itemize}
\end{sphinxadmonition}


\subsection{Generating PSF file}
\label{\detokenize{input_file:generating-psf-file}}
\sphinxAtStartPar
The PSF file is typically generated using PSFGen. It is convenient to make a script, such as the example below, to do this:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
package require psfgen
topology  ./Top\PYGZus{}branched\PYGZus{}Alaknes.inp
segment ISB \PYGZob{}
  pdb   ./STEP2\PYGZus{}ISB\PYGZus{}packed\PYGZus{}BOX 0.pdb
  first   none
  last  none
\PYGZcb{}

coordpdb ./STEP2\PYGZus{}ISB\PYGZus{}packed\PYGZus{}BOX 0.pdb ISB

writepsf ./STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}0.psf
writepdb ./STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}0.pdb
\end{sphinxVerbatim}

\sphinxAtStartPar
Typically, one script is run per box to generate a finalized PDB/PSF for that box. The script requires one additional file, the NAMD\sphinxhyphen{}style topology file. While GOMC does not directly read or interact with this file, it’s typically used to generate the PSF and, hence, is considered one of the integral file types. It will be briefly discussed in the following section.


\section{Topology File}
\label{\detokenize{input_file:topology-file}}
\sphinxAtStartPar
A CHARMM forcefield topology file contains all of the information needed to convert a list of residue names into a complete PSF structure file. The topology is a whitespace separated file format, which contains a list of atoms and their corresponding masses, and a list of residue information (charges, composition, and topology). Essentially, it is a non\sphinxhyphen{}redundant lookup table equivalent to the PSF file.

\sphinxAtStartPar
This is followed by a series of residues, which tell PSFGen what atoms are bonded to a given atom. Each residue is comprised of four key elements:
\begin{itemize}
\item {} 
\sphinxAtStartPar
A header beginning with the keyword RESI with the residue name and net charge

\item {} 
\sphinxAtStartPar
A body with multiple ATOM entries (not to be confused with the PDB\sphinxhyphen{}style entries of the same name), which list the partial charge on the particle and what kind of atom each named atom in a specific molecule/residue is.

\item {} 
\sphinxAtStartPar
A section of lines starting with the word BOND contains pairs of bonded atoms (typically 3 per line)

\item {} 
\sphinxAtStartPar
A closing section with instructions for PSFGen.

\end{itemize}

\sphinxAtStartPar
Here’s an example of topology file for isobutane:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
* Custom top file \PYGZhy{}\PYGZhy{} branched alkanes *
11
!
MASS 1 CH3 15.035 C !
MASS 2 CH1 13.019 C !

AUTOGENERATE ANGLES DIHEDRALS

RESI ISB    0.00 !  isobutane \PYGZhy{} TraPPE
GROUP
ATOM  C1  CH1   0.00 !  C3\PYGZbs{}
ATOM  C2  CH3   0.00 !     C1\PYGZhy{}C2
ATOM  C3  CH3   0.00 !  C4/
ATOM  C4  CH3   0.00 !
BOND  C1  C2  C1  C3  C1  C4
PATCHING FIRS NONE LAST NONE

END
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The keyword END must be used to terminate this file and keywords related to the auto\sphinxhyphen{}generation process must be placed near the top of the file, after the MASS definitions.
\end{sphinxadmonition}

\begin{sphinxadmonition}{tip}{Tip:}
\sphinxAtStartPar
More in\sphinxhyphen{}depth information can be found in the following links:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{http://www.ks.uiuc.edu/Training/Tutorials/science/topology/topology-tutorial.pdf}{Topology Tutorial}

\end{itemize}
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{http://www.ks.uiuc.edu/Training/Tutorials/science/topology/topology-html/node4.html}{NAMD Tutorial: Examining the Topology File}

\end{itemize}
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{http://www.ks.uiuc.edu/Training/Tutorials/science/forcefield-tutorial/forcefield-html/node6.html}{Developing Topology and Parameter Files}

\end{itemize}
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{http://www.ks.uiuc.edu/Training/Tutorials/namd/namd-tutorial-win-html/node25.html}{NAMD Tutorial: Topology Files}

\end{itemize}
\end{sphinxadmonition}


\section{Parameter File}
\label{\detokenize{input_file:parameter-file}}
\sphinxAtStartPar
Currently, GOMC uses a single parameter file and the user has the two kinds of parameter file choices:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{CHARMM}} (Chemistry at Harvard Molecular Mechanics) compatible parameter file

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{EXOTIC}} or \sphinxcode{\sphinxupquote{Mie}} parameter file

\end{itemize}

\sphinxAtStartPar
If the parameter file type is not specified or if the chosen file is missing, an error will result.

\sphinxAtStartPar
Both force field file options are whitespace separated files with sections preceded by a tag. When a known tag (representing a molecular interaction in the model) is encountered, reading of that section of the force field begins. Comments (anything after a \sphinxcode{\sphinxupquote{*}} or \sphinxcode{\sphinxupquote{!}}) and whitespace are ignored. Reading concludes when the end of the file is reached or another section tag is encountered.


\subsection{CHARMM format parameter file}
\label{\detokenize{input_file:charmm-format-parameter-file}}
\sphinxAtStartPar
CHARMM contains a widely used model for describing energies in Monte Carlo and molecular dynamics simulations. It is intended to be compatible with other codes that use such a format, such as NAMD. See \sphinxhref{http://www.charmmtutorial.org/index.php/The\_Energy\_Function}{here} for a general overview of the CHARMM force field.

\sphinxAtStartPar
Here’s the basic CHARMM contributions that are supported in GOMC:
\begin{equation*}
\begin{split}U_{\texttt{bond}}&=\sum_{\texttt{bonds}} K_b(b-b_0)^2\\
U_{\texttt{angle}}&=\sum_{\texttt{angles}} K_{\theta}(\theta-\theta_0)^2\\
U_{\texttt{dihedral}}&=\sum_{\texttt{dihedrals}} K_{\phi} [1+\cos(n\phi - \delta)]\\
U_{\texttt{LJ}}&=\sum_{\texttt{nonbonded}} \epsilon_{ij}\left[\left(\frac{R_{min_{ij}}}{r_{ij}}\right)^{12}-2\left(\frac{R_{min_{ij}}}{r_{ij}}\right)^6\right]+ \frac{q_i q_j}{\epsilon r_{ij}}\end{split}
\end{equation*}
\sphinxAtStartPar
As seen above, the following are recognized, read and used:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{BONDS}}
\sphinxhyphen{} Quadratic expression describing bond stretching based on bond length (b) in Angstrom
\textendash{} Typically, it is ignored as bonds are rigid for Monte Carlo simulations.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
GOMC does not sample bond stretch. To ignore the relative bond energy, set the \(K_b\) to a large value i.e. “999999999999”.
\end{sphinxadmonition}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{bonds}.png}
\caption{Oscillations about the equilibrium bond length}\label{\detokenize{input_file:id1}}\end{figure}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{ANGLES}}
\sphinxhyphen{} Describe the conformational behavior of an angle (\(\delta\)) between three atoms, one of which is shared branch point to the other two.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
To fix any angle and ignore the related angle energy, set the \(K_\theta\) to a large value i.e. “999999999999”.
\end{sphinxadmonition}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{angle}.png}
\caption{Oscillations of 3 atoms about an equilibrium bond angle}\label{\detokenize{input_file:id2}}\end{figure}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{DIHEDRALS}}
\sphinxhyphen{} Describes crankshaft\sphinxhyphen{}like rotation behavior about a central bond in a series of three consecutive bonds (rotation is given as \(\phi\)).

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{dihedrals}.png}
\caption{Torsional rotation of 4 atoms about a central bond}\label{\detokenize{input_file:id3}}\end{figure}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NONBONDED}}
\sphinxhyphen{} This tag name only should be used if CHARMM force files are being used. This section describes 12\sphinxhyphen{}6 (Lennard\sphinxhyphen{}Jones) non\sphinxhyphen{}bonded interactions. Non\sphinxhyphen{}bonded parameters are assigned by specifying atom type name followed by polarizabilities (which will be ignored), minimum energy, and (minimum radius)/2. In order to modify 1\sphinxhyphen{}4 interaction, a second polarizability (again, will be ignored), minimum energy, and (minimum radius)/2 need to be defined; otherwise, the same parameter will be considered for 1\sphinxhyphen{}4 interaction.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=1.000\linewidth]{{nonbonded}.png}
\caption{Non\sphinxhyphen{}bonded energy terms (electrostatics and Lennard\sphinxhyphen{}Jones)}\label{\detokenize{input_file:id4}}\end{figure}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NBFIX}}
\sphinxhyphen{} This tag name only should be used if CHARMM force field is being used. This section allows in\sphinxhyphen{} teraction between two pairs of atoms to be modified, done by specifying two atom type names followed by minimum energy and minimum radius. In order to modify 1\sphinxhyphen{}4 interaction, a second minimum energy and minimum radius need to be defined; otherwise, the same parameter will be considered for 1\sphinxhyphen{}4 interaction.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Please pay attention that in this section we define minimum radius, not (minimum radius)/2 as it is defined in the NONBONDED section.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
This does not modify the 1\sphinxhyphen{}4 electrostatic interactions.
\end{sphinxadmonition}

\sphinxAtStartPar
Currently, supported sections of the \sphinxcode{\sphinxupquote{CHARMM}} compliant file include \sphinxcode{\sphinxupquote{BONDS}}, \sphinxcode{\sphinxupquote{ANGLES}}, \sphinxcode{\sphinxupquote{DIHEDRALS}}, \sphinxcode{\sphinxupquote{NONBONDED}}, \sphinxcode{\sphinxupquote{NBFIX}}. Other sections such as \sphinxcode{\sphinxupquote{CMAP}} are not currently read or supported.

\end{itemize}


\subsection{BONDS}
\label{\detokenize{input_file:bonds}}
\sphinxAtStartPar
(“bond stretching”) is one key section of the CHARMM\sphinxhyphen{}compliant file. Units for the \(K_b\) variable in this section are in kcal/mol; the \(b_0\) section (which represents the equilibrium bond length for that kind of pair) is measured in Angstroms.
\begin{equation*}
\begin{split}U_{\texttt{bond}}&=\sum_{\texttt{bonds}} K_b(b-b_0)^2\\\end{split}
\end{equation*}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
BONDS
!V(bond) = Kb(b \PYGZhy{} b0)**2
!
!Kb:  kcal/mole/A**2
!b0:  A
!
!Kb (kcal/mol) = Kb (K) * Boltz.  const.;
!
!atom type      Kb          b0     description
CH3   CH1   9999999999    1.540 !  TraPPE 2
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The \(K_b\) value may appear odd, but this is because a larger value corresponds to a more rigid bond. As Monte Carlo force fields (e.g. TraPPE) typically treat molecules as rigid constructs, \(K_b\) is set to a large value \sphinxhyphen{} 9999999999. Sampling bond stretch is not supported in GOMC.
\end{sphinxadmonition}


\subsection{ANGLES}
\label{\detokenize{input_file:angles}}
\sphinxAtStartPar
(“bond bending”), where \(\theta\) is the measured bond angle and \(\theta_0\) is the equilibrium bond angle for that kind of pair, are commonly measured in degrees and \(K_\theta\) is the force constant measured in kcal/mol/K. These values, in literature, are often expressed in Kelvin (K).

\sphinxAtStartPar
To convert Kelvin to kcal/mol/K, multiply by the Boltzmann constant \textendash{} \(K_\theta\), 0.0019872041 kcal/mol. In order to fix the angle, it requires to set a large value for \(K_\theta\). By assigning a large value like 9999999999, specified angle will be fixed and energy of that angle will considered to be zero.
\begin{equation*}
\begin{split}U_{\texttt{angle}}&=\sum_{\texttt{angles}} K_{\theta}(\theta-\theta_0)^2\\\end{split}
\end{equation*}
\sphinxAtStartPar
Here is an example of what is necessary for isobutane:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
ANGLES
!
!V(angle) = Ktheta(Theta \PYGZhy{} Theta0)**2
!
!V(Urey\PYGZhy{}Bradley) = Kub(S \PYGZhy{} S0)**2
!
!Ktheta:  kcal/mole/rad**2
!Theta0:  degrees
!S0:  A
!
!Ktheta (kcal/mol) = Ktheta (K) * Boltz.  const.
!
!atom types         Ktheta        Theta0
CH3   CH1   CH3     62.100125     112.00 !  TraPPE 2
\end{sphinxVerbatim}

\sphinxAtStartPar
Some CHARMM ANGLES section entries include \sphinxcode{\sphinxupquote{Urey\sphinxhyphen{}Bradley}} potentials (\(K_{ub}\), \(b_{ub}\)), in addition to the standard quadratic angle potential. The constants related to this potential function are currently read, but the logic has not been added to calculate this potential function. Support for this potential function will be added in later versions of the code.


\subsection{DIHEDRALS}
\label{\detokenize{input_file:dihedrals}}
\sphinxAtStartPar
The final major bonded interactions section of the CHARMM compliant parameter file are the DIHEDRALS. Dihedral energies were represented by a cosine series where \(\phi\) is the dihedral angle, \(C_n\) are dihedral force constants, \(n\) is the multiplicity, and \(\delta_n\) is the phase shift.
Often, there are 4 to 6 terms in a dihedral. Angles for the dihedrals’ deltas are given in degrees.
\begin{equation*}
\begin{split}U_{\texttt{dihedral}}&= C_0 + \sum_{\texttt{n = 1}} C_n [1+\cos(n\phi_i - \delta_n)]\\\end{split}
\end{equation*}
\sphinxAtStartPar
Since isobutane has no dihedral, here are the parameters pertaining to 2,3\sphinxhyphen{}dimethylbutane:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
DIHEDRALS
!
!V(dihedral) = Kchi(1 + cos(n(chi) \PYGZhy{} delta))
!
!Kchi:  kcal/mole
!n:  multiplicity
!delta:  degrees
!
!Kchi (kcal/mol) = Kchi (K) * Boltz.  const.
!
!atom types             Kchi    n     delta   description
X   CH1   CH1   X    \PYGZhy{}0.498907  0     0.0   !  TraPPE 2
X   CH1   CH1   X     0.851974  1     0.0   !  TraPPE 2
X   CH1   CH1   X    \PYGZhy{}0.222269  2   180.0   !  TraPPE 2
X   CH1   CH1   X     0.876894  3     0.0   !  TraPPE 2
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The code allows the use of ‘X’ to indicate ambiguous positions on the ends. This is useful because this kind is often determined solely by the two middle atoms in the middle of the dihedral, according to literature.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
If a dihedral parameter was defined with multiplicity value of zero (\(n\) = 0), GOMC will automatically assign the phase shift value to 90 (\(\delta_n\) = 90) to recover the above dihedral expresion.
\end{sphinxadmonition}


\subsection{IMPROPERS}
\label{\detokenize{input_file:impropers}}
\sphinxAtStartPar
Energy parameters used to describe out\sphinxhyphen{}of\sphinxhyphen{}plane rocking are currently read, but unused. The section is often blank. If it becomes necessary, algorithms to calculate the improper energy will need to be added.


\subsection{NONBONDED}
\label{\detokenize{input_file:nonbonded}}
\sphinxAtStartPar
The next section of the CHARMM style parameter file is the NONBONDED. The nonbonded energy in CHARMM is presented as 12\sphinxhyphen{}6 potential
where, \(r_{ij}\), \(\epsilon_{ij}\), \({R_{min}}_{ij}\) are the separation, minimum potential, and minimum potential distance, respectively.
In order to use TraPPE this section of the CHARMM compliant file is critical.
\begin{equation*}
\begin{split}U_{\texttt{LJ}}&=\sum_{\texttt{nonbonded}} \epsilon_{ij}\left[\left(\frac{R_{min_{ij}}}{r_{ij}}\right)^{12}-2\left(\frac{R_{min_{ij}}}{r_{ij}}\right)^6\right] \\\end{split}
\end{equation*}
\sphinxAtStartPar
Here’s an example with our isobutane potential model:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
NONBONDED
!
!V(Lennard\PYGZhy{}Jones) = Eps,i,j[(Rmin,i,j/ri,j)**12 \PYGZhy{} 2(Rmin,i,j/ri,j)**6]
!
!atom ignored epsilon         Rmin/2        ignored   eps,1\PYGZhy{}4     Rmin/2,1\PYGZhy{}4
CH3   0.0     \PYGZhy{}0.194745992  2.10461634058     0.0       0.0       0.0 !  TraPPE 1
CH1   0.0     \PYGZhy{}0.019872040  2.62656119304     0.0       0.0       0.0 !  TraPPE 2
End
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The \(R_{min}\) is the potential well\sphinxhyphen{}depth, where the attraction is maximum. However, \(\sigma\) is the particle diameter, where the interaction energy is zero. To convert \(\sigma\) to \(R_{min}\), simply multiply \(\sigma\) by 0.56123102415.
\end{sphinxadmonition}

\begin{sphinxadmonition}{important}{Important:}
\sphinxAtStartPar
If no parameter was defined for 1\sphinxhyphen{}4 interaction e.g (\(\epsilon_{1-4}, Rmin_{1-4}/2\)), GOMC will use the  \(\epsilon, Rmin/2\) for 1\sphinxhyphen{}4 interaction.
\end{sphinxadmonition}


\subsection{NBFIX}
\label{\detokenize{input_file:nbfix}}
\sphinxAtStartPar
The last section of the CHARMM style parameter file is the NBFIX. In this section, individual pair interaction will be modified. First, pseudo non\sphinxhyphen{}bonded parameters have to be defined in NONBONDED and modified in NBFIX. Here iss an example if it is required to modify interaction between CH3 and CH1 atoms:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
NBFIX
!V(Lennard\PYGZhy{}Jones) = Eps,i,j[(Rmin,i,j/ri,j)**12 \PYGZhy{} 2(Rmin,i,j/ri,j)**6]
!
!atom atom  epsilon         Rmin          eps,1\PYGZhy{}4   Rmin,1\PYGZhy{}4
CH3   CH1   \PYGZhy{}0.294745992    1.10461634058 !
End
\end{sphinxVerbatim}

\begin{sphinxadmonition}{important}{Important:}
\sphinxAtStartPar
If no parameter was defined for 1\sphinxhyphen{}4 interaction e.g (\(\epsilon_{1-4}, Rmin_{1-4}\)), GOMC will use the  \(\epsilon, Rmin\) for 1\sphinxhyphen{}4 interaction.
\end{sphinxadmonition}


\section{Exotic or Mie Parameter File}
\label{\detokenize{input_file:exotic-or-mie-parameter-file}}
\sphinxAtStartPar
The Mie file is intended for use with nonstandard/specialty models of molecular interaction, which are not included in CHARMM standard.


\subsection{Mie Potential}
\label{\detokenize{input_file:mie-potential}}\begin{equation*}
\begin{split}E_{ij} = C_{n_{ij}} \epsilon_{ij} \bigg[\bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg]\end{split}
\end{equation*}
\sphinxAtStartPar
where \(r_{ij}\), \(\epsilon_{ij}\), and \(\sigma_{ij}\) are, respectively, the separation, minimum potential, and collision diameter for the pair of interaction sites \(i\) and \(j\). The constant \(C_n\) is a normalization factor such that the minimum of the potential remains at \(-\epsilon_{ij}\) for all \(n_{ij}\). In the 12\sphinxhyphen{}6 potential, \(C_n\) reduces to the familiar value of 4.
\begin{equation*}
\begin{split}C_{n_{ij}} = \bigg(\frac{n_{ij}}{n_{ij} - 6} \bigg)\bigg(\frac{n_{ij}}{6} \bigg)^{6/(n_{ij} - 6)}\end{split}
\end{equation*}

\subsection{Buckingham Potential (Exp\sphinxhyphen{}6)}
\label{\detokenize{input_file:buckingham-potential-exp-6}}\begin{equation*}
\begin{split}E_{ij} =
\begin{cases}
  \frac{\alpha_{ij}\epsilon_{ij}}{\alpha_{ij}-6} \bigg[\frac{6}{\alpha_{ij}} exp\bigg(\alpha_{ij} \bigg[1-\frac{r_{ij}}{R_{min,ij}} \bigg]\bigg) - {\bigg(\frac{R_{min,ij}}{r_{ij}}\bigg)}^6 \bigg] &  r_{ij} \geq R_{max,ij} \\
  \infty & r_{ij} < R_{max,ij}
\end{cases}\end{split}
\end{equation*}
\sphinxAtStartPar
where \(r_{ij}\), \(\epsilon_{ij}\), and \(R_{min,ij}\) are, respectively, the separation, minimum potential, and minimum potential distance for the pair of interaction sites \(i\) and \(j\).
The constant \(\alpha_{ij}\) is an  exponential\sphinxhyphen{}6 parameter. The cutoff distance \(R_{max,ij}\) is the smallest positive value for which \(\frac{dE_{ij}}{dr_{ij}}=0\).

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In order to use \sphinxcode{\sphinxupquote{Mie}} or \sphinxcode{\sphinxupquote{Exotice}} potential file format for \sphinxcode{\sphinxupquote{Buckingham}} potential, instead of defining \(R_{min}\), we define \(\sigma\) (collision diameter or the distance, where potential is zero)
and GOMC will calculate the \(R_{min}\) and \(R_{max}\) using \sphinxcode{\sphinxupquote{Buckingham}} potential equation.
\end{sphinxadmonition}

\sphinxAtStartPar
Currently, two custom interaction are included:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NONBODED\_MIE}} This section describes n\sphinxhyphen{}6 (Lennard\sphinxhyphen{}Jones) or Exp\sphinxhyphen{}6 (Buckingham) non\sphinxhyphen{}bonded interactions. The Lennard\sphinxhyphen{}Jones potential (12\sphinxhyphen{}6) is a subset of Mie potential.
Non\sphinxhyphen{}bonded parameters are assigned by specifying the following fields in order:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Atom type name

\item {} 
\sphinxAtStartPar
Minimum energy (\(\epsilon\))

\item {} 
\sphinxAtStartPar
Atom diameter (\(\sigma\))

\item {} 
\sphinxAtStartPar
Repulsion exponent (\(n\)) in \sphinxcode{\sphinxupquote{Mie}} potential or \(\alpha\) in \sphinxcode{\sphinxupquote{Buckingham}} potential.

\end{enumerate}

\sphinxAtStartPar
The 1\sphinxhyphen{}4 interaction can be modified by specifying the following fields in order:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{4}
\item {} 
\sphinxAtStartPar
Minimum energy (\(\epsilon_{1-4}\))

\item {} 
\sphinxAtStartPar
Atom diameter (\(\sigma_{1-4}\))

\item {} 
\sphinxAtStartPar
Repulsion exponent (\(n_{1-4}\)) in \sphinxcode{\sphinxupquote{Mie}} potential or \(\alpha_{1-4}\) in \sphinxcode{\sphinxupquote{Buckingham}} potential.

\end{enumerate}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
If no parameter is provided for 1\sphinxhyphen{}4 interaction, same parameters (item 2, 3, 4) would be considered for 1\sphinxhyphen{}4 interaction.
\end{sphinxadmonition}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NBFIX\_MIE}} This section allows n\sphinxhyphen{}6 (Lennard\sphinxhyphen{}Jones) or Exp\sphinxhyphen{}6 (Buckingham) interaction between two pairs of atoms to be modified.
Interaction between two pairs of atoms can be modified by specifying the following fields in order:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Atom type 1 name

\item {} 
\sphinxAtStartPar
Atom type 2 name

\item {} 
\sphinxAtStartPar
Minimum energy (\(\epsilon\))

\item {} 
\sphinxAtStartPar
Atom diameter (\(\sigma\))

\item {} 
\sphinxAtStartPar
Repulsion exponent (\(n\)) in \sphinxcode{\sphinxupquote{Mie}} potential or \(\alpha\) in \sphinxcode{\sphinxupquote{Buckingham}} potential.

\end{enumerate}

\sphinxAtStartPar
The 1\sphinxhyphen{}4 interaction between two pairs of atoms can be modified by specifying the following fields in order:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{5}
\item {} 
\sphinxAtStartPar
Minimum energy (\(\epsilon_{1-4}\))

\item {} 
\sphinxAtStartPar
Atom diameter (\(\sigma_{1-4}\))

\item {} 
\sphinxAtStartPar
Repulsion exponent (\(n_{1-4}\)) in \sphinxcode{\sphinxupquote{Mie}} potential or \(\alpha_{1-4}\) in \sphinxcode{\sphinxupquote{Buckingham}} potential.

\end{enumerate}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
If no parameter is provided for 1\sphinxhyphen{}4 interaction, same parameters (item 3, 4, 5) would be considered for 1\sphinxhyphen{}4 interaction.
\end{sphinxadmonition}

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In \sphinxcode{\sphinxupquote{Mie}} or \sphinxcode{\sphinxupquote{Buckingham}} potential, the definition of atom diameter(\(\sigma\)) is same for both \sphinxcode{\sphinxupquote{NONBONDED\_MIE}} and \sphinxcode{\sphinxupquote{NBFIX\_MIE}}.
\end{sphinxadmonition}

\begin{sphinxadmonition}{important}{Important:}
\sphinxAtStartPar
If no parameter was defined for 1\sphinxhyphen{}4 interaction e.g (\(\epsilon_{1-4}, \sigma_{1-4}, n_{1-4}\)), GOMC will use the  \(\epsilon, \sigma, n\) for 1\sphinxhyphen{}4 interaction.
\end{sphinxadmonition}

\sphinxAtStartPar
Otherwise, the Mie file reuses the same geometry section headings \sphinxhyphen{} BONDS / ANGLES / DIHEDRALS / etc. The only difference in these sections versus in the CHARMM format force field file is that the energies are in Kelvin (‘K’),
the unit most commonly found for parameters in Monte Carlo chemical simulation literature. This precludes the need to convert to kcal/mol, the energy unit used in CHARMM.
The most frequently used section of the Mie files in the Mie potential section is NONBONDED\_MIE.

\sphinxAtStartPar
Here is the example of \sphinxcode{\sphinxupquote{Mie}} or \sphinxcode{\sphinxupquote{Exotic}} parameters file format that are used to simulate alkanes with \sphinxcode{\sphinxupquote{Mie}} potential:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
NONBONDED\PYGZus{}MIE
!
!V(Mie) = const*eps*((sig/r)\PYGZca{}n\PYGZhy{}(sig/r)\PYGZca{}6)
!
!atom eps       sig     n     eps,1\PYGZhy{}4   sig,1\PYGZhy{}4   n,1\PYGZhy{}4
CH4   161.00    3.740   14    0.0       0.0       0.0 ! Potoff, et al. \PYGZsq{}09
CH3   121.25    3.783   16    0.0       0.0       0.0 ! Potoff, et al. \PYGZsq{}09
CH2    61.00    3.990   16    0.0       0.0       0.0 ! Potoff, et al. \PYGZsq{}09

NBFIX\PYGZus{}MIE
!V(Mie) = const*eps*((sig/r)\PYGZca{}n\PYGZhy{}(sig/r)\PYGZca{}6)
!
!atom atom  epsilon  sig     n     eps,1\PYGZhy{}4   sig,1\PYGZhy{}4   n,1\PYGZhy{}4
CH3   CH2   100.00   3.8     16    0.0       0.0       0.0 !
End
\end{sphinxVerbatim}

\sphinxAtStartPar
Here is the example of \sphinxcode{\sphinxupquote{Mie}} or \sphinxcode{\sphinxupquote{Exotic}} parameters file format that are used to simulate water with \sphinxcode{\sphinxupquote{Buckingham}} potential:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
NONBONDED\PYGZus{}MIE
!
!V(exp\PYGZhy{}6) = ((eps\PYGZhy{}ij * alpha)/(alpha \PYGZhy{} 6)) * ((6 / alpha) * exp(alpha * [1 \PYGZhy{} (r / rmin)]) \PYGZhy{} (rmin / r)\PYGZca{}6))
!
!atom eps       sig     alpha     eps,1\PYGZhy{}4   sig,1\PYGZhy{}4   n,1\PYGZhy{}4
OT    159.78    3.195   12        0.0       0.0       0.0 ! Errington, et al. 1998
HT      0.0     0.0      0        0.0       0.0       0.0 ! Errington, et al. 1998

NBFIX\PYGZus{}MIE
!V(exp\PYGZhy{}6) = ((eps\PYGZhy{}ij * alpha)/(alpha \PYGZhy{} 6)) * ((6 / alpha) * exp(alpha * [1 \PYGZhy{} (r / rmin)]) \PYGZhy{} (rmin / r)\PYGZca{}6))
!
!atom atom  epsilon  sig     alpha     eps,1\PYGZhy{}4   sig,1\PYGZhy{}4   n,1\PYGZhy{}4
HT   OT      0.00    0.0     0         0.0       0.0       0.0 !
End
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Although the units (Angstroms) are the same, the Mie file uses \(\sigma\), not the \(R_{min}\) used by CHARMM. The energy in the exotic file are expressed in Kelvin (K), as this is the standard convention in the literature.
\end{sphinxadmonition}


\section{Control File (*.conf)}
\label{\detokenize{input_file:control-file-conf}}
\sphinxAtStartPar
The control file is GOMC’s proprietary input file. It contains key settings. The settings generally fall under three categories:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Input/Simulation Setup

\item {} 
\sphinxAtStartPar
System Settings for During Run

\item {} 
\sphinxAtStartPar
Output Settings

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The control file is designed to recognize logic values, such as “yes/true/on” or “no/false/off”. The keyword in control file is not case sensitive.
\end{sphinxadmonition}


\subsection{Input/Simulation Setup}
\label{\detokenize{input_file:input-simulation-setup}}
\sphinxAtStartPar
In this section, input file names are listed. In addition, if you want to restart your simulation or use integer seed for running your simulation, you need to modify this section according to your purpose.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Restart}}}] \leavevmode
\sphinxAtStartPar
Determines whether to restart the simulation from restart file (\sphinxtitleref{*\_restart.pdb}) or not.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True if restart, false otherwise.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{ExpertMode}}}] \leavevmode
\sphinxAtStartPar
Determines whether to perform error checking of move selection to ensure correct ensemble is sampled.  This allows the user to run a simulation with no volume moves in NPT, NPT\sphinxhyphen{}GEMC; no molecule transfers in GCMC, GEMC.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True if enable expert mode; false otherwise.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{Checkpoint}}}] \leavevmode
\sphinxAtStartPar
Determines whether to restart the simulation from checkpoint file or not. Restarting the simulation with would result in
an identitcal outcome, as if previous simulation was continued.  This is required for hybrid Monte\sphinxhyphen{}Carlo Molecular Dyanamics in open\sphinxhyphen{}ensembles (GCMC/GEMC) to concatenate trajectory files since the molecular transfers rearranges the order of the molecules.  Checkpointing will ensure the molecules are loaded in the same order each cycle.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True if restart with checkpoint file, false otherwise.

\item {} 
\sphinxAtStartPar
Value 2: String \sphinxhyphen{} Sets the name of the checkpoint file.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Checkpoint   true        AR\PYGZus{}KR\PYGZus{}continued.chk
\end{sphinxVerbatim}

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{PRNG}}}] \leavevmode
\sphinxAtStartPar
Dictates how to start the pseudo\sphinxhyphen{}random number generator (PRNG)
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String
\begin{itemize}
\item {} 
\sphinxAtStartPar
RANDOM: Randomizes Mersenne Twister PRNG with random bits based on the system time.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} kind \PYGZob{}RANDOM, INTSEED\PYGZcb{}
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
PRNG   RANDOM
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
INTSEED: This option “seeds” the Mersenne Twister PRNG with a standard integer. When the same integer is used, the generated PRNG stream should be the same every time, which is helpful in tracking down bugs.

\end{itemize}

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{Random\_Seed}}}] \leavevmode
\sphinxAtStartPar
Defines the seed number. If “INTSEED” is chosen, seed number needs to be specified; otherwise, the program will terminate.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: ULONG \sphinxhyphen{} If “INTSEED” option is selected for PRNG (See bellow example)

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} kind \PYGZob{}RANDOM, INTSEED\PYGZcb{}
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
PRNG          INTSEED
Random\PYGZus{}Seed    50
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{ParaTypeCHARMM}}}] \leavevmode
\sphinxAtStartPar
Sets force field type to CHARMM style.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True if it is CHARMM forcefield, false otherwise.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} FORCE FIELD TYPE
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
ParaTypeCHARMM    true
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{ParaTypeEXOTIC}} or \sphinxcode{\sphinxupquote{ParaTypeMie}}}] \leavevmode
\sphinxAtStartPar
Sets force field type to Mie style.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True if it is Mie forcefield, false otherwise.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} FORCE FIELD TYPE
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
ParaTypeEXOTIC    true
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{ParaTypeMARTINI}}}] \leavevmode
\sphinxAtStartPar
Sets force field type to MARTINI style.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True if it is MARTINI forcefield, false otherwise.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} FORCE FIELD TYPE
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
ParaTypeMARTINI     true
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{Parameters}}}] \leavevmode
\sphinxAtStartPar
Provides the name and location of the parameter file to use for the simulation.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Sets the name of the parameter file.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} FORCE FIELD TYPE
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
ParaTypeCHARMM    yes
Parameters        ../../common/Par\PYGZus{}TraPPE\PYGZus{}Alkanes.inp
Parameters        ../../common/Par\PYGZus{}TIP4P.inp
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
More than one parameter file can be provided.
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{Coordinates}}}] \leavevmode
\sphinxAtStartPar
Defines the PDB file names (coordinates) and location for each box in the system.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Sets box number (starts from ‘0’).

\item {} 
\sphinxAtStartPar
Value 2: String \sphinxhyphen{} Sets the name of PDB file.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
NVT and NPT ensembles requires only one PDB file and GEMC/GCMC requires two PDB files. If the number of PDB files is not compatible with the simulation type, the program will terminate.
\end{sphinxadmonition}

\sphinxAtStartPar
Example of NVT or NPT ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES \PYGZhy{} NVT or NPT ensemble
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Coordinates   0   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys.pdb
\end{sphinxVerbatim}

\sphinxAtStartPar
Example of Gibbs or GC ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES \PYGZhy{} Gibbs or GCMC ensemble
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Coordinates   0   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}0.pdb
Coordinates   1   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}1.pdb
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In case of \sphinxcode{\sphinxupquote{Restart}} true, the restart PDB output file from GOMC (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_N\_restart.pdb) can be used for each box.
\end{sphinxadmonition}

\sphinxAtStartPar
Example of Gibbs ensemble when Restart mode is active:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Coordinates   0   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.pdb
Coordinates   1   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}1\PYGZus{}restart.pdb
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{Structures}}}] \leavevmode
\sphinxAtStartPar
Defines the PSF filenames (structures) for each box in the system.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Sets box number (start from ‘0’)

\item {} 
\sphinxAtStartPar
Value 2: String \sphinxhyphen{} Sets the name of PSF file.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
NVT and NPT ensembles requires only one PSF file and GEMC/GCMC requires two PSF files. If the number of PSF files is not compatible with the simulation type, the program will terminate.
\end{sphinxadmonition}

\sphinxAtStartPar
Example of NVT or NPT ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PSF FILES
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Structure   0   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys.psf
\end{sphinxVerbatim}

\sphinxAtStartPar
Example of Gibbs or GC ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PSF FILES
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Structure   0   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}0.psf
Structure   1   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}1.psf
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In case of \sphinxcode{\sphinxupquote{Restart}} true, the PSF output file from GOMC (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_N\_restart.psf) can be used for both boxes.
\end{sphinxadmonition}

\sphinxAtStartPar
Example of Gibbs ensemble when \sphinxcode{\sphinxupquote{Restart}} mode is active:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PSF FILES
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Structure   0   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.psf
Structure   1   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}1\PYGZus{}restart.psf
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{binCoordinates}}}] \leavevmode
\sphinxAtStartPar
Defines the DCD file names (coordinates) and location for each box in the system.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Sets box number (starts from ‘0’).

\item {} 
\sphinxAtStartPar
Value 2: String \sphinxhyphen{} Sets the name of PDB file.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
NVT and NPT ensembles requires only one DCD file and GEMC/GCMC requires only one PDB files, although loading two is supported. This is different from PDB files, for which two are required in GEMC/GCMC.  This allows the user to only load binary coordinates for one box.
\end{sphinxadmonition}

\sphinxAtStartPar
Example of NVT or NPT ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES \PYGZhy{} NVT or NPT ensemble
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
binCoordinates   0   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys.coor
\end{sphinxVerbatim}

\sphinxAtStartPar
Example of Gibbs or GC ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES \PYGZhy{} Gibbs or GCMC ensemble
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
binCoordinates   0   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}0.coor
binCoordinates   1   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}1.coor
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In case of \sphinxcode{\sphinxupquote{Restart}}, the restart DCD output file from GOMC (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_N\_restart.coor) can be used for each box.
\end{sphinxadmonition}

\sphinxAtStartPar
Example of Gibbs ensemble when Restart mode is active:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
binCoordinates   0   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.coor
binCoordinates   1   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}1\PYGZus{}restart.coor
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{binVelocities}}}] \leavevmode
\sphinxAtStartPar
Defines the VEL file names (velocities) and location for each box in the system.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Sets box number (starts from ‘0’).

\item {} 
\sphinxAtStartPar
Value 2: String \sphinxhyphen{} Sets the name of VEL file.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Originate from a Molecular Dynamics softwrae such as NAMD.  GOMC will only output a velocity restart file if it is provided one using this keyword.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In hybrid Monte\sphinxhyphen{}Carlo Molecular Dynamics, the velocities of the atoms should be preserved across cycles to increase accuracy.  These files are not used internally by GOMC, only maintained.  If a molecular transfer occurs, a new velocity is generated by Langevin dynamics.
\end{sphinxadmonition}

\sphinxAtStartPar
Example of NVT or NPT ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES \PYGZhy{} NVT or NPT ensemble
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
binVelocities   0   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys.vel
\end{sphinxVerbatim}

\sphinxAtStartPar
Example of Gibbs or GC ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES \PYGZhy{} Gibbs or GCMC ensemble
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
binVelocities   0   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}0.vel
binVelocities   1   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}1.vel
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In case of \sphinxcode{\sphinxupquote{Restart}}, the restart VEL output file from GOMC (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_N\_restart.vel) can be used for each box.
\end{sphinxadmonition}

\sphinxAtStartPar
Example of Gibbs ensemble when Restart mode is active:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
binVelocities   0   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.vel
binVelocities   1   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}1\PYGZus{}restart.vel
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{extendedSystem}}}] \leavevmode
\sphinxAtStartPar
Defines the XSC file names (box dimensions and origin) and location for each box in the system.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Sets box number (starts from ‘0’).

\item {} 
\sphinxAtStartPar
Value 2: String \sphinxhyphen{} Sets the name of XSC file.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Previously, this information was stored in plain\sphinxhyphen{}text format at the top of restart PDB files.  This will be deprecated in favor of binary XSC.
\end{sphinxadmonition}

\sphinxAtStartPar
Example of NVT or NPT ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES \PYGZhy{} NVT or NPT ensemble
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
extendedSystem   0   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys.xsc
\end{sphinxVerbatim}

\sphinxAtStartPar
Example of Gibbs or GC ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES \PYGZhy{} Gibbs or GCMC ensemble
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
extendedSystem   0   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}0.xsc
extendedSystem   1   STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}1.xsc
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In case of \sphinxcode{\sphinxupquote{Restart}}, the restart XSC output file from GOMC (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_N\_restart.xsc) can be used for each box.
\end{sphinxadmonition}

\sphinxAtStartPar
Example of Gibbs ensemble when Restart mode is active:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} INPUT PDB FILES
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
extendedSystem   0   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.xsc
extendedSystem   1   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}1\PYGZus{}restart.xsc
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{MultiSimFolderName}}}] \leavevmode
\sphinxAtStartPar
The name of the folder to be created which contains output from the multisim.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Name of the folder to contain output

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
MultiSimFolderName  outputFolderName
\end{sphinxVerbatim}

\end{description}


\section{Binary input file types}
\label{\detokenize{input_file:binary-input-file-types}}\begin{description}
\item[{Binary representations of the system.}] \leavevmode\begin{itemize}
\item {} 
\sphinxAtStartPar
XSC

\item {} 
\sphinxAtStartPar
COOR

\item {} 
\sphinxAtStartPar
VEL

\item {} 
\sphinxAtStartPar
CHK

\item {} 
\sphinxAtStartPar
“NAMD uses a trivial double\sphinxhyphen{}precision binary file format for coordinates, velocities, and forces. Due to its high precision this is the default output and restart format. VMD refers to these files as the \textasciigrave{}\textasciigrave{}namdbin\textquotesingle{}\textquotesingle{} format. The file consists of the atom count as a 32\sphinxhyphen{}bit integer followed by all three position or velocity components for each atom as 64\sphinxhyphen{}bit double\sphinxhyphen{}precision floating point, i.e., NXYZXYZXYZXYZ... where N is a 4\sphinxhyphen{}byte int and X, Y, and Z are 8\sphinxhyphen{}byte doubles. If the number of atoms the file contains is known then the atom count can be used to determine endianness. The file readers in NAMD and VMD can detect and adapt to the endianness of the machine on which the binary file was written, and the utility program flipbinpdb is also provided to reformat these files if needed. Positions in NAMD binary files are stored in Å. Velocities in NAMD binary files are stored in NAMD internal units and must be multiplied by PDBVELFACTOR=20.45482706 to convert to Å/ps. Forces in NAMD binary files are stored in kcal/mol/Å.”

\end{itemize}
\begin{itemize}
\item {} 
\sphinxAtStartPar
source : \sphinxurl{https://www.ks.uiuc.edu/Research/namd/2.9/ug/node11.html}

\end{itemize}

\end{description}


\section{XSC (eXtended System Configuration file) File}
\label{\detokenize{input_file:xsc-extended-system-configuration-file-file}}
\sphinxAtStartPar
GOMC allows the box dimensions to be defined in one of three ways:
\begin{itemize}
\item {} 
\sphinxAtStartPar
In the control file

\item {} 
\sphinxAtStartPar
In the header of restart PDB file

\item {} 
\sphinxAtStartPar
In a binary XSC file

\end{itemize}

\sphinxAtStartPar
The XSC file contains the first step of the simulation, cell vectors, and cell origin.  Currently, GOMC only uses the cell vectors.


\section{COOR (binary coordinates) File}
\label{\detokenize{input_file:coor-binary-coordinates-file}}
\sphinxAtStartPar
GOMC allows the box coordinates to be overwritten by a binary coordinates file.  The COOR file should have the same number of atoms in it as the PDB file which it is overwriting.  The actual coordinates can vary dramatically, which allows the user to sample the coordinates with other engines (MCMD), or transform it however one sees fit.


\section{VEL (binary velocity) File}
\label{\detokenize{input_file:vel-binary-velocity-file}}
\sphinxAtStartPar
GOMC allows the velocities associated with each atom to be maintained and output for continuing MD simulations.  In the event a molecule transfer occurs, all the atoms of the transferred molecule are given new velocities by Langevin dynamics.  These VEL files must originate from NAMD, as GOMC will not produce them without first being provided them.


\section{CHK (checkpoint) File}
\label{\detokenize{input_file:chk-checkpoint-file}}
\sphinxAtStartPar
GOMC contains several variables which, if not accounted for, will produce different outputs even if the initial conditions are exactly the same.  These variables are contained in the checkpoint file, and allow the user to pick up a GOMC simulation where it left off without altering the course of the simulation.  Also, the checkpoint file is essential for MCMD as molecules are treated as distinguishable in molecular dynamics due to the fact that MD is a continuous trajectory through time.  The checkpoint file contains the original atom order of the molecules, and coordinates and velocities are loaded into this order to ensure the trajectories are consistently arranged.

\sphinxAtStartPar
Checkpoint file contents:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Last simulation step that saved into checkpoint file (Start step can be overriden).

\item {} 
\sphinxAtStartPar
True number of simulation steps that have been run.

\item {} 
\sphinxAtStartPar
Maximum amount of displacement (Å), rotation (\(\delta\)), and volume (\(\AA^3\)) that used in Displacement, Rotation, MultiParticle, and Volume move.

\item {} 
\sphinxAtStartPar
Number of Monte Carlo move trial and acceptance.

\item {} 
\sphinxAtStartPar
Random number sequence.

\item {} 
\sphinxAtStartPar
Molecule lookup object.

\item {} 
\sphinxAtStartPar
Original pdb atoms object to reload new positions into.

\item {} 
\sphinxAtStartPar
Original molecule setup object generated from parsing first PSF files.

\item {} 
\sphinxAtStartPar
Accessory data for coordinating loading the restart coordinates into the original ordering.

\item {} 
\sphinxAtStartPar
If built with MPI and parallel tempering was enabled: Random number sequence for parallel tempering.

\end{itemize}


\subsection{System Settings for During Run Setup}
\label{\detokenize{input_file:system-settings-for-during-run-setup}}
\sphinxAtStartPar
This section contains all the variables not involved in the output of data during the simulation, or in the reading of input files at the start of the simulation. In other words, it contains settings related to the moves, the thermodynamic constants (based on choice of ensemble), and the length of the simulation.
Note that some tags, or entries for tags, are only used in certain ensembles (e.g. Gibbs ensemble). These cases are denoted with colored text.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{GEMC}}}] \leavevmode
\sphinxAtStartPar
\sphinxstyleemphasis{(For Gibbs Ensemble runs only)} Defines the type of Gibbs Ensemble simulation you want to run. If neglected in Gibbs Ensemble, it simply defaults to const volume (NVT) Gibbs Ensemble.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Allows you to pick between isovolumetric (“NVT”) and isobaric (“NPT”) Gibbs ensemble simulations.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{GEMC}} is NVT.
\end{sphinxadmonition}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} GEMC TYPE (DEFAULT IS NVT GEMC)
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
GEMC    NVT
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{Pressure}}}] \leavevmode
\sphinxAtStartPar
For \sphinxcode{\sphinxupquote{NPT}} or \sphinxcode{\sphinxupquote{NPT\sphinxhyphen{}GEMC}} simulation, imposed pressure (in bar) needs to be specified; otherwise, the program will terminate.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} Constant pressure in bar.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} GEMC TYPE (DEFAULT IS NVT GEMC)
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
GEMC        NPT
Pressure    5.76
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{Temperature}}}] \leavevmode
\sphinxAtStartPar
Sets the temperature at which the system will run.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} Constant temperature of simulation in degrees Kelvin.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} SIMULATION CONDITION
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Temperature   270.00
\end{sphinxVerbatim}

\sphinxAtStartPar
(MPI\sphinxhyphen{}GOMC Only)
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: List of Doubles \sphinxhyphen{} A list of constant temperatures for simulations in degrees Kelvin.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} SIMULATION CONDITION
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Temperature   270.00    280.00    290.00    300.00
\end{sphinxVerbatim}

\end{description}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
To use more than one temperature, GOMC must be compiled in MPI mode.  Also, if GOMC is compiled in MPI mode, more than one temperature is required.  To use only one temperature, use standard GOMC.
\end{sphinxadmonition}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Rcut}}}] \leavevmode
\sphinxAtStartPar
Sets a specific radius that non\sphinxhyphen{}bonded interaction energy and force will be considered and calculated using defined potential function.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} The distance to truncate the Lennard\sphinxhyphen{}Jones potential at.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{RcutLow}}}] \leavevmode
\sphinxAtStartPar
Sets a specific minimum possible in angstrom that reject any move that places any atom closer than specified distance.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} The minimum possible distance between any atoms.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{RcutCoulomb}}}] \leavevmode
\sphinxAtStartPar
Sets a specific radius for each box in the system that short range electrostatic energy will be calculated.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Sets box number (start from ‘0’)

\item {} 
\sphinxAtStartPar
Value 2: Double \sphinxhyphen{} The distance to truncate the short rage electrostatic energy at.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{RcutCoulomb}} is the value of \sphinxcode{\sphinxupquote{Rcut}}
\end{sphinxadmonition}

\begin{sphinxadmonition}{important}{Important:}\begin{itemize}
\item {} 
\sphinxAtStartPar
In Ewald Summation method, at constant \sphinxcode{\sphinxupquote{Tolerance}} and box volume, increasing \sphinxcode{\sphinxupquote{RcutCoulomb}} would result is decreasing reciprocal vector {[}\sphinxhref{https://www.tandfonline.com/doi/abs/10.1080/08927029408022180}{Fincham 1993}{]}.
Decreasing the reciprocal vector decreases the computation time in long range electrostatic calculation.

\item {} 
\sphinxAtStartPar
Increasing the \sphinxcode{\sphinxupquote{RcutCoulomb}} results in increasing the computation time in short range electrostatic calculation.

\item {} 
\sphinxAtStartPar
Parallelization of Ewald summation method is done on reciprocal vector loop, rather than molecule loop.
So, in case of running on multiple CPU threads or GPU, it is better to use the lower value for \sphinxcode{\sphinxupquote{RcutCoulomb}}, to maximize the parallelization efficiency.

\item {} 
\sphinxAtStartPar
There is an optimum value for \sphinxcode{\sphinxupquote{RcutCoulomb}}, where result in maximum effeciency of the method. We encourage to run a short simulation with various \sphinxcode{\sphinxupquote{RcutCoulomb}} to find the optimum value.

\end{itemize}
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{LRC}}}] \leavevmode
\sphinxAtStartPar
Defines whether or not long range corrections are used.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True to consider long range correction.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In case of using \sphinxcode{\sphinxupquote{SHIFT}} or \sphinxcode{\sphinxupquote{SWITCH}} potential functions, \sphinxcode{\sphinxupquote{LRC}} will be ignored.
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{Exclude}}}] \leavevmode
\sphinxAtStartPar
Defines which pairs of bonded atoms should be excluded from non\sphinxhyphen{}bonded interactions.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Allows you to choose between “1\sphinxhyphen{}2”, “1\sphinxhyphen{}3”, and “1\sphinxhyphen{}4”.
\begin{itemize}
\item {} 
\sphinxAtStartPar
1\sphinxhyphen{}2: All interactions pairs of bonded atoms, except the ones that separated with one bond, will be considered and modified using 1\sphinxhyphen{}4 parameters defined in parameter file.

\item {} 
\sphinxAtStartPar
1\sphinxhyphen{}3: All interaction pairs of bonded atoms, except the ones that separated with one or two bonds, will be considered and modified using 1\sphinxhyphen{}4 parameters defined in parameter file.

\item {} 
\sphinxAtStartPar
1\sphinxhyphen{}4: All interaction pairs of bonded atoms, except the ones that separated with one, two or three bonds, will be considered using non\sphinxhyphen{}bonded parameters defined in parameter file.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{Exclude}} is “1\sphinxhyphen{}4”.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In CHARMM force field, the 1\sphinxhyphen{}4 interaction needs to be considered. Choosing “\sphinxcode{\sphinxupquote{Exclude}} 1\sphinxhyphen{}3” will modify 1\sphinxhyphen{}4 interaction based on 1\sphinxhyphen{}4 parameters in parameter file. If a kind of force field is used, where 1\sphinxhyphen{}4 interaction needs to be ignored, such as TraPPE, either “\sphinxcode{\sphinxupquote{Exclude}} 1\sphinxhyphen{}4” needs to be chosen or 1\sphinxhyphen{}4 parameter needs to be assigned to zero in the parameter file.
\end{sphinxadmonition}

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{Potential}}}] \leavevmode
\sphinxAtStartPar
Defines the potential function type to calculate non\sphinxhyphen{}bonded interaction energy and force between atoms.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Allows you to pick between “VDW”, “EXP6”, “SHIFT” and “SWITCH”.
\begin{itemize}
\item {} 
\sphinxAtStartPar
VDW: Nonbonded interaction energy and force calculated based on n\sphinxhyphen{}6 (Lennard\sphinxhyphen{}Johns) equation. This function will be discussed further in the Intermolecular energy and Virial calculation section.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} SIMULATION CONDITION
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Temperature   270.00
Potential     VDW
LRC           true
Rcut          10
Exclude       1\PYGZhy{}4
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
EXP6: Nonbonded interaction energy and force calculated based on exp\sphinxhyphen{}6 (Buckingham potential) equation. This function will be discussed further in the Intermolecular energy and Virial calculation section.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} SIMULATION CONDITION
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Temperature   270.00
Potential     EXP6
LRC           true
Rcut          10
Exclude       1\PYGZhy{}4
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
SHIFT: This option forces the potential energy to be zero at \sphinxcode{\sphinxupquote{Rcut}} distance. This function will be discussed further in the Intermolecular energy and Virial calculation section.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} SIMULATION CONDITION
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Temperature     270.00
Potential       SHIFT
LRC             false
Rcut            10
Exclude         1\PYGZhy{}4
RcutCoulomb  0  12.0
RcutCoulomb  1  20.0
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
SWITCH: This option smoothly forces the potential energy to be zero at \sphinxcode{\sphinxupquote{Rcut}} distance and starts modifying the potential at \sphinxcode{\sphinxupquote{Rswitch}} distance. Depending on force field type, specific potential function will be applied. These functions will be discussed further in the Intermolecular energy and Virial calculation section.

\end{itemize}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Rswitch}}}] \leavevmode
\sphinxAtStartPar
In the case of choosing “SWITCH” as potential function, a distance is set in which non\sphinxhyphen{}bonded interaction energy is truncated smoothly at \sphinxcode{\sphinxupquote{Rcut}} distance.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} Define switch distance in angstrom. If the “SWITCH” function is chosen, \sphinxcode{\sphinxupquote{Rswitch}} needs to be defined; otherwise, the program will be terminated.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} SIMULATION CONDITION
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Temperature   270.00
Potential     SWITCH
LRC           false
Rcut          12
Rswitch       9
Exclude       1\PYGZhy{}4
\end{sphinxVerbatim}

\end{description}

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{VDWGeometricSigma}}}] \leavevmode
\sphinxAtStartPar
Use geometric mean, as required by OPLS force field, to combining Lennard\sphinxhyphen{}Jones sigma parameters for different atom types.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True, uses geometric mean to combine L\sphinxhyphen{}J sigmas

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The default setting of \sphinxcode{\sphinxupquote{VDWGeometricSigma}} is false to use arithmetic mean when combining Lennard\sphinxhyphen{}Jones sigma parameters for different atom types.
\end{sphinxadmonition}

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{ElectroStatic}}}] \leavevmode
\sphinxAtStartPar
Considers coulomb interaction or not. This function will be discussed further in the Inter\sphinxhyphen{} molecular energy and Virial calculation section.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True if coulomb interaction needs to be considered and false if not.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
To simulate the polar molecule in MARTINI force field, \sphinxcode{\sphinxupquote{ElectroStatic}} needs to be turn on. MARTINI force field uses short range coulomb interaction with constant \sphinxcode{\sphinxupquote{Dielectric}} 15.0.
\end{sphinxadmonition}

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{Ewald}}}] \leavevmode
\sphinxAtStartPar
Considers standard Ewald summation method for electrostatic calculation. This function will be discussed further in the Intermolecular energy and Virial calculation section.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} True if Ewald summation calculation needs to be considered and false if not.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
By default, \sphinxcode{\sphinxupquote{ElectroStatic}} will be set to true if Ewald summation method was used to calculate coulomb interaction.
\end{sphinxadmonition}

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{CachedFourier}}}] \leavevmode
\sphinxAtStartPar
Considers storing the reciprocal terms for Ewald summation calculation in order to improve the code performance. This option would increase the code performance with the cost of memory usage.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True to store reciprocal terms of Ewald summation calculation and false if not.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
By default, \sphinxcode{\sphinxupquote{CachedFourier}} will be set to true if not value was set.
\end{sphinxadmonition}

\begin{sphinxadmonition}{warning}{Warning:}
\sphinxAtStartPar
Monte Carlo moves, such as \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}1}}, \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}2}}, \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}3}}, \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}1}}, \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}2}}, \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}3}} does not support \sphinxcode{\sphinxupquote{CachedFourier}}.
\end{sphinxadmonition}

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{Tolerance}}}] \leavevmode
\sphinxAtStartPar
Specifies the accuracy of the Ewald summation calculation. Ewald separation parameter and number of reciprocal vectors for the Ewald summation are determined based on the accuracy parameter.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} Sets the accuracy in Ewald summation calculation.

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
A reasonable value for te accuracy is 0.00001.

\item {} 
\sphinxAtStartPar
If “Ewald” was chosen and no value was set for Tolerance, the program will be terminated.

\end{itemize}
\end{sphinxadmonition}

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{Dielectric}}}] \leavevmode
\sphinxAtStartPar
Defines dielectric constant for coulomb interaction in MARTINI force field.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} Sets dielectric value used in coulomb interaction.

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
In MARTINI force field, \sphinxcode{\sphinxupquote{Dielectric}} needs to be set to 15.0.

\item {} 
\sphinxAtStartPar
If MARTINI force field was chosen and \sphinxcode{\sphinxupquote{Dielectric}} was not specified, a default value of 15.0 will be assigned.

\end{itemize}
\end{sphinxadmonition}

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{PressureCalc}}}] \leavevmode
\sphinxAtStartPar
Considers to calculate the pressure or not. If it is set to true, the frequency of pressure calculation need to be set.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True enabling pressure calculation during the simulation, false disabling pressure calculation.

\item {} 
\sphinxAtStartPar
Value 2: Ulong \sphinxhyphen{} The frequency of calculating the pressure.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{1\sphinxhyphen{}4scaling}}}] \leavevmode
\sphinxAtStartPar
Defines constant factor to modify intra\sphinxhyphen{}molecule coulomb interaction.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} A fraction number between 0.0 and 1.0.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
CHARMM force field uses a value between 0.0 and 1.0. In MARTINI force field, it needs to be set to 1.0 because 1\sphinxhyphen{}4 interaction will not be modified in this force field.
\end{sphinxadmonition}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} SIMULATION CONDITION
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
ElectroStatic   true
Ewald           true
Tolerance       0.00001
CachedFourier   false
1\PYGZhy{}4scaling      0.0
\end{sphinxVerbatim}

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{RunSteps}}}] \leavevmode
\sphinxAtStartPar
Sets the total number of steps to run (one move is performed for each step) (cycles = this value / number of molecules in the system)
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Ulong \sphinxhyphen{} Total run steps

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
RunSteps is a delta.
\end{sphinxadmonition}

\begin{sphinxadmonition}{important}{Important:}
\sphinxAtStartPar
Seting the \sphinxcode{\sphinxupquote{RunSteps}} to zero, and activating \sphinxcode{\sphinxupquote{Restart}} simulation, will recalculate the energy of stored simulation’s snapshots.
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{EqSteps}}}] \leavevmode
\sphinxAtStartPar
Sets the number of steps necessary to equilibrate the system; averaging will begin at this step.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Ulong \sphinxhyphen{} Equilibration steps

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
EqSteps is not a delta.  If restarting a simulation with a start step greater than EqSteps, no equilibration is performed.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In GCMC simulation, the \sphinxcode{\sphinxupquote{Histogram}} files will be outputed at \sphinxcode{\sphinxupquote{EqSteps}}.
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{AdjSteps}}}] \leavevmode
\sphinxAtStartPar
Sets the number of steps per adjustment of the parameter associated with each move (e.g. maximum translate distance, maximum rotation, maximum volume exchange, etc.)
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Ulong \sphinxhyphen{} Number of steps per move adjustment

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{InitStep}}}] \leavevmode
\sphinxAtStartPar
Sets the first step of the simulation.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Ulong \sphinxhyphen{} Number of first step of simulation.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Hybrid Monte\sphinxhyphen{}Carlo Molecular Dynamics (py\sphinxhyphen{}MCMD) requires resetting start step to 0 for combination of NAMD and GOMC data.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} STEPS
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
RunSteps    25000000
EqSteps     5000000
AdjSteps    1000
InitStep    0
\end{sphinxVerbatim}
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{ChemPot}}}] \leavevmode
\sphinxAtStartPar
For Grand Canonical (GC) ensemble runs only: Chemical potential at which simulation is run.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} The residue name to apply this chemical potential.

\item {} 
\sphinxAtStartPar
Value 2: Double \sphinxhyphen{} The chemical potential value in degrees Kelvin (should be negative).

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
For binary systems, include multiple copies of the tag (one per residue kind).

\item {} 
\sphinxAtStartPar
If there is a molecule kind that cannot be transfer between boxes (in PDB file the beta value is set to 1.00 or 2.00), an arbitrary value (e.g. 0.00) can be assigned to the residue name.

\end{itemize}
\end{sphinxadmonition}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} Mol.  Name Chem.  Pot.  (K)
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
ChemPot   ISB     \PYGZhy{}968
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{Fugacity}}}] \leavevmode
\sphinxAtStartPar
For Grand Canonical (GC) ensemble runs only: Fugacity at which simulation is run.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} The residue to apply this fugacity.

\item {} 
\sphinxAtStartPar
Value 2: Double \sphinxhyphen{} The fugacity value in bar.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
For binary systems, include multiple copies of the tag (one per residue kind).

\item {} 
\sphinxAtStartPar
If there is a molecule kind that cannot be transfer between boxes (in PDB file the beta value is set to 1.00 or 2.00) an arbitrary value e.g. 0.00 can be assigned to the residue name.

\end{itemize}
\end{sphinxadmonition}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} Mol.  Name Fugacity (bar)
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Fugacity  ISB   10.0
Fugacity  Si     0.0
Fugacity  O      0.0
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{DisFreq}}}] \leavevmode
\sphinxAtStartPar
Fractional percentage at which displacement move will occur.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Displacement

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{RotFreq}}}] \leavevmode
\sphinxAtStartPar
Fractional percentage at which rigid rotation move will occur.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Rotatation

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{IntraSwapFreq}}}] \leavevmode
\sphinxAtStartPar
Fractional percentage at which molecule will be removed from a box and inserted into the same box using coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias algorithm.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Intra molecule swap

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{IntraSwapFreq}} is 0.000
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{IntraTargetedSwapFreq}}}] \leavevmode
\sphinxAtStartPar
Fractional percentage at which molecule will be removed from the box and inserted into a subvolume within the same box, or deleted from the subvolume and inserted into the same box using coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias algorithm.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Intra molecule swap

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{IntraTargetedSwapFreq}} is 0.000
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{RegrowthFreq}}}] \leavevmode
\sphinxAtStartPar
Fractional percentage at which part of the molecule will be deleted and then regrown using coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias algorithm.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Molecular growth

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{RegrowthFreq}} is 0.000
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{CrankShaftFreq}}}] \leavevmode
\sphinxAtStartPar
Fractional percentage at which crankshaft move will occur. In this move, two atoms that are forming angle or dihedral are selected randomely and form a shaft.
Then any atoms or group that are within these two selected atoms, will rotate around the shaft to sample intramolecular degree of freedom.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Crankshaft

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{CrankShaftFreq}} is 0.000
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{MultiParticleFreq}}}] \leavevmode
\sphinxAtStartPar
Fractional percentage at which multi\sphinxhyphen{}particle move will occur. In this move, all molecules in the selected simulation box will be rigidly rotated or displaced
simultaneously, along the calculated torque or force, respectively.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Multiparticle

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{MultiParticleFreq}} is 0.000
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{MultiParticleBrownianFreq}}}] \leavevmode
\sphinxAtStartPar
Fractional percentage at which multi\sphinxhyphen{}particle brownian move will occur. In this move, all molecules in the selected simulation box will be rigidly rotated or displaced
simultaneously, along the calculated torque or force, respectively.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Multiparticle

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{MultiParticleBrownianFreq}} is 0.000
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}1Freq}}}] \leavevmode
\sphinxAtStartPar
Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub\sphinxhyphen{}volume within same simulation box.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Molecular exchange

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}1Freq}} is 0.000

\item {} 
\sphinxAtStartPar
This move need additional information such as \sphinxcode{\sphinxupquote{ExchangeVolumeDim}}, \sphinxcode{\sphinxupquote{ExchangeRatio}}, \sphinxcode{\sphinxupquote{ExchangeSmallKind}}, and \sphinxcode{\sphinxupquote{ExchangeLargeKind}}, which will be explained later.

\item {} 
\sphinxAtStartPar
For more information about this move, please refere to \sphinxhref{https://aip.scitation.org/doi/abs/10.1063/1.5025184}{MEMC\sphinxhyphen{}GCMC} and \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0378381218305351}{MEMC\sphinxhyphen{}GEMC} papers.

\end{itemize}
\end{sphinxadmonition}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}2Freq}}}] \leavevmode
\sphinxAtStartPar
Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub\sphinxhyphen{}volume within same simulation box. Backbone of small and large molecule kind will be used to insert the large molecule more efficiently.

\end{description}
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Molecular exchange

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}2Freq}} is 0.000

\item {} 
\sphinxAtStartPar
This move need additional information such as \sphinxcode{\sphinxupquote{ExchangeVolumeDim}}, \sphinxcode{\sphinxupquote{ExchangeRatio}}, \sphinxcode{\sphinxupquote{ExchangeSmallKind}}, \sphinxcode{\sphinxupquote{ExchangeLargeKind}}, \sphinxcode{\sphinxupquote{SmallKindBackBone}}, and \sphinxcode{\sphinxupquote{LargeKindBackBone}}, which will be explained later.

\item {} 
\sphinxAtStartPar
For more information about this move, please refere to \sphinxhref{https://aip.scitation.org/doi/abs/10.1063/1.5025184}{MEMC\sphinxhyphen{}GCMC} and \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0378381218305351}{MEMC\sphinxhyphen{}GEMC} papers.

\end{itemize}
\end{sphinxadmonition}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}3Freq}}}] \leavevmode
\sphinxAtStartPar
Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub\sphinxhyphen{}volume within same simulation box. Specified atom of the large molecule kind will be used to insert the large molecule using coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias.

\end{description}
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Molecular exchange

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}3Freq}} is 0.000

\item {} 
\sphinxAtStartPar
This move need additional information such as \sphinxcode{\sphinxupquote{ExchangeVolumeDim}}, \sphinxcode{\sphinxupquote{ExchangeRatio}}, \sphinxcode{\sphinxupquote{ExchangeSmallKind}}, \sphinxcode{\sphinxupquote{ExchangeLargeKind}}, and \sphinxcode{\sphinxupquote{LargeKindBackBone}}, which will be explained later.

\item {} 
\sphinxAtStartPar
For more information about this move, please refere to \sphinxhref{https://aip.scitation.org/doi/abs/10.1063/1.5025184}{MEMC\sphinxhyphen{}GCMC} and \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0378381218305351}{MEMC\sphinxhyphen{}GEMC} papers.

\end{itemize}
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}1Freq}}}] \leavevmode
\sphinxAtStartPar
For Gibbs and Grand Canonical (GC) ensemble runs only: Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub\sphinxhyphen{}volume in dense simulation box.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Molecular exchange

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}1Freq}} is 0.000

\item {} 
\sphinxAtStartPar
This move need additional information such as \sphinxcode{\sphinxupquote{ExchangeVolumeDim}}, \sphinxcode{\sphinxupquote{ExchangeRatio}}, \sphinxcode{\sphinxupquote{ExchangeSmallKind}}, and \sphinxcode{\sphinxupquote{ExchangeLargeKind}}, which will be explained later.

\item {} 
\sphinxAtStartPar
For more information about this move, please refere to \sphinxhref{https://aip.scitation.org/doi/abs/10.1063/1.5025184}{MEMC\sphinxhyphen{}GCMC} and \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0378381218305351}{MEMC\sphinxhyphen{}GEMC} papers.

\end{itemize}
\end{sphinxadmonition}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}2Freq}}}] \leavevmode
\sphinxAtStartPar
For Gibbs and Grand Canonical (GC) ensemble runs only: Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub\sphinxhyphen{}volume in dense simulation box. Backbone of small and large molecule kind will be used to insert the large molecule more efficiently.

\end{description}
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Molecular exchange

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}2Freq}} is 0.000

\item {} 
\sphinxAtStartPar
This move need additional information such as \sphinxcode{\sphinxupquote{ExchangeVolumeDim}}, \sphinxcode{\sphinxupquote{ExchangeRatio}}, \sphinxcode{\sphinxupquote{ExchangeSmallKind}}, \sphinxcode{\sphinxupquote{ExchangeLargeKind}}, \sphinxcode{\sphinxupquote{SmallKindBackBone}}, and \sphinxcode{\sphinxupquote{LargeKindBackBone}}, which will be explained later.

\item {} 
\sphinxAtStartPar
For more information about this move, please refere to \sphinxhref{https://aip.scitation.org/doi/abs/10.1063/1.5025184}{MEMC\sphinxhyphen{}GCMC} and \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0378381218305351}{MEMC\sphinxhyphen{}GEMC} papers.

\end{itemize}
\end{sphinxadmonition}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}3Freq}}}] \leavevmode
\sphinxAtStartPar
For Gibbs and Grand Canonical (GC) ensemble runs only: Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub\sphinxhyphen{}volume in dense simulation box. Specified atom of the large molecule kind will be used to insert the large molecule using coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias.

\end{description}
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Molecular exchange

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
The default value for \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}3Freq}} is 0.000

\item {} 
\sphinxAtStartPar
This move need additional information such as \sphinxcode{\sphinxupquote{ExchangeVolumeDim}}, \sphinxcode{\sphinxupquote{ExchangeRatio}}, \sphinxcode{\sphinxupquote{ExchangeSmallKind}}, \sphinxcode{\sphinxupquote{ExchangeLargeKind}}, and \sphinxcode{\sphinxupquote{LargeKindBackBone}}, which will be explained later.

\item {} 
\sphinxAtStartPar
For more information about this move, please refere to \sphinxhref{https://aip.scitation.org/doi/abs/10.1063/1.5025184}{MEMC\sphinxhyphen{}GCMC} and \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0378381218305351}{MEMC\sphinxhyphen{}GEMC} papers.

\end{itemize}
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{SwapFreq}}}] \leavevmode
\sphinxAtStartPar
For Gibbs and Grand Canonical (GC) ensemble runs only: Fractional percentage at which molecule swap move will occur using coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Molecule swaps

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{TargetedSwapFreq}}}] \leavevmode
\sphinxAtStartPar
For Gibbs and Grand Canonical (GC) ensemble runs only: Fractional percentage at which targeted molecule swap move will occur using coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias in the sub\sphinxhyphen{}volumes specified.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Molecule targeted swaps

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{VolFreq}}}] \leavevmode
\sphinxAtStartPar
For isobaric\sphinxhyphen{}isothermal ensemble and Gibbs ensemble runs only: Fractional percentage at which molecule will be removed from one box and inserted into the other box using configurational bias algorithm.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \% Volume swaps

\end{itemize}

\end{description}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} MOVE FREQEUNCY
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
DisFreq         0.39
RotFreq         0.10
IntraSwapFreq   0.10
RegrowthFreq    0.10
CrankShaftFreq  0.10
SwapFreq        0.20
VolFreq         0.01
\end{sphinxVerbatim}

\begin{sphinxadmonition}{warning}{Warning:}
\sphinxAtStartPar
All move percentages should add up to 1.0; otherwise, the program will terminate.
\end{sphinxadmonition}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{ExchangeVolumeDim}}}] \leavevmode
\sphinxAtStartPar
To use all variation of \sphinxcode{\sphinxupquote{MEMC}} and \sphinxcode{\sphinxupquote{IntraMEMC}} Monte Carlo moves, the exchange sub\sphinxhyphen{}volume must be defined. The exchange sub\sphinxhyphen{}volume is defined as an orthogonal box with x\sphinxhyphen{}, y\sphinxhyphen{}, and z\sphinxhyphen{}dimensions, where small molecule/molecules kind will be selected from to be exchanged with a large molecule kind.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} X dimension in \(Å\)

\item {} 
\sphinxAtStartPar
Value 2: Double \sphinxhyphen{} Y dimension in \(Å\)

\item {} 
\sphinxAtStartPar
Value 3: Double \sphinxhyphen{} Z dimension in \(Å\)

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
Currently, the X and Y dimension cannot be set independently (X = Y = max(X, Y))

\item {} 
\sphinxAtStartPar
A heuristic for setting good values of the x\sphinxhyphen{}, y\sphinxhyphen{}, and z\sphinxhyphen{}dimensions is to use the geometric size of the large molecule plus 1\sphinxhyphen{}2 Å in each dimension.

\item {} 
\sphinxAtStartPar
In case of exchanging 1 small molecule kind with 1 large molecule kind in \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}2}}, \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}3}}, \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}2}}, \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}3}} Monte Carlo moves, the sub\sphinxhyphen{}volume dimension has no effect on acceptance rate.

\end{itemize}
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{ExchangeSmallKind}}}] \leavevmode
\sphinxAtStartPar
To use all variation of \sphinxcode{\sphinxupquote{MEMC}} and \sphinxcode{\sphinxupquote{IntraMEMC}} Monte Carlo moves, the small molecule kind to be exchanged with a large molecule kind must be defined. Multiple small molecule kind can be specified.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Small molecule kind (resname) to be exchanged.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{ExchangeLargeKind}}}] \leavevmode
\sphinxAtStartPar
To use all variation of \sphinxcode{\sphinxupquote{MEMC}} and \sphinxcode{\sphinxupquote{IntraMEMC}} Monte Carlo moves, the large molecule kind to be exchanged with small molecule kind must be defined. Multiple large molecule kind can be specified.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Large molecule kind (resname) to be exchanged.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{ExchangeRatio}}}] \leavevmode
\sphinxAtStartPar
To use all variation of \sphinxcode{\sphinxupquote{MEMC}} and \sphinxcode{\sphinxupquote{IntraMEMC}} Monte Carlo moves, the exchange ratio must be defined. The exchange ratio defines how many small molecule will be exchanged with 1 large molecule. For each large\sphinxhyphen{}small molecule pairs, one exchange ratio must be defined.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Ratio of exchanging small molecule/molecules with 1 large molecule.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{LargeKindBackBone}}}] \leavevmode
\sphinxAtStartPar
To use \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}2}}, \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}3}}, \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}2}}, and \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}3}} Monte Carlo moves, the large molecule backbone must be defined. The backbone of the molecule is defined as a vector that connects two atoms belong to the large molecule.
The large molecule backbone will be used to align the sub\sphinxhyphen{}volume in \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}2}} and \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}2}} moves, while in \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}3}} and \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}3}} moves, it uses the atom name to start growing the large molecule using coupled\sphinxhyphen{}decoupled configurational\sphinxhyphen{}bias.
For each large\sphinxhyphen{}small molecule pairs, two atom names must be defined.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Atom name 1 belong to the large molecule’s backbone

\item {} 
\sphinxAtStartPar
Value 2: String \sphinxhyphen{} Atom name 2 belong to the large molecule’s backbone

\end{itemize}

\begin{sphinxadmonition}{important}{Important:}\begin{itemize}
\item {} 
\sphinxAtStartPar
In \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}3}} and \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}3}} Monte Carlo moves, both atom names must be same, otherwise program will be terminated.

\item {} 
\sphinxAtStartPar
If large molecule has only one atom (mono atomic molecules), same atom name must be used for \sphinxcode{\sphinxupquote{Value 1}} and \sphinxcode{\sphinxupquote{Value 2}} of the \sphinxcode{\sphinxupquote{LargeKindBackBone}}.

\end{itemize}
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{SmallKindBackBone}}}] \leavevmode
\sphinxAtStartPar
To use \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}2}}, and \sphinxcode{\sphinxupquote{IntraMEMC\sphinxhyphen{}2}} Monte Carlo moves, the small molecule backbone must be defined. The backbone of the molecule is defined as a vector that connects two atoms belong to the small molecule and will be used to align the sub\sphinxhyphen{}volume.
For each large\sphinxhyphen{}small molecule pairs, two atom names must be defined.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Atom name 1 belong to the small molecule’s backbone

\item {} 
\sphinxAtStartPar
Value 2: String \sphinxhyphen{} Atom name 2 belong to the small molecule’s backbone

\end{itemize}

\begin{sphinxadmonition}{important}{Important:}\begin{itemize}
\item {} 
\sphinxAtStartPar
If small molecule has only one atom (mono atomic molecules), same atom name must be used for \sphinxcode{\sphinxupquote{Value 1}} and \sphinxcode{\sphinxupquote{Value 2}} of the \sphinxcode{\sphinxupquote{SmallKindBackBone}}.

\end{itemize}
\end{sphinxadmonition}

\end{description}

\sphinxAtStartPar
Here is the example of \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}2}} Monte Carlo moves, where 7 large\sphinxhyphen{}small molecule pairs are defined with an exchange ratio of 1:1: (ethane, methane), (propane, ethane), (n\sphinxhyphen{}butane, propane), (n\sphinxhyphen{}pentane, nbutane), (n\sphinxhyphen{}hexane, n\sphinxhyphen{}pentane), (n\sphinxhyphen{}heptane, n\sphinxhyphen{}hexane), and (noctane, n\sphinxhyphen{}heptane).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} MEMC PARAMETER
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
ExchangeVolumeDim   1.0   1.0   1.0
ExchangeRatio       1       1       1      1      1      1      1
ExchangeLargeKind   C8P    C7P    C6P    C5P    C4P    C3P    C2P
ExchangeSmallKind   C7P    C6P    C5P    C4P    C3P    C2P    C1P
LargeKindBackBone   C1 C8  C1 C7  C1 C6  C1 C5  C1 C4  C1 C3  C1 C2
SmallKindBackBone   C1 C7  C1 C6  C1 C5  C1 C4  C1 C3  C1 C2  C1 C1
\end{sphinxVerbatim}

\sphinxAtStartPar
Here is the example of \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}3}} Monte Carlo moves, where 7 large\sphinxhyphen{}small molecule pairs are defined with an exchange ratio of 1:1: (ethane, methane), (propane, ethane), (n\sphinxhyphen{}butane, propane), (n\sphinxhyphen{}pentane, nbutane), (n\sphinxhyphen{}hexane, n\sphinxhyphen{}pentane), (n\sphinxhyphen{}heptane, n\sphinxhyphen{}hexane), and (noctane, n\sphinxhyphen{}heptane).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} MEMC PARAMETER
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
ExchangeVolumeDim   1.0   1.0   1.0
ExchangeRatio       1       1       1      1      1      1      1
ExchangeLargeKind   C8P    C7P    C6P    C5P    C4P    C3P    C2P
ExchangeSmallKind   C7P    C6P    C5P    C4P    C3P    C2P    C1P
LargeKindBackBone   C4 C4  C4 C4  C3 C3  C3 C3  C2 C2  C2 C2  C1 C1
SmallKindBackBone   C1 C7  C1 C6  C1 C5  C1 C4  C1 C3  C1 C2  C1 C1
\end{sphinxVerbatim}

\sphinxAtStartPar
Here is the example of \sphinxcode{\sphinxupquote{MEMC\sphinxhyphen{}2}} Monte Carlo moves, where 1 large\sphinxhyphen{}small molecule pair is defined with an exchange ratio of 1:2: (xenon, methane).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} MEMC PARAMETER
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
ExchangeVolumeDim   5.0   5.0   5.0
ExchangeRatio       2
ExchangeLargeKind   XE
ExchangeSmallKind   C1P
LargeKindBackBone   Xe Xe
SmallKindBackBone   C1 C1
\end{sphinxVerbatim}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{SubVolumeBox}}}] \leavevmode
\sphinxAtStartPar
Define which box the subvolume occupies.
\sphinxhyphen{} Value 1: Integer \sphinxhyphen{} Sub\sphinxhyphen{}volume id.
\sphinxhyphen{} Value 2: Integer \sphinxhyphen{} Sets box number (first box is box ‘0’).

\item[{\sphinxcode{\sphinxupquote{SubVolumeCenter}}}] \leavevmode
\sphinxAtStartPar
Define the center of the static subvolume.
\sphinxhyphen{} Value 1: Integer \sphinxhyphen{} Sub\sphinxhyphen{}volume id.
\sphinxhyphen{} Value 2: Double \sphinxhyphen{} x value of SubVolumeCenter \(Å\).
\sphinxhyphen{} Value 3: Double \sphinxhyphen{} y value of SubVolumeCenter \(Å\).
\sphinxhyphen{} Value 4: Double \sphinxhyphen{} z value of SubVolumeCenter \(Å\).

\item[{\sphinxcode{\sphinxupquote{SubVolumePBC}}}] \leavevmode
\sphinxAtStartPar
Define which dimensions periodic box wrapping is applied in the subvolume.
\sphinxhyphen{} Value 1: Integer \sphinxhyphen{} Sub\sphinxhyphen{}volume id.
\sphinxhyphen{} Value 2: String \sphinxhyphen{} X, Y, Z, XY, XZ, YZ, XYZ (Axes should have no spaced between them)

\item[{\sphinxcode{\sphinxupquote{SubVolumeCenterList}}}] \leavevmode
\sphinxAtStartPar
Define the center of the dynamic subvolume by defining the atoms to use for the geometric mean calculation.
\sphinxhyphen{} Value 1: Integer \sphinxhyphen{} Sub\sphinxhyphen{}volume id.
\sphinxhyphen{} Value 2: Integer Range \sphinxhyphen{} Atom indices used to calculate geometric center of subvolume.

\item[{\sphinxcode{\sphinxupquote{SubVolumeDim}}}] \leavevmode
\sphinxAtStartPar
Define the dimensions of the subvolume.
\sphinxhyphen{} Value 1: Integer \sphinxhyphen{} Sub\sphinxhyphen{}volume id.
\sphinxhyphen{} Value 2: Double \sphinxhyphen{} x value of SubVolumeDim \(Å\).
\sphinxhyphen{} Value 3: Double \sphinxhyphen{} y value of SubVolumeDim \(Å\).
\sphinxhyphen{} Value 4: Double \sphinxhyphen{} z value of SubVolumeDim \(Å\).

\item[{\sphinxcode{\sphinxupquote{SubVolumeResidueKind}}}] \leavevmode
\sphinxAtStartPar
Define which residue kinds can be inserted or deleted from the subvolume.
\sphinxhyphen{} Value 1: Integer \sphinxhyphen{} Sub\sphinxhyphen{}volume id.
\sphinxhyphen{} Value 2: String \sphinxhyphen{} Residue kind inserted/deleted from subvolume
\sphinxhyphen{} Value .: String \sphinxhyphen{} Residue kind inserted/deleted from subvolume
\sphinxhyphen{} Value .: String \sphinxhyphen{} Residue kind inserted/deleted from subvolume
\sphinxhyphen{} Value N: String \sphinxhyphen{} Residue kind inserted/deleted from subvolume

\item[{\sphinxcode{\sphinxupquote{SubVolumeRigidSwap}}}] \leavevmode
\sphinxAtStartPar
Define whether molecules are held rigid or the geometry is sampled per the coupled\sphinxhyphen{}decoupled CBMC scheme.
\sphinxhyphen{} Value 1: Integer \sphinxhyphen{} Sub\sphinxhyphen{}volume id.
\sphinxhyphen{} Value 2: Boolean \sphinxhyphen{} If true the molecule is held rigid.  If false, geometry is sampled when inserting in the subvolume.

\item[{\sphinxcode{\sphinxupquote{SubVolumeChemPot}}}] \leavevmode
\sphinxAtStartPar
Define the chemical potential of a residue kind in the subvolume.  Only used in TargetedSwap, not IntraTargetedSwap.
\sphinxhyphen{} Value 1: Integer \sphinxhyphen{} Sub\sphinxhyphen{}volume id.
\sphinxhyphen{} Value 2: String \sphinxhyphen{} Residue kind
\sphinxhyphen{} Value 3: Double \sphinxhyphen{} Chemical potential

\item[{\sphinxcode{\sphinxupquote{SubVolumeFugacity}}}] \leavevmode
\sphinxAtStartPar
Define the fugacity of a residue kind in the subvolume.  Only used in TargetedSwap, not IntraTargetedSwap.
\sphinxhyphen{} Value 1: Integer \sphinxhyphen{} Sub\sphinxhyphen{}volume id.
\sphinxhyphen{} Value 2: String \sphinxhyphen{} Residue kind
\sphinxhyphen{} Value 3: Double \sphinxhyphen{} Chemical potential

\end{description}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} TARGETED SWAP (Static subVolume)
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
SubVolumeBox                  1       0
SubVolumeCenter               1       25.0 25.0 25.0
SubVolumeDim                  1       35 35 5
SubVolumeResidueKind          1       TIP3
SubVolumeRigidSwap            1       false
SubVolumeChemPot              1       TIP3    \PYGZhy{}800
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} TARGETED SWAP (Dynamic subVolume)
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
SubVolumeBox                  1       0
SubVolumeCenterList           1       1\PYGZhy{}402
SubVolumeDim                  1       35 35 5
SubVolumeResidueKind          1       TIP3
SubVolumeRigidSwap            1       false
SubVolumeChemPot              1       TIP3    \PYGZhy{}800
\end{sphinxVerbatim}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{useConstantArea}}}] \leavevmode
\sphinxAtStartPar
For Isobaric\sphinxhyphen{}Isothermal ensemble and Gibbs ensemble runs only: Considers to change the volume of the simulation box by fixing the cross\sphinxhyphen{}sectional area (x\sphinxhyphen{}y plane).
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} If true volume will change only in z axis, If false volume will change with constant axis ratio.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
By default, \sphinxcode{\sphinxupquote{useConstantArea}} will be set to false if no value was set. It means, the volume of the box will change in a way to maintain the constant axis ratio.
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{FixVolBox0}}}] \leavevmode
\sphinxAtStartPar
For adsorption simulation in NPT Gibbs ensemble runs only: Changing the volume of fluid phase (Box 1) to maintain the constant imposed pressure and temperature, while keeping the volume of adsorbed phase (Box 0) fix.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} If true volume of adsorbed phase will remain constant, If false volume of adsorbed phase will change.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{CellBasisVector}}}] \leavevmode
\sphinxAtStartPar
Defines the shape and size of the simulation periodic cell. \sphinxcode{\sphinxupquote{CellBasisVector1}}, \sphinxcode{\sphinxupquote{CellBasisVector2}}, \sphinxcode{\sphinxupquote{CellBasisVector3}} represent the cell basis vector \(a,b,c\), respectively. This tag may occur multiple times. It occurs once for NVT and NPT, but twice for Gibbs ensemble or GC ensemble.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Sets box number (first box is box ‘0’).

\item {} 
\sphinxAtStartPar
Value 2: Double \sphinxhyphen{} x value of cell basis vector \(Å\).

\item {} 
\sphinxAtStartPar
Value 3: Double \sphinxhyphen{} y value of cell basis vector \(Å\).

\item {} 
\sphinxAtStartPar
Value 4: Double \sphinxhyphen{} z value of cell basis vector \(Å\).

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
If the number of defined boxes were not compatible to simulation type, the program will be terminated.
\end{sphinxadmonition}

\sphinxAtStartPar
Example for NVT and NPT ensemble. In this example, each vector is perpendicular to the other two (\(\alpha = 90, \beta = 90, \gamma = 90\)), as indicated by a single x, y, or z value being specified by each and making a rectangular 3\sphinxhyphen{}D box:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} BOX DIMENSION \PYGZsh{}, X, Y, Z
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
CellBasisVector1  0   40.00   00.00   00.00
CellBasisVector2  0   00.00   40.00   00.00
CellBasisVector3  0   00.00   00.00   80.00
\end{sphinxVerbatim}

\sphinxAtStartPar
Example for Gibbs ensemble and GC ensemble ensemble. In this example, In the first box, only vector \(a\) and \(c\) are perpendicular to each other (\(\alpha = 90, \beta = 90, \gamma = 120\)), and making a non\sphinxhyphen{}orthogonal simulation cell with the cell length \(a = 36.91 Å, b = 36.91 Å, c = 76.98 Å\). In the second box, each vector is perpendicular to the other two (\(\alpha = 90, \beta = 90, \gamma = 90\)), as indicated by a single x, y, or z value being specified by each and making a cubic box:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} BOX DIMENSION \PYGZsh{}, X, Y, Z
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
CellBasisVector1  0   36.91   00.00   00.00
CellBasisVector2  0   \PYGZhy{}18.45  31.96   00.00
CellBasisVector3  0   00.00   00.00   76.98

CellBasisVector1  1   60.00   00.00   00.00
CellBasisVector2  1   00.00   60.00   00.00
CellBasisVector3  1   00.00   00.00   60.00
\end{sphinxVerbatim}

\begin{sphinxadmonition}{warning}{Warning:}
\sphinxAtStartPar
If \sphinxcode{\sphinxupquote{Restart}} was activated, box dimension does not need to be specified. If it is specified, program will read it but it will be ignored and replaced by the printed cell dimensions and angles in the restart PDB output file from GOMC (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_0\_restart.pdb and \sphinxcode{\sphinxupquote{OutputName}}\_BOX\_1\_restart.pdb).
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{CBMC\_First}}}] \leavevmode
\sphinxAtStartPar
Number of CD\sphinxhyphen{}CBMC trials to choose the first atom position (Lennard\sphinxhyphen{}Jones trials for first seed growth).
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Number of initial insertion sites to try.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{CBMC\_Nth}}}] \leavevmode
\sphinxAtStartPar
Number of CD\sphinxhyphen{}CBMC trials to choose the later atom positions (Lennard\sphinxhyphen{}Jones trials for first seed growth).
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Number of LJ trials for growing later atom positions.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{CBMC\_Ang}}}] \leavevmode
\sphinxAtStartPar
Number of CD\sphinxhyphen{}CBMC bending angle trials to perform for geometry (per the coupled\sphinxhyphen{}decoupled CBMC scheme).
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Number of trials per angle.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{CBMC\_Dih}}}] \leavevmode
\sphinxAtStartPar
Number of CD\sphinxhyphen{}CBMC dihedral angle trials to perform for geometry (per the coupled\sphinxhyphen{}decoupled CBMC scheme).
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} Number of trials per dihedral.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} CBMC TRIALS
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
CBMC\PYGZus{}First  10
CBMC\PYGZus{}Nth    4
CBMC\PYGZus{}Ang    100
CBMC\PYGZus{}Dih    30
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxstylestrong{Next section specifies the parameters that will be used for free energy calculation in NVT and NPT ensembles.}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{FreeEnergyCalc}}}] \leavevmode
\sphinxAtStartPar
For NVT and NPT ensemble only: Considers to calculate the free energy data (the energy different between current lambda
state and all other neighboring lambda states, and calculate the derivative of energy with respective to current lambda) or not.
If it is set to true, the frequency of free energy calculation need to be set. The free energy data will be printed into
Free\_Energy\_BOX\_0\_ \sphinxcode{\sphinxupquote{OutputName}}.dat.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True enabling free energy calculation during the simulation, false disabling the calculation.

\item {} 
\sphinxAtStartPar
Value 2: Ulong \sphinxhyphen{} The frequency of calculating the free energy.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{MoleculeType}}}] \leavevmode
\sphinxAtStartPar
Sets the solute molecule kind (residue name) and molecule number (residue ID), which absolute solvation free will be calculated for.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} The solute name (residue name).

\item {} 
\sphinxAtStartPar
Value 2: Integer \sphinxhyphen{} The solute molecule number (residue ID).

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{InitialState}}}] \leavevmode
\sphinxAtStartPar
Sets the index of the \sphinxcode{\sphinxupquote{LambdaCoulomb}} and \sphinxcode{\sphinxupquote{LambdaVDW}} vectors, to determine the simulation lambda value for VDW and Coulomb interactions.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} The index of \sphinxcode{\sphinxupquote{LambdaCoulomb}} and \sphinxcode{\sphinxupquote{LambdaVDW}} vectors.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
Multiple initial states need to be run to perform a free energy analysis on the system.

\end{itemize}
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{LambdaVDW}}}] \leavevmode
\sphinxAtStartPar
Sets the intermediate lambda states to which solute\sphinxhyphen{}solvent VDW interaction to be scaled.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} Lambda values for VDW interaction in ascending order.

\end{itemize}

\begin{sphinxadmonition}{warning}{Warning:}
\sphinxAtStartPar
All lambda values must be stated in the ascending order, otherwise the program will terminate.
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{LambdaCoulomb}}}] \leavevmode
\sphinxAtStartPar
Sets the intermediate lambda states to which solute\sphinxhyphen{}solvent Coulombic interaction to be scaled.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} Lambda values for Coulombic interaction in ascending order.

\end{itemize}

\begin{sphinxadmonition}{warning}{Warning:}
\sphinxAtStartPar
All lambda values must be stated in the ascending order, otherwise the program will terminate.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
By default, the lambda values for Coulombic interaction will be set to zero if \sphinxcode{\sphinxupquote{ElectroStatic}} or \sphinxcode{\sphinxupquote{Ewald}} is \sphinxstylestrong{deactivated}.

\item {} 
\sphinxAtStartPar
By default, the lambda values for Coulombic interaction will be set to Lambda values for VDW interaction if \sphinxcode{\sphinxupquote{ElectroStatic}} or \sphinxcode{\sphinxupquote{Ewald}} is \sphinxstylestrong{activated}.

\end{itemize}

\sphinxAtStartPar
\sphinxhyphen{}The LambdaVDW and LambdaCoulomb lists must be equal in length.
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{ScaleCoulomb}}}] \leavevmode
\sphinxAtStartPar
Determines to scale the Coulombic interaction non\sphinxhyphen{}linearly (soft\sphinxhyphen{}core scheme) or not.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} True if coulombic interaction needs to be scaled non\sphinxhyphen{}linearly, False if coulombic interaction needs to be scaled linearly.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
By default, the \sphinxcode{\sphinxupquote{ScaleCoulomb}} will be set to false.
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{ScalePower}}}] \leavevmode
\sphinxAtStartPar
Sets the \(p\) value in soft\sphinxhyphen{}core scaling scheme, where the distance between solute and solvent is scaled non\sphinxhyphen{}linearly.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Integer \sphinxhyphen{} The \(p\) value in the soft\sphinxhyphen{}core scaling scheme.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
By default, the \sphinxcode{\sphinxupquote{ScalePower}} will be set to 2.
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{ScaleAlpha}}}] \leavevmode
\sphinxAtStartPar
Sets the \(\alpha\) value in soft\sphinxhyphen{}core scaling scheme, where the distance between solute and solvent is scaled non\sphinxhyphen{}linearly.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} \(\alpha\) value in the soft\sphinxhyphen{}core scaling scheme.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
By default, the \sphinxcode{\sphinxupquote{ScaleAlpha}} will be set to 0.5.
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{MinSigma}}}] \leavevmode
\sphinxAtStartPar
Sets the minimum \(\sigma\) value in soft\sphinxhyphen{}core scaling scheme, where the distance between solute and solvent is scaled non\sphinxhyphen{}linearly.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Double \sphinxhyphen{} Minimum \(\sigma\) value in the soft\sphinxhyphen{}core scaling scheme.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
By default, the \sphinxcode{\sphinxupquote{MinSigma}} will be set to 3.0.
\end{sphinxadmonition}

\end{description}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Scaling the distance between solute and solvent using soft\sphinxhyphen{}core scheme:
\begin{equation*}
\begin{split}r_{sc} = \bigg[\alpha {\big(1 - \lambda \big)}^{p}{\sigma}^6 + {r}^6 \bigg]^{\frac{1}{6}}\end{split}
\end{equation*}\end{sphinxadmonition}

\sphinxAtStartPar
Here is the example of solvation free energy of CO2, in intermediate state 3.

\end{description}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} FREE ENERGY PARAMETERS
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
FreeEnergyCalc true   1000
MoleculeType   CO2   1
InitialState   3
ScalePower     2
ScaleAlpha     0.5
MinSigma       3.0
ScaleCoulomb   false
\PYGZsh{}states        0    1    2    3    4
LambdaVDW      0.00 0.50 1.00 1.00 1.00
LambdaCoulomb  0.00 0.00 0.00 0.50 1.00
\end{sphinxVerbatim}


\subsection{Output Controls}
\label{\detokenize{input_file:output-controls}}
\sphinxAtStartPar
This section contains all the values that control output in the control file. For example, certain variables control the naming of files outputed of the block\sphinxhyphen{}averaged thermodynamic variables of interest, the PDB files, etc.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{OutputName}}}] \leavevmode
\sphinxAtStartPar
Unique name with no space for simulation used to name the block average, PDB, and PSF output files.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Unique phrase to identify this system.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} OUTPUT FILE NAME
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
OutputName  ISB\PYGZus{}T\PYGZus{}270\PYGZus{}K
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{CoordinatesFreq}}}] \leavevmode
\sphinxAtStartPar
Controls output of PDB file (coordinates). If PDB outputing was enabled, one file for NVT or NPT and
two files for Gibbs ensemble or GC ensemble will be outputed into \sphinxcode{\sphinxupquote{OutputName}}\_BOX\_n.pdb, where n defines the box number.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} “true” enables outputing these files; “false” disables outputing.

\item {} 
\sphinxAtStartPar
Value 2: Ulong \sphinxhyphen{} Steps per dump PDB frame. It should be less than or equal to RunSteps. If this
keyword could not be found in configuration file, its value will be assigned a default value to dump 10 frames.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
DCDFreq should be used unless the low precision and slower PDB trajectory is needed,
perhaps beta and occupancy values are desired.  The PDB trajectory is much larger and consumes more disk space.

\item {} 
\sphinxAtStartPar
The PDB file contains an entry for every ATOM, in all boxes read. This allows VMD (which requires a
constant number of atoms) to properly parse frames, with a bit of help. Atoms that are not currently
in a specific box are given the coordinate (0.00, 0.00, 0.00). The occupancy value corresponds to the
box a molecule is currently in (e.g. 0.00 for box 0; 1.00 for box 1).

\item {} 
\sphinxAtStartPar
At the beginning of simulation, a merged PSF file will be outputed into \sphinxcode{\sphinxupquote{OutputName}}\_merged.psf,
in which all boxes will be outputed. It also contains the topology for every molecule in both boxes,
corresponding to the merged PDB format. Loading PDB files into merged PSF file in VMD allows the user
to visualize and analyze the results.

\end{itemize}
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{DCDFreq}}}] \leavevmode
\sphinxAtStartPar
Controls output of DCD file (binary coordinates). If DCD outputing was enabled, one file for NVT or NPT and
two files for Gibbs ensemble or GC ensemble will be outputed into \sphinxcode{\sphinxupquote{OutputName}}\_BOX\_n.dcd, where n defines the box number.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} “true” enables outputing these files; “false” disables outputing.

\item {} 
\sphinxAtStartPar
Value 2: Ulong \sphinxhyphen{} Steps per dump PDB frame. It should be less than or equal to RunSteps. If this
keyword could not be found in configuration file, its value will be assigned a default value to dump 10 frames.

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
The DCD file contains an entry for every ATOM, in all boxes read. This allows VMD (which requires a
constant number of atoms) to properly parse frames, with a bit of help. Atoms that are not currently
in a specific box are given the coordinate (0.00, 0.00, 0.00). The occupancy value corresponds to the
box a molecule is currently in (e.g. 0.00 for box 0; 1.00 for box 1).

\item {} 
\sphinxAtStartPar
At the beginning of simulation, a merged PSF file will be outputed into \sphinxcode{\sphinxupquote{OutputName}}\_merged.psf,
in which all boxes will be outputed. It also contains the topology for every molecule in both boxes,
corresponding to the merged PDB format. Loading DCD files into merged PSF file in VMD allows the user
to visualize and analyze the results.

\item {} 
\sphinxAtStartPar
“The DCD files are single precision binary FORTRAN files, so are transportable between computer architectures. The file readers in NAMD and VMD can detect and adapt to the endianness of the machine on which the DCD file was written, and the utility program flipdcd is also provided to reformat these files if needed. The exact format of these files is very ugly but supported by a wide range of analysis and display programs. The timestep is stored in the DCD file in NAMD internal units and must be multiplied by TIMEFACTOR=48.88821 to convert to fs. Positions in DCD files are stored in Å. Velocities in DCD files are stored in NAMD internal units and must be multiplied by PDBVELFACTOR=20.45482706 to convert to Å/ps. Forces in DCD files are stored in kcal/mol/Å.”

\item {} 
\sphinxAtStartPar
source : \sphinxurl{https://www.ks.uiuc.edu/Research/namd/2.9/ug/node11.html}

\end{itemize}
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{RestartFreq}}}] \leavevmode
\sphinxAtStartPar
Controls the output of the last state of simulation at a specified step in
\begin{itemize}
\item {} 
\sphinxAtStartPar
PDB files (coordinates)

\item {} 
\sphinxAtStartPar
PSF files (structure)

\item {} 
\sphinxAtStartPar
XSC files (binary box dimensions)

\item {} 
\sphinxAtStartPar
COOR files (binary coordinates)

\item {} 
\sphinxAtStartPar
CHK files (checkpoint)

\item {} 
\sphinxAtStartPar
If provided as input: VEL files (binary velocity)

\end{itemize}

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_n\_restart.*, where n defines the box number. Header part of this file contains
important information and will be needed to restart the simulation:

\sphinxAtStartPar
Restart PDB files, one file for NVT or NPT and two files for Gibbs ensemble or GC ensemble, will be outputed with the following information.
\sphinxhyphen{} Simulation cell dimensions and angles.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Maximum amount of displacement (Å), rotation (\(\delta\)), and volume (\(Å^3\)) that used in Displacement, Rotation, and Volume move.

\end{itemize}

\end{description}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
The restart PDB/PSF/COOR/VEL files contains only ATOM that exist in each boxes at specified steps.  These box restart files allows the user to load a box into NAMD and run molecular dynamics in Hybrid Monte\sphinxhyphen{}Carlo Molecular Dynamics (py\sphinxhyphen{}MCMD).

\item {} 
\sphinxAtStartPar
When restarting the GOMC simulation from two restart files, the order of the molecules in the trajectory may differ preventing trajectory concatenation, unless the CHK file is loaded.

\item {} 
\sphinxAtStartPar
Only restart files must be used to begin a GOMC simulation with \sphinxcode{\sphinxupquote{Restart}} simulation active.  The merged psf is NOT a restart file.

\item {} 
\sphinxAtStartPar
CoordinatesFreq must be a common multiple of RestartFreq or vice versa.

\end{itemize}
\end{sphinxadmonition}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{ConsoleFreq}}}] \leavevmode
\sphinxAtStartPar
Controls the output to STDIO (“the console”) of messages such as acceptance statistics, and run timing info. In addition, instantaneously\sphinxhyphen{}selected thermodynamic properties will be output to this file.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} “true” enables message printing; “false” disables outputing.

\item {} 
\sphinxAtStartPar
Value 2: Ulong \sphinxhyphen{} Number of steps per print. If this keyword could not be found in the configuration file, the value will be assigned by default to dump 1000 output for RunSteps greater than 1000 steps and 100 output for RunSteps less than 1000 steps.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{BlockAverageFreq}}}] \leavevmode
\sphinxAtStartPar
Controls the block averages output of selected thermodynamic properties. Block averages are averages of thermodynamic values of interest for chunks of the simulation (for post\sphinxhyphen{}processing of averages or std. dev. in those values).
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} “true” enables printing block average; “false” disables it.

\item {} 
\sphinxAtStartPar
Value 2: Ulong \sphinxhyphen{} Number of steps per block\sphinxhyphen{}average output file. If this keyword cannot be found in the configuration file, its value will be assigned a default to dump 100 output.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{HistogramFreq}}}] \leavevmode
\sphinxAtStartPar
Controls the histograms. Histograms are a binned listing of observation frequency for a specific thermodynamic variable. In this code, they also control the output of a file containing energy/molecule samples; it only will be used in GC ensemble simulations for histogram reweighting purposes.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \sphinxhyphen{} “true” enables printing histogram; “false” disables it.

\item {} 
\sphinxAtStartPar
Value 2: Ulong \sphinxhyphen{} Number of steps per histogram output file. If this keyword cannot be found in the configuration file, a value will be assigned by default to dump 1000 output for RunSteps greater than 1000 steps and 100 output for RunSteps less than 1000 steps.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} STATISTICS Enable, Freq.
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
CoordinatesFreq   true 10000000
RestartFreq       true 1000000
CheckpointFreq    true 1000000
ConsoleFreq       true 100000
BlockAverageFreq  true 100000
HistogramFreq     true 10000
\end{sphinxVerbatim}

\end{description}

\sphinxAtStartPar
The next section controls the output of the energy/molecule sample file and the distribution file f
or molecule counts, commonly referred to as the “histogram” output. This section is only required
if Grand Canonical ensemble simulation was used.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{DistName}}}] \leavevmode
\sphinxAtStartPar
Sets short phrase to naming molecule distribution file.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Short phrase which will be combined with \sphinxstyleemphasis{RunNumber} and \sphinxstyleemphasis{RunLetter} to use in the name of the binned histogram for molecule distribution.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{HistName}}}] \leavevmode
\sphinxAtStartPar
Sets short phrase to naming energy sample file.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: String \sphinxhyphen{} Short phrase, which will be combined with \sphinxstyleemphasis{RunNumber} and \sphinxstyleemphasis{RunLetter}, to use in the name of the energy/molecule count sample file.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{RunNumber}}}] \leavevmode
\sphinxAtStartPar
Sets a number, which is a part of \sphinxstyleemphasis{DistName} and \sphinxstyleemphasis{HistName} file name.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Uint \textendash{} Run number to be used in the above file names.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{RunLetter}}}] \leavevmode
\sphinxAtStartPar
Sets a letter, which is a part of \sphinxstyleemphasis{DistName} and \sphinxstyleemphasis{HistName} file name.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Character \textendash{} Run letter to be used in above file names.

\end{itemize}

\item[{\sphinxcode{\sphinxupquote{SampleFreq}}}] \leavevmode
\sphinxAtStartPar
Controls histogram sampling frequency.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Uint \textendash{} the number of steps per histogram sample.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} OutHistSettings
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
DistName   dis
HistName   his
RunNumber  5
RunLetter  a
SampleFreq 200
\end{sphinxVerbatim}

\item[{\sphinxcode{\sphinxupquote{OutEnergy, OutPressure, OutMolNumber, OutDensity, OutVolume, OutSurfaceTension}}}] \leavevmode
\sphinxAtStartPar
Enables/Disables for specific kinds of file output for tracked thermodynamic quantities
\begin{itemize}
\item {} 
\sphinxAtStartPar
Value 1: Boolean \textendash{} “true” enables message output of block averages via this tracked parameter (and in some cases such as entry, components); “false” disables it.

\item {} 
\sphinxAtStartPar
Value 2: Boolean \textendash{} “true” enables message output of a fluctuation into the console file via this tracked parameter (and in some cases, such as entry, components); “false” disables it.

\end{itemize}

\sphinxAtStartPar
The keywords are available for the following ensembles


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|T|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Keyword
&\sphinxstyletheadfamily 
\sphinxAtStartPar
NVT
&\sphinxstyletheadfamily 
\sphinxAtStartPar
NPT \& Gibbs
&\sphinxstyletheadfamily 
\sphinxAtStartPar
GC
\\
\hline
\sphinxAtStartPar
OutEnergy
&
\sphinxAtStartPar
\(\checkmark\)
&
\sphinxAtStartPar
\(\checkmark\)
&
\sphinxAtStartPar
\(\checkmark\)
\\
\hline
\sphinxAtStartPar
OutPressure
&
\sphinxAtStartPar
\(\checkmark\)
&
\sphinxAtStartPar
\(\checkmark\)
&
\sphinxAtStartPar
\(\checkmark\)
\\
\hline
\sphinxAtStartPar
OutMolNumber
&&
\sphinxAtStartPar
\(\checkmark\)
&
\sphinxAtStartPar
\(\checkmark\)
\\
\hline
\sphinxAtStartPar
OutDensity
&&
\sphinxAtStartPar
\(\checkmark\)
&
\sphinxAtStartPar
\(\checkmark\)
\\
\hline
\sphinxAtStartPar
OutVolume
&&
\sphinxAtStartPar
\(\checkmark\)
&
\sphinxAtStartPar
\(\checkmark\)
\\
\hline
\sphinxAtStartPar
OutSurfaceTension
&
\sphinxAtStartPar
\(\checkmark\)
&&\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
Here is an example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} ENABLE: BLK AVE., FLUC.
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
OutEnergy         true true
OutPressure       true true
OutMolNum         true true
OutDensity        true true
OutVolume         true true
OutSurfaceTension false false
\end{sphinxVerbatim}

\end{description}


\chapter{GOMC’s Output Files}
\label{\detokenize{output_file:gomc-s-output-files}}\label{\detokenize{output_file::doc}}\begin{description}
\item[{GOMC currently supports several kinds of output:}] \leavevmode\begin{itemize}
\item {} 
\sphinxAtStartPar
STDIO (“console”) output

\item {} 
\sphinxAtStartPar
File output
\begin{itemize}
\item {} 
\sphinxAtStartPar
Block Averages

\item {} 
\sphinxAtStartPar
PDB

\item {} 
\sphinxAtStartPar
PSF

\item {} 
\sphinxAtStartPar
Molecule distribution (GCMC only)

\item {} 
\sphinxAtStartPar
Histogram (GCMC only)

\item {} 
\sphinxAtStartPar
Free energy data (NVT and NPT only)

\end{itemize}

\end{itemize}

\end{description}

\sphinxAtStartPar
GOMC output units:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Properties
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Units
\\
\hline
\sphinxAtStartPar
Energy
&
\sphinxAtStartPar
\(K\)
\\
\hline
\sphinxAtStartPar
Pressure, Pressure Tensor
&
\sphinxAtStartPar
bar
\\
\hline
\sphinxAtStartPar
Heat of vaporization
&
\sphinxAtStartPar
\(kJ/mol\)
\\
\hline
\sphinxAtStartPar
Volume
&
\sphinxAtStartPar
\(\AA^3\)
\\
\hline
\sphinxAtStartPar
Density
&
\sphinxAtStartPar
\(kg/m^3\)
\\
\hline
\sphinxAtStartPar
Mol Density
&
\sphinxAtStartPar
\(molecule/Å^3\)
\\
\hline
\sphinxAtStartPar
Surface Tension
&
\sphinxAtStartPar
\(mN/m\)
\\
\hline
\sphinxAtStartPar
Free Energy
&
\sphinxAtStartPar
\(kJ/mol\)
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}


\section{Console Output}
\label{\detokenize{output_file:console-output}}
\sphinxAtStartPar
A variety of useful information relating to instantaneous statistical and thermodynamic data (move trials, acceptance rates, file I/O messages warnings, and other kinds of information) is printed to the STDIO, which, in Linux, will typically be displayed in the terminal. This output can be redirected into a log file in Linux using the \sphinxcode{\sphinxupquote{\textgreater{}}} operator.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} GOMC CPU NVT in.conf \PYGZgt{} out\PYGZus{}isobutane.log \PYG{p}{\PYGZam{}}
\end{sphinxVerbatim}

\sphinxAtStartPar
Statistical and thermodynamic information is provided in console output.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Energy

\sphinxAtStartPar
\textendash{} Intermolecular (LJ)

\sphinxAtStartPar
\textendash{} Intramolecular bonded

\sphinxAtStartPar
\textendash{} Intramolecular nonbonded

\sphinxAtStartPar
\textendash{} Tail corrections

\sphinxAtStartPar
\textendash{} Electrostatic real

\sphinxAtStartPar
\textendash{} Electrostatic Reciprocal

\sphinxAtStartPar
\textendash{} Electrostatic self

\sphinxAtStartPar
\textendash{} Electrostatic correction

\sphinxAtStartPar
\textendash{} Total electrostatic energy (sum of real, reciprocal, self, and correction)

\sphinxAtStartPar
\textendash{} Total Energy (sum of the all energies)

\item {} 
\sphinxAtStartPar
Pressure
\begin{itemize}
\item {} 
\sphinxAtStartPar
Pressure Tensor (\(P_{xx},P_{yy},P_{zz}\))

\item {} 
\sphinxAtStartPar
Total pressure

\end{itemize}

\item {} 
\sphinxAtStartPar
Statistic
\begin{itemize}
\item {} 
\sphinxAtStartPar
Volume

\item {} 
\sphinxAtStartPar
Pressure

\item {} 
\sphinxAtStartPar
Total molecule number

\item {} 
\sphinxAtStartPar
Total Density

\item {} 
\sphinxAtStartPar
Surface Tension

\item {} 
\sphinxAtStartPar
Mol fraction of each species

\item {} 
\sphinxAtStartPar
Mol density of each species

\end{itemize}

\end{itemize}

\sphinxAtStartPar
Detailed move, energy, and statistical or thermodynamic information for each simulation box will be printed in
three different sections. Each section’s title will start with \sphinxcode{\sphinxupquote{MTITLE}}, \sphinxcode{\sphinxupquote{ETITLE}}, and \sphinxcode{\sphinxupquote{STITLE}} for move,
energy, and statistical information, respectively. The instantaneous values for each section will start with
\sphinxcode{\sphinxupquote{MOVE\_\#}}, \sphinxcode{\sphinxupquote{ENER\_\#}}, and \sphinxcode{\sphinxupquote{STAT\_\#}} for move, energy, and statistical values, respectively. Where, \# is the
simulation box number. In addition, if pressure calculation is activated and enabled to print, pressure tensor
will be printed in the console output file. This section starts with \sphinxcode{\sphinxupquote{PRES\_\#}} and print the diagonal value of
pressure tensor \(P_{xx}\), \(P_{yy}\),and \(P_{zz}\), respectively. The second element after the
title of each section is the step number.

\sphinxAtStartPar
In order to extract the desired information from the console file, “grep” and “awk” commands can be used with
a proper title section. For example, in order to extract total energy of the system, the following command needs
to be executed in terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} grep \PYG{l+s+s2}{\PYGZdq{}ENER\PYGZus{}0\PYGZdq{}} output\PYGZus{}console.log \PYG{p}{|} awk \PYG{l+s+s1}{\PYGZsq{}\PYGZob{}print \PYGZdl{}3\PYGZcb{}\PYGZsq{}}
\end{sphinxVerbatim}

\sphinxAtStartPar
Here, “output\_console.log” is the console output file and “\$3” represents the second element of the “ENERGY\_BOX\_0” section.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Surface Tension is calculated using Virial method according to following equation,
\begin{equation*}
\begin{split}\gamma = \frac{1}{2A_{xy}} \int_{0}^{L} \bigg(P_{zz} - \frac{P_{xx} + P_{yy}}{2} \bigg) dz\end{split}
\end{equation*}\end{sphinxadmonition}

\sphinxAtStartPar
The first section of this console output typically includes some information relating the system, CPU, GPU, and RAM. In continue, console output includes information regarding the input file (configuration file), force field reading, summary of the structure of the molecule, bonded and non\sphinxhyphen{}bonded parameters, and minimum and maximum coordinate of molecules. This output is important; it may contain text relating to issues encountered if there was an error in the current run (e.g. a bad parameter, unknown keyword, missing parameters in the configuration file, etc.)

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{out1}.png}
\caption{Printing summary of configuration file.}\label{\detokenize{output_file:id1}}\end{figure}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{out2}.png}
\caption{Reading parameter file and printing the summary of the force field.}\label{\detokenize{output_file:id2}}\end{figure}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{out3}.png}
\caption{Reading the PDB files for each box, printing the min and max coordinates.}\label{\detokenize{output_file:id3}}\end{figure}

\sphinxAtStartPar
Next, the energy and statistic title, initial energy and statistic of the system’s starting configuration will print:

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The frequency of printing \sphinxcode{\sphinxupquote{MOVE\_\#}}, \sphinxcode{\sphinxupquote{ENER\_\#}}, \sphinxcode{\sphinxupquote{STAT\_\#}}, and \sphinxcode{\sphinxupquote{PRES\_\#}} is controlled by \sphinxcode{\sphinxupquote{ConsoleFreq}}
parameter in configuration file.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
User can control the output of the thermodynamic properties in \sphinxcode{\sphinxupquote{ENER\_\#}} and \sphinxcode{\sphinxupquote{STAT\_\#}} using the following
parameters in configuration file:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} ENABLE:         BLK, FLUC.
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
OutEnergy         true  true
OutPressure       true  true
OutMolNum         true  true
OutDensity        true  true
OutVolume         true  true
OutSurfaceTension false false
\end{sphinxVerbatim}
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
If total energy of simulation is greater that \(1.0e^{12}\), System Total Energy
Calculation will be performed at EqSteps to preserve energy value.
\end{sphinxadmonition}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{out4}.png}
\caption{Printing initial energy of the system and statistical values.}\label{\detokenize{output_file:id4}}\end{figure}

\sphinxAtStartPar
After the simulation starts, move, energy, and statistical title, followed by their values for each simulation box, will print:

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{out5}.png}
\end{figure}

\sphinxAtStartPar
At the end of the run, Monte Carlo move acceptance for each molecule kind and simulation box, total amount of time spent on each
Monte Carlo move, total timing information, and other wrap up info will be printed.

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
Printed energy and statistical values are instantaneous values.

\item {} 
\sphinxAtStartPar
In order to keep the format of console file consistent and print the calculated properties with high accuracy, scientific format is used.

\item {} 
\sphinxAtStartPar
It’s important to watch the acceptance rates and adjust the move percentages and CBMC trial amounts to get the desired rate of move acceptance.

\end{itemize}
\end{sphinxadmonition}


\section{Block Output Files}
\label{\detokenize{output_file:block-output-files}}
\sphinxAtStartPar
GOMC tracks a number of thermodynamic variables of interest during the simulation and prints them all in one file for each box.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Energy

\sphinxAtStartPar
\textendash{} Intermolecular (LJ)

\sphinxAtStartPar
\textendash{} Intramolecular bonded

\sphinxAtStartPar
\textendash{} Intramolecular nonbonded

\sphinxAtStartPar
\textendash{} Tail corrections

\sphinxAtStartPar
\textendash{} Electrostatic real

\sphinxAtStartPar
\textendash{} Electrostatic Reciprocal

\sphinxAtStartPar
\textendash{} Total Energy (sum of the all energies)

\item {} 
\sphinxAtStartPar
Virial

\item {} 
\sphinxAtStartPar
Statistic
\begin{itemize}
\item {} 
\sphinxAtStartPar
Pressure

\item {} 
\sphinxAtStartPar
Surface Tension (using virial method)

\item {} 
\sphinxAtStartPar
Volume

\item {} 
\sphinxAtStartPar
Total molecule number

\item {} 
\sphinxAtStartPar
Total Density

\item {} 
\sphinxAtStartPar
Mol fraction of each species

\item {} 
\sphinxAtStartPar
Mol density of each species

\item {} 
\sphinxAtStartPar
Heat of vaporization

\end{itemize}

\end{itemize}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{Blk}.png}
\caption{Printing the average energy of the system and statistical values.}\label{\detokenize{output_file:id5}}\end{figure}

\sphinxAtStartPar
At the beginning of each file, the title of each property followed by their average values is printed.
Desired data can be extracted, as explained before, using the “awk” command. For example, in order to
extract total density of the system, the following command need to be executed in terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} cat Blk\PYGZus{}OutputName\PYGZus{}BOX\PYGZus{}0.dat \PYG{p}{|} awk \PYG{l+s+s1}{\PYGZsq{}\PYGZob{}print \PYGZdl{}2\PYGZcb{}\PYGZsq{}}
\end{sphinxVerbatim}

\sphinxAtStartPar
Here, “Blk\_OutputName\_BOX\_0.dat” is the block\sphinxhyphen{}average file for simulation box 0 and “\$2” represents the
second column of the block file.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The frequency of printing average thermodynamic properties is controlled by \sphinxcode{\sphinxupquote{BlockAverageFreq}}
parameter in configuration file.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
User can control the output of the average thermodynamic properties,  using the following
parameters in configuration file:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} ENABLE:         BLK, FLUC.
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
OutEnergy         true  true
OutPressure       true  true
OutMolNum         true  true
OutDensity        true  true
OutVolume         true  true
OutSurfaceTension false false
\end{sphinxVerbatim}
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In order to keep the format of BlockOutput file consistent and print the calculated properties
with high accuracy, scientific format is used.
\end{sphinxadmonition}


\section{PDB Output Files}
\label{\detokenize{output_file:pdb-output-files}}
\sphinxAtStartPar
GOMC capables of outputing the molecular coordinates during the simulation in PDB format.
GOMC outputs two type of PDB files:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
The last state of simulation at a specified step (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_n.pdb,
where n defines the box number).

\item {} 
\sphinxAtStartPar
The state of simulation at a specified step (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_n\_restart.pdb,
where n defines the box number).

\end{enumerate}


\subsection{1.  Restart Trajectory}
\label{\detokenize{output_file:restart-trajectory}}
\sphinxAtStartPar
The restart PDB file contains only ATOM that exist in each boxes at specified steps. This allows the
user to load this file into GOMC once \sphinxcode{\sphinxupquote{Restart}} simulation was active. If restart PDB output was enabled,
one file for NVT or NPT and two files for Gibbs ensemble or grand canonical ensemble will be outputed.
Header part of this file contains important information and will be needed to restart the simulation:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Simulation cell dimensions and angles.

\item {} 
\sphinxAtStartPar
Maximum amount of displacement (Å), rotation (\(\delta\)), and volume (\(\AA^3\)) that used in Displacement,
Rotation, and Volume move.

\end{itemize}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{pdb_restart_0}.png}
\caption{The coordinates of isobutane molecules in simulation Box 0, at steps 30000, in \sphinxcode{\sphinxupquote{OutputName}}\_BOX\_0\_restart.pdb file.}\label{\detokenize{output_file:id6}}\end{figure}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{pdb_restart_1}.png}
\caption{The coordinates of isobutane molecules in simulation Box 1, at steps 30000, in \sphinxcode{\sphinxupquote{OutputName}}\_BOX\_1\_restart.pdb file.}\label{\detokenize{output_file:id7}}\end{figure}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The frequency of printing restart PDB file is controlled by \sphinxcode{\sphinxupquote{RestartFreq}}
parameter in configuration file.
\end{sphinxadmonition}

\begin{sphinxadmonition}{important}{Important:}
\sphinxAtStartPar
The beta value in restart PDB file defines the mobility of the molecule.
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{Beta = 0.00}}: molecule can move and transfer within and between boxes.

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{Beta = 1.00}}: molecule is fixed in its position.

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{Beta = 2.00}}: molecule can move within the box but cannot be transferred between boxes.

\end{itemize}
\end{sphinxadmonition}


\subsection{2.  Simulation Trajectories}
\label{\detokenize{output_file:simulation-trajectories}}
\sphinxAtStartPar
The trajectory PDB file contains an entry for every ATOM, in all boxes read. This allows VMD
(which requires a constant number of atoms) to properly parse the simulation frames. If PDB output was enabled,
one file for NVT or NPT and two files for Gibbs ensemble or grand canonical ensemble will be outputed.
Header part of this file contains simulation cell dimensions and angles, frame number, and simulation steps.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{pdb_0}.png}
\caption{The coordinates of all isobutane molecules at beginning of the simulation, in \sphinxcode{\sphinxupquote{OutputName}}\_BOX\_0.pdb file.}\label{\detokenize{output_file:id8}}\end{figure}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{pdb_1}.png}
\caption{The coordinates of all isobutane molecules at beginning of the simulation, in \sphinxcode{\sphinxupquote{OutputName}}\_BOX\_1.pdb file.}\label{\detokenize{output_file:id9}}\end{figure}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The frequency of printing trajectory PDB file is controlled by \sphinxcode{\sphinxupquote{CoordinatesFreq}}
parameter in configuration file.
\end{sphinxadmonition}

\begin{sphinxadmonition}{important}{Important:}\begin{itemize}
\item {} 
\sphinxAtStartPar
For atoms not currently in a box, the coordinates are set to \sphinxcode{\sphinxupquote{\textless{} 0.00, 0.00, 0.00 \textgreater{}}}.

\item {} 
\sphinxAtStartPar
The occupancy value defines the box, which molecule is in (box 0 occupancy=0.00 ; box 1 occupancy=1.00)

\item {} 
\sphinxAtStartPar
The beta value in trajectory PDB file defines the mobility of the molecule.
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{Beta = 0.00}}: molecule can move and transfer within and between boxes.

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{Beta = 1.00}}: molecule is fixed in its position.

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{Beta = 2.00}}: molecule can move within the box but cannot be transferred between boxes.

\end{itemize}

\end{itemize}
\end{sphinxadmonition}


\section{PSF Output File}
\label{\detokenize{output_file:psf-output-file}}
\sphinxAtStartPar
At the beginning of the simulation, a merged PSF file will be outputed into \sphinxcode{\sphinxupquote{OutputName}}\_merged.psf,
It contains the topology for every molecule in all simulation boxes, corresponding to the merged PDB format.
Loading PDB files into merged PSF file in VMD allows the user to visualize and analyze the results.

\sphinxAtStartPar
The PSF file contains six main sections: \sphinxcode{\sphinxupquote{remarks}}, \sphinxcode{\sphinxupquote{atoms}}, \sphinxcode{\sphinxupquote{bonds}}, \sphinxcode{\sphinxupquote{angles}}, and \sphinxcode{\sphinxupquote{dihedrals}}.
Each section starts with a specific header described bellow:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NTITLE}}: Remarks on the file.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{merged_psf_remark}.png}
\caption{Remaks generated by GOMC.}\label{\detokenize{output_file:id10}}\end{figure}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NATOM}}: The atom names, residue name, atom types, and partial charges of each atom.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{merged_psf_atom}.png}
\caption{Atom section, taken from a merged PSF file for isobutane. The fields in the atom section,
from left to right are atom ID, segment name, residue ID, residue name, atom name, atom type,
charge, mass, and an unused 0.}\label{\detokenize{output_file:id11}}\end{figure}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NBOND}}: The covalent bond section lists four pairs of atoms per line.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{merged_psf_bond}.png}
\caption{Bond section, taken from a merged PSF file for isobutane.}\label{\detokenize{output_file:id12}}\end{figure}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NTHETA}}: The angle section lists three triples of atoms per line.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{merged_psf_angle}.png}
\caption{Angle section, taken from a merged PSF file for isobutane.}\label{\detokenize{output_file:id13}}\end{figure}

\item {} 
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{NPHI}}: The dihedral sections list two quadruples of atoms per line.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{merged_psf_dihedral}.png}
\caption{Dihedral section, taken from a merged PSF file for isobutane.}\label{\detokenize{output_file:id14}}\end{figure}

\end{itemize}


\section{Molecule Distribution Output File}
\label{\detokenize{output_file:molecule-distribution-output-file}}
\sphinxAtStartPar
In grand canonical Monte Carlo (GCMC) simulation, GOMC outputs a binned number of molecules, observed
in the system. This file can be used to detect the overlap between various GCMC simulation states.
Sufficient overlap between various GCMC simulation is required in histogram reweighting method.

\sphinxAtStartPar
The molecule distribution will be outputed to a file, with a name constructed from parameters defined
in configuration file (\sphinxcode{\sphinxupquote{DistName}}, \sphinxcode{\sphinxupquote{RunNumber}}, and \sphinxcode{\sphinxupquote{RunLetter}}). For instance, for the first
molecule kind and following parameters in configuration file

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} OutHistSettings
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
DistName   dis
RunNumber  3
RunLetter  a
\end{sphinxVerbatim}

\sphinxAtStartPar
GOMC will output the molecule distribution into “n1dis3a.dat” file.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{ndis}.png}
\caption{Molecule number distribution taken for isobutane simulation in GCMC simulation. The field in
molecule distribution file, from left to right are number of molecule observed in the
simulation and number of samples.}\label{\detokenize{output_file:id15}}\end{figure}

\begin{sphinxadmonition}{important}{Important:}
\sphinxAtStartPar
In case of system with multiple molecule kinds, multiple molecule distribution files will be
outputed by GOMC (“n1dis3a.dat”, “n2dis3a.dat”, …).
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
The Molecule distribution files will be outputed at \sphinxcode{\sphinxupquote{EqSteps}}.

\item {} 
\sphinxAtStartPar
The frequency of outputing molecule distribution file is controlled by \sphinxcode{\sphinxupquote{HistogramFreq}}
parameter in configuration file.

\item {} 
\sphinxAtStartPar
The observation frequency is ontrolled by \sphinxcode{\sphinxupquote{SampleFreq}} parameter in configuration file.

\end{itemize}
\end{sphinxadmonition}


\section{Histogram Output File}
\label{\detokenize{output_file:histogram-output-file}}
\sphinxAtStartPar
In grand canonical Monte Carlo (GCMC) simulation, GOMC outputs the observed number of molecule (for each
molecule kind) and energy of the system (nonbonded + LRC). This file only will be used for histogram
reweighting purposes.

\sphinxAtStartPar
The histogram will be outputed to a file, with a name constructed from parameters defined
in configuration file (\sphinxcode{\sphinxupquote{HistName}}, \sphinxcode{\sphinxupquote{RunNumber}}, and \sphinxcode{\sphinxupquote{RunLetter}}). For instance, for
the following parameters in configuration file

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} OutHistSettings
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
HistName   his
RunNumber  3
RunLetter  a
\end{sphinxVerbatim}

\sphinxAtStartPar
GOMC will output the histogram data into “his3a.dat” file.

\sphinxAtStartPar
The header of the histogram file contains information of the simulated system, such as temperature,
number of molecule kind, chemical potential, and x, y, z dimensions of simulation box.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{hist}.png}
\caption{The histogram taken for isobutane simulation in GCMC simulation. The field in
histogram file, from left to right are number of molecule observed for the first molecule kind
in the simulation and energy of the system (nonbonded + LRC).}\label{\detokenize{output_file:id16}}\end{figure}

\begin{sphinxadmonition}{important}{Important:}
\sphinxAtStartPar
In case of system with multiple molecule kinds, multiple column will be printed, which each column
represents the number of molecule for each molecule kind.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}\begin{itemize}
\item {} 
\sphinxAtStartPar
The Histogram file will be outputed at \sphinxcode{\sphinxupquote{EqSteps}}.

\item {} 
\sphinxAtStartPar
The frequency of outputing Histogram file is controlled by \sphinxcode{\sphinxupquote{HistogramFreq}}
parameter in configuration file.

\item {} 
\sphinxAtStartPar
The observation frequency is ontrolled by \sphinxcode{\sphinxupquote{SampleFreq}} parameter in configuration file.

\end{itemize}
\end{sphinxadmonition}


\section{Free Energy Output File}
\label{\detokenize{output_file:free-energy-output-file}}
\sphinxAtStartPar
GOMC is capable of calculating absolute solvation free energy in NVT and NPT ensemble, using
thermodynamic integration and free energy purturbation methods.
GOMC outputs the raw informations, such as the derivative of energy with respective to current
lambda (\(\frac{dE}{d\lambda}\)) and energy different between current lambda state
and all other neighboring lambda states (\(\Delta E_{{\lambda}_i \rightarrow {\lambda}_j}\)),
which is essential to calculate solvation free energy with various estimators, such as TI, BAR, MBAR, and more.

\sphinxAtStartPar
The header of Free\_Energy\_BOX\_0\_ \sphinxcode{\sphinxupquote{OutputName}}.dat contains the following information:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Temperature of the simulation.

\item {} 
\sphinxAtStartPar
The index of the lambda vector.

\item {} 
\sphinxAtStartPar
Value of \({\lambda}_{Coulomb}\) and \({\lambda}_{VDW}\).

\item {} 
\sphinxAtStartPar
Monte Carlo step number.

\item {} 
\sphinxAtStartPar
Total energy of the system.

\item {} 
\sphinxAtStartPar
Derivative of energy with respective to lambda for coulomb interaction (\(\frac{dE}{d\lambda_{Coulomb}}\)).

\item {} 
\sphinxAtStartPar
Derivative of energy with respective to lambda for VDW interaction (\(\frac{dE}{d\lambda_{VDW}}\)).

\item {} 
\sphinxAtStartPar
Energy different between current lambda state and all other neighboring lambda states
(\(\Delta E_{{\lambda}_i \rightarrow {\lambda}_j} = E_{{\lambda}_j} - E_{{\lambda}_i}\))

\end{itemize}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=1.000\linewidth]{{FE-snapshot}.png}
\caption{Snapshot of GOMC free energy output file (Free\_Energy\_BOX\_0\_ \sphinxcode{\sphinxupquote{OutputName}}.dat).}\label{\detokenize{output_file:id17}}\end{figure}

\begin{sphinxadmonition}{important}{Important:}
\sphinxAtStartPar
For simulation in NPT ensemble or NVT ensemble with activated pressure calculation (\sphinxcode{\sphinxupquote{PressureCalc}}   True),
additional column will be printed to represent \(PV\) term.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The frequency of outputing free energy data is controlled by \sphinxcode{\sphinxupquote{FreeEnergyCalc}} parameter in configuration file.
\end{sphinxadmonition}


\chapter{Putting it all together: Running a GOMC Simulation}
\label{\detokenize{putting_all_together:putting-it-all-together-running-a-gomc-simulation}}\label{\detokenize{putting_all_together::doc}}
\sphinxAtStartPar
It is strongly recommended that you download the test system provided at \sphinxhref{http://gomc.eng.wayne.edu/downloads.html}{GOMC Website} or \sphinxhref{https://github.com/GOMC-WSU/GOMC\_Examples/tree/master}{Our Github Page}

\sphinxAtStartPar
Run different simulation types in order to become more familiar with different parameter and configuration files (*.conf).

\sphinxAtStartPar
To recap the previous examples, a simulation of isobutane will be completed for a single temperature point on the saturated vapor\sphinxhyphen{}liquid coexistence curve.

\sphinxAtStartPar
The general plan for running the simulation is:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Build GOMC (if not done already)

\item {} 
\sphinxAtStartPar
Copy GOMC executable to build directory

\item {} 
\sphinxAtStartPar
Create scripts, PDB, and topology file to build the system, plus in.dat file and parameter files to prepare for runtime

\item {} 
\sphinxAtStartPar
Build finished PDBs and PSFs using the simulation.

\item {} 
\sphinxAtStartPar
Run the simulation in the terminal.

\item {} 
\sphinxAtStartPar
Analyze the output.

\end{enumerate}

\sphinxAtStartPar
Please, complete steps 1 and 2; then, traverse to the directory, which should now contain a single file “GOMC\_CPU\_GEMC”. Next, six files need to be made:
\begin{itemize}
\item {} 
\sphinxAtStartPar
PDB file for isobutane

\item {} 
\sphinxAtStartPar
Topology file describing isobutane residue

\item {} 
\sphinxAtStartPar
Two \sphinxcode{\sphinxupquote{*.inp}} packmol scripts to pack two system boxes

\item {} 
\sphinxAtStartPar
Two \sphinxstyleemphasis{TCL} scripts to input into \sphinxcode{\sphinxupquote{PSFGen}} to generate the final configuration

\end{itemize}

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{isobutane.pdb}}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
REMARK   1 File  created   by  GaussView   5.0.8
ATOM          1       C1  ISB          1   0.911   \PYGZhy{}0.313    0.000  C
ATOM          2       C2  ISB          1   1.424   \PYGZhy{}1.765    0.000  C
ATOM          3       C3  ISB          1  \PYGZhy{}0.629   \PYGZhy{}0.313    0.000  C
ATOM          4       C4  ISB          1   1.424    0.413   \PYGZhy{}1.257  C
END
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{Top\_Branched\_Alkane.inp}}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
* Custom top file \PYGZhy{}\PYGZhy{} branched alkanes
*
MASS     1    CH3      15.035 C !
MASS     2    CH1      13.019 C !

AUTOGENERATE ANGLES DIHEDRALS

RESI   ISB   0.00               !  isobutane \PYGZob{} TraPPE \PYGZcb{}
GROUP
ATOM    C1    CH1       0.00    !  C3\PYGZbs{}
ATOM    C2    CH3       0.00    !     C2\PYGZhy{}C1
ATOM    C3    CH3       0.00    !  C4/
ATOM    C4    CH3       0.00    !
BOND    C1  C2   C1  C3   C1  C4
PATCHING FIRS NONE LAST NONE
END
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{pack\_box\_0.inp}}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
tolerance   3.0
filetype    pdb
output      STEP2\PYGZus{}ISB\PYGZus{}packed\PYGZus{}BOX\PYGZus{}0.pdb

structure     isobutane.pdb
number        1000
inside cube   0.  0.  0.  68.00
end structure
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{pack\_box\_1.inp}}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
tolerance   3.0
filetype    pdb
output      STEP2\PYGZus{}ISB\PYGZus{}packed\PYGZus{}BOX\PYGZus{}1.pdb

structure     isobutane.pdb
number        1000
inside cube   0.  0.  0.  68.00
end structure
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{build\_box\_0.inp}}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
package require psfgen

topology  ./Top Branched Alkane.inp segment ISB \PYGZob{}
  pdb     ./STEP2\PYGZus{}ISB\PYGZus{}packed\PYGZus{}BOX\PYGZus{}0.pdb
  first   none
  last    none
\PYGZcb{}
coordpdb  ./STEP2 ISB\PYGZus{}packed\PYGZus{}BOX\PYGZus{}0.pdb ISB

writepsf  ./STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}0.psf
writepdb  ./STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}0.pdb
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{build\_box\_1.inp}}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
package require psfgen

topology  ./Top Branched Alkane.inp segment ISB \PYGZob{}
  pdb     ./STEP2\PYGZus{}ISB\PYGZus{}packed\PYGZus{}BOX\PYGZus{}1.pdb
  first   none
  last    none
\PYGZcb{}
coordpdb  ./STEP2 ISB\PYGZus{}packed\PYGZus{}BOX\PYGZus{}1.pdb ISB

writepsf  ./STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}1.psf
writepdb  ./STEP3\PYGZus{}START\PYGZus{}ISB\PYGZus{}sys\PYGZus{}BOX\PYGZus{}1.pdb
\end{sphinxVerbatim}

\sphinxAtStartPar
These files can be created with a standard Linux or Windows text editor. Please, also copy a Packmol executable into the working directory.

\sphinxAtStartPar
Once those files are created, run in the terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./packmol   \PYGZlt{}   pack\PYGZus{}box\PYGZus{}0.inp
\PYGZdl{} ./packmol   \PYGZlt{}   pack\PYGZus{}box\PYGZus{}1.inp
\end{sphinxVerbatim}

\sphinxAtStartPar
This will create the intermediate PDBs.

\sphinxAtStartPar
Then, run the PSFGen scripts to finish the system using the following commands:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} vmd \PYGZhy{}dispdev text \PYGZlt{} ./build\PYGZus{}box\PYGZus{}0.inp
\PYGZdl{} vmd \PYGZhy{}dispdev text \PYGZlt{} ./build\PYGZus{}box\PYGZus{}1.inp
\end{sphinxVerbatim}

\sphinxAtStartPar
This will create the intermediate PDBs.

\sphinxAtStartPar
To run the code a few additional things will be needed:
\begin{itemize}
\item {} 
\sphinxAtStartPar
A GOMC Gibbs ensemble executable

\item {} 
\sphinxAtStartPar
A control file

\item {} 
\sphinxAtStartPar
Parameter files

\end{itemize}

\sphinxAtStartPar
Enter the control file (in.conf) in the text editor in order to modify it. Example files for different simulation types can be found in previous section.

\sphinxAtStartPar
Once these four files have been added to the output directory, the simulation is ready.

\sphinxAtStartPar
Assuming the code is named GOMC\_CPU\_GEMC, run in the terminal using:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./GOMC CPU GEMC in.conf \PYGZgt{} out\PYGZus{}ISB\PYGZus{}T\PYGZus{}330.00\PYGZus{}K\PYGZus{}RUN\PYGZus{}0.log \PYG{p}{\PYGZam{}}
\end{sphinxVerbatim}

\sphinxAtStartPar
For running GOMC in parallel, using openmp, run in the terminal using:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./GOMC CPU GEMC +p4 in.conf \PYGZgt{} out\PYGZus{}ISB\PYGZus{}T\PYGZus{}330.00\PYGZus{}K\PYGZus{}RUN\PYGZus{}0.log\PYG{p}{\PYGZam{}}
\end{sphinxVerbatim}

\sphinxAtStartPar
Here, 4 defines the number of processors that will be used to run the simulation in parallel.

\sphinxAtStartPar
Progress can be monitored in the terminal with the tail command:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} tail \PYGZhy{}f out\PYGZus{}ISB.log
\end{sphinxVerbatim}

\begin{sphinxadmonition}{attention}{Attention:}
\sphinxAtStartPar
Congratulations! You have examined a single\sphinxhyphen{}phase coexistence point on the saturated vapor\sphinxhyphen{}liquid curve using GOMC operating in the Gibbs ensemble.
\end{sphinxadmonition}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=1.000\linewidth]{{isobutane_result}.png}
\caption{Repeating this process for multiple temperatures will allow you to obtain the following results.}\label{\detokenize{putting_all_together:id1}}\end{figure}


\chapter{Intermolecular Energy and Virial Function (Van der Waals)}
\label{\detokenize{vdw_energy:intermolecular-energy-and-virial-function-van-der-waals}}\label{\detokenize{vdw_energy::doc}}
\sphinxAtStartPar
In this section, the virial and energy equation of Van der Waals interaction for different potential function are discussed in details.


\section{VDW}
\label{\detokenize{vdw_energy:vdw}}
\sphinxAtStartPar
This option calculates potential energy without any truncation.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Potential Calculation}}}] \leavevmode
\sphinxAtStartPar
Interactions between atoms can be modeled with an n\sphinxhyphen{}6 potential, a Mie potential in which the attractive exponent is fixed. The Mie potential can be viewed as a generalized version of the 12\sphinxhyphen{}6 Lennard\sphinxhyphen{}Jones potential,
\begin{equation*}
\begin{split}E_{\texttt{VDW}}(r_{ij}) = C_{n_{ij}} \epsilon_{ij} \bigg[\bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg]\end{split}
\end{equation*}
\sphinxAtStartPar
where \(r_{ij}\), \(\epsilon_{ij}\), and \(\sigma_{ij}\) are, respectively, the separation, minimum potential, and collision diameter for the pair of interaction sites \(i\) and \(j\). The constant \(C_n\) is a normalization factor such that the minimum of the potential remains at \(-\epsilon_{ij}\) for all \(n_{ij}\). In the 12\sphinxhyphen{}6 potential, \(C_n\) reduces to the familiar value of 4.
\begin{equation*}
\begin{split}C_{n_{ij}} = \bigg(\frac{n_{ij}}{n_{ij} - 6} \bigg)\bigg(\frac{n_{ij}}{6} \bigg)^{6/(n_{ij} - 6)}\end{split}
\end{equation*}
\item[{\sphinxcode{\sphinxupquote{Virial Calculation}}}] \leavevmode
\sphinxAtStartPar
Virial is basically the negative derivative of energy with respect to distance, multiplied by distance.
\begin{equation*}
\begin{split}W_{\texttt{VDW}}(r_{ij}) = -\frac{dE_{\texttt{VDW}}(r_{ij})}{r_{ij}}\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}} = F_{\texttt{VDW}}(r_{ij}) \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}}\end{split}
\end{equation*}
\sphinxAtStartPar
Using n\sphinxhyphen{}6 LJ potential defined above:
\begin{equation*}
\begin{split}F_{\texttt{VDW}}(r_{ij}) = 6C_{n_{ij}} \epsilon_{ij} \bigg[\frac{n_{ij}}{6} \times \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg]\times \frac{1}{{r_{ij}}}\end{split}
\end{equation*}
\end{description}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
This option only evaluates the energy up to specified \sphinxcode{\sphinxupquote{Rcut}} distance. Tail correction to energy and pressure can be specified to account for infinite cutoff distance.
\end{sphinxadmonition}


\section{EXP6}
\label{\detokenize{vdw_energy:exp6}}
\sphinxAtStartPar
This option calculates potential energy without any truncation.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Potential Calculation}}}] \leavevmode
\sphinxAtStartPar
Interactions between atoms can be modeled with an exp\sphinxhyphen{}6 (Buckingham) potential,
\begin{equation*}
\begin{split}E_{\texttt{VDW}}(r_{ij}) =
\begin{cases}
  \frac{\alpha_{ij}\epsilon_{ij}}{\alpha_{ij}-6} \bigg[\frac{6}{\alpha_{ij}} \exp\bigg(\alpha_{ij} \bigg[1-\frac{r_{ij}}{R_{min,ij}} \bigg]\bigg) - {\bigg(\frac{R_{min,ij}}{r_{ij}}\bigg)}^6 \bigg] &  r_{ij} \geq R_{max,ij} \\
  \infty & r_{ij} < R_{max,ij}
\end{cases}\end{split}
\end{equation*}
\sphinxAtStartPar
where \(r_{ij}\), \(\epsilon_{ij}\), and \(R_{min,ij}\) are, respectively, the separation, minimum potential, and minimum potential distance for the pair of interaction sites \(i\) and \(j\).
The constant \(\alpha_{ij}\) is an  exponential\sphinxhyphen{}6 parameter. The cutoff distance \(R_{max,ij}\) is the smallest positive value for which \(\frac{dE_{\texttt{VDW}}(r_{ij})}{dr_{ij}}=0\).

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
In order to use \sphinxcode{\sphinxupquote{Mie}} or \sphinxcode{\sphinxupquote{Exotice}} potential file format for \sphinxcode{\sphinxupquote{Buckingham}} potential, instead of defining \(R_{min}\), we define \(\sigma\) (collision diameter or the distance, where potential is zero)
and GOMC will calculate the \(R_{min}\) and \(R_{max}\) using \sphinxcode{\sphinxupquote{Buckingham}} potential equation.
\end{sphinxadmonition}

\item[{\sphinxcode{\sphinxupquote{Virial Calculation}}}] \leavevmode
\sphinxAtStartPar
Virial is basically the negative derivative of energy with respect to distance, multiplied by distance.
\begin{equation*}
\begin{split}W_{\texttt{VDW}}(r_{ij}) = -\frac{dE_{\texttt{VDW}}(r_{ij})}{r_{ij}}\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}} = F_{\texttt{VDW}}(r_{ij}) \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}}\end{split}
\end{equation*}
\sphinxAtStartPar
Using exp\sphinxhyphen{}6 potential defined above:
\begin{equation*}
\begin{split}F_{\texttt{VDW}}(r_{ij}) =
\begin{cases}
  \frac{6 \alpha_{ij}\epsilon_{ij}}{r_{ij}\big(\alpha_{ij}-6\big)} \bigg[\frac{r_{ij}}{R{min,ij}} \exp\bigg(\alpha_{ij} \bigg[1-\frac{r_{ij}}{R_{min,ij}} \bigg]\bigg) - {\bigg(\frac{R_{min,ij}}{r_{ij}}\bigg)}^6 \bigg] &  r_{ij} \geq R_{max,ij} \\
  \infty & r_{ij} < R_{max,ij}
\end{cases}\end{split}
\end{equation*}
\end{description}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
This option only evaluates the energy up to specified \sphinxcode{\sphinxupquote{Rcut}} distance. Tail correction to energy and pressure can be specified to account for infinite cutoff distance.
\end{sphinxadmonition}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=1.000\linewidth]{{VDW_Exp6}.png}
\caption{Graph of Van der Waals interaction for comparison of \sphinxcode{\sphinxupquote{VDW}} and \sphinxcode{\sphinxupquote{EXP6}} potentials.}\label{\detokenize{vdw_energy:id1}}\end{figure}


\section{SHIFT}
\label{\detokenize{vdw_energy:shift}}
\sphinxAtStartPar
This option forces the potential energy to be zero at \sphinxcode{\sphinxupquote{Rcut}} distance.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Potential Calculation}}}] \leavevmode
\sphinxAtStartPar
Interactions between atoms can be modeled with an n\sphinxhyphen{}6 potential,
\begin{equation*}
\begin{split}E_{\texttt{VDW}}(r_{ij}) = C_{n_{ij}} \epsilon_{ij} \bigg[\bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg] - C_{n_{ij}} \epsilon_{ij} \bigg[\bigg(\frac{\sigma_{ij}}{r_{cut}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{cut}}\bigg)^6\bigg]\end{split}
\end{equation*}
\sphinxAtStartPar
where \(r_{ij}\), \(\epsilon_{ij}\), and \(\sigma_{ij}\) are, respectively, the separation, minimum potential, and collision diameter for the pair of interaction sites \(i\) and \(j\). The constant \(C_n\) is a normalization factor according to Eq. 3, such that the minimum of the potential remains at \(-\epsilon_{ij}\) for all \(n_{ij}\). In the 12\sphinxhyphen{}6 potential, \(C_n\) reduces to the familiar value of 4.

\item[{\sphinxcode{\sphinxupquote{Virial Calculation}}}] \leavevmode
\sphinxAtStartPar
Virial is basically the negative derivative of energy with respect to distance, multiplied by distance.
\begin{equation*}
\begin{split}W_{\texttt{VDW}}(r_{ij}) = -\frac{dE_{\texttt{VDW}}(r_{ij})}{r_{ij}}\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}} = F_{\texttt{VDW}}(r_{ij}) \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}}\end{split}
\end{equation*}
\sphinxAtStartPar
Using \sphinxcode{\sphinxupquote{SHIFT}} potential function defined above:
\begin{equation*}
\begin{split}F_{\texttt{VDW}}(r_{ij}) = 6C_{n_{ij}} \epsilon_{ij} \bigg[\frac{n_{ij}}{6} \times \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg]\times \frac{1}{{r_{ij}}}\end{split}
\end{equation*}
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{VDW_SHIFT}.png}
\caption{Graph of Van der Waals potential with and without the application of the \sphinxcode{\sphinxupquote{SHIFT}} function. With the \sphinxcode{\sphinxupquote{SHIFT}} function active, the potential by force was reduced to 0.0 at the \sphinxcode{\sphinxupquote{Rcut}} distance. With the \sphinxcode{\sphinxupquote{SHIFT}} function, there is a discontinuity where the potential is truncated.}\label{\detokenize{vdw_energy:id2}}\end{figure}

\end{description}


\section{SWITCH}
\label{\detokenize{vdw_energy:switch}}
\sphinxAtStartPar
This option in \sphinxcode{\sphinxupquote{CHARMM}} or \sphinxcode{\sphinxupquote{EXOTIC}} force field smoothly forces the potential energy to be zero at Rcut distance and starts modifying the potential at Rswitch distance.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Potential Calculation}}}] \leavevmode
\sphinxAtStartPar
Interactions between atoms can be modeled with an n\sphinxhyphen{}6 potential,
\begin{equation*}
\begin{split}E_{\texttt{VDW}}(r_{ij}) = C_{n_{ij}} \epsilon_{ij} \bigg[\bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg]\times \varphi_E(r_{ij})\end{split}
\end{equation*}
\sphinxAtStartPar
where \(r_{ij}\), \(\epsilon_{ij}\), and \(\sigma_{ij}\) are, respectively, the separation, minimum potential, and collision diameter for the pair of interaction sites \(i\) and \(j\). The constant \(C_n\) is a normalization factor according to Eq. 3, such that the minimum of the potential remains at \(-\epsilon_{ij}\) for all \(n_{ij}\). In the 12\sphinxhyphen{}6 potential, \(C_n\) reduces to the familiar value of 4.

\sphinxAtStartPar
The factor \(\varphi_E\) is defined as:
\begin{equation*}
\begin{split}\varphi_E(r_{ij}) =
\begin{cases}
  1 & r_{ij} \leq r_{switch} \\
  \frac{\big({r_{cut}}^2 - {r_{ij}}^2 \big)^2 \times \big({r_{cut}}^2 - 3{r_{switch}}^2 + 2{r_{ij}}^2 \big)}{\big({r_{cut}}^2 - {r_{switch}}^2 \big)^3} & r_{switch} < r_{ij} < r_{cut} \\
  0 & r_{ij} \geq r_{cut}
\end{cases}\end{split}
\end{equation*}
\item[{\sphinxcode{\sphinxupquote{Virial Calculation}}}] \leavevmode
\sphinxAtStartPar
Virial is basically the negative derivative of energy with respect to distance, multiplied by distance.
\begin{equation*}
\begin{split}W_{\texttt{VDW}}(r_{ij}) = -\frac{dE_{\texttt{VDW}}(r_{ij})}{r_{ij}}\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}} = F_{\texttt{VDW}}(r_{ij}) \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}}\end{split}
\end{equation*}
\sphinxAtStartPar
Using SWITCH potential function defined above:
\begin{align*}\!\begin{aligned}
F_{\texttt{VDW}}(r_{ij}) = \Bigg[6 C_{n_{ij}} \epsilon_{ij} \bigg[\frac{n_{ij}}{6} \times \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg]\times \frac{\varphi_E(r_{ij})}{{r_{ij}}}  -\\
C_{n_{ij}} \epsilon_{ij} \bigg[\bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg] \times \varphi_F(r_{ij}) \Bigg]\\
\end{aligned}\end{align*}
\sphinxAtStartPar
The factor \(\varphi_F\) is defined as:
\begin{equation*}
\begin{split}\varphi_F(r_{ij}) =
\begin{cases}
  0 & r_{ij} \leq r_{switch} \\
  \frac{12r_{ij}\big({r_{cut}}^2 - {r_{ij}}^2 \big) \times \big({r_{switch}}^2 - {r_{ij}}^2 \big)}{\big({r_{cut}}^2 - {r_{switch}}^2 \big)^3} & r_{switch} < r_{ij} < r_{cut} \\
  0 & r_{ij} \geq r_{cut}
\end{cases}\end{split}
\end{equation*}
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{SWITCH}.png}
\caption{Graph of Van der Waals potential with and without the application of the \sphinxcode{\sphinxupquote{SWITCH}} function. With the \sphinxcode{\sphinxupquote{SWITCH}} function active, the potential is smoothly reduced to 0.0 at the \sphinxcode{\sphinxupquote{Rcut}} distance.}\label{\detokenize{vdw_energy:id3}}\end{figure}

\end{description}


\section{SWITCH (MARTINI)}
\label{\detokenize{vdw_energy:switch-martini}}
\sphinxAtStartPar
This option in \sphinxcode{\sphinxupquote{MARTINI}} force field smoothly forces the potential energy to be zero at Rcut distance and starts modifying the potential at \sphinxcode{\sphinxupquote{Rswitch}} distance.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Potential Calculation}}}] \leavevmode
\sphinxAtStartPar
Potential Calculation: Interactions between atoms can be modeled with an n\sphinxhyphen{}6 potential. In standard MARTINI, \(n\) is equal to 12,
\begin{equation*}
\begin{split}E_{\texttt{VDW}}(r_{ij}) = C_{n_{ij}}\epsilon_{ij} \Bigg[ {\sigma_{ij}}^{n} \bigg(\frac{1}{{r_{ij}}^{n}} + \varphi_{E, n} (r_{ij}) \bigg) - {\sigma_{ij}}^{6} \bigg(\frac{1}{{r_{ij}}^{6}} + \varphi_{E, 6} (r_{ij}) \bigg) \Bigg]\end{split}
\end{equation*}
\sphinxAtStartPar
where \(r_{ij}\), \(\epsilon_{ij}\), and \(\sigma_{ij}\) are, respectively, the separation, minimum potential, and collision diameter for the pair of interaction sites \(i\) and \(j\). The constant \(C_n\) is a normalization factor according to Eq. 3, such that the minimum of the potential remains at \(-\epsilon_{ij}\) for all \(n_{ij}\). In the 12\sphinxhyphen{}6 potential, \(C_n\) reduces to the familiar value of 4.

\sphinxAtStartPar
The factor \(\varphi_{E, \alpha}\) and constants are defined as:
\begin{equation*}
\begin{split}\varphi_{E, \alpha}(r_{ij}) =
\begin{cases}
  -C_{\alpha} & r_{ij} \leq r_{switch} \\
  -\frac{A_{\alpha}}{3} (r_{ij} - r_{switch})^3 -\frac{B_{\alpha}}{4} (r_{ij} - r_{switch})^4 - C_{\alpha} & r_{switch} < r_{ij} < r_{cut} \\
  0 & r_{ij} \geq r_{cut}
\end{cases}\end{split}
\end{equation*}\begin{equation*}
\begin{split}A_{\alpha} = \alpha \frac{(\alpha + 1) r_{switch} - (\alpha +4) r_{cut}} {{r_{cut}}^{(\alpha + 2)} {(r_{cut} - r_{switch})}^2}\end{split}
\end{equation*}\begin{equation*}
\begin{split}B_{\alpha} = \alpha \frac{(\alpha + 1) r_{switch} - (\alpha +3) r_{cut}} {{r_{cut}}^{(\alpha + 2)} {(r_{cut} - r_{switch})}^3}\end{split}
\end{equation*}\begin{equation*}
\begin{split}C_{\alpha} =  \frac{1}{{r_{cut}}^{\alpha}} -\frac{A_{\alpha}}{3} (r_{cut} - r_{switch})^3 -\frac{B_{\alpha}}{4} (r_{cut} - r_{switch})^4\end{split}
\end{equation*}
\item[{\sphinxcode{\sphinxupquote{Virial Calculation}}}] \leavevmode
\sphinxAtStartPar
Virial is basically the negative derivative of energy with respect to distance, multiplied by distance.
\begin{equation*}
\begin{split}W_{\texttt{VDW}}(r_{ij}) = -\frac{dE_{\texttt{VDW}}(r_{ij})}{r_{ij}}\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}} = F_{\texttt{VDW}}(r_{ij}) \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}}\end{split}
\end{equation*}
\sphinxAtStartPar
Using the \sphinxcode{\sphinxupquote{SWITCH}} potential function defined for \sphinxcode{\sphinxupquote{MARTINI}} force field:
\begin{equation*}
\begin{split}F_{\texttt{VDW}}(r_{ij}) = C_{n_{ij}}\epsilon_{ij} \Bigg[ {\sigma_{ij}}^{n} \bigg(\frac{n}{{r_{ij}}^{(n+1)}} + \varphi_{F, n} (r_{ij}) \bigg) - {\sigma_{ij}}^{6} \bigg(\frac{6}{{r_{ij}}^{(6+1)}} + \varphi_{F, 6} (r_{ij}) \bigg) \Bigg]\end{split}
\end{equation*}
\sphinxAtStartPar
The constants defined in Eq. 14\sphinxhyphen{}16 and the factor \(\varphi_{F, \alpha}\) defined as:
\begin{equation*}
\begin{split}\varphi_{F, \alpha}(r_{ij}) =
\begin{cases}
  0 & r_{ij} \leq r_{switch} \\
  A_{\alpha} (r_{ij} - r_{switch})^2 + B_{\alpha} (r_{ij} - r_{switch})^3 & r_{switch} < r_{ij} < r_{cut} \\
  0 & r_{ij} \geq r_{cut}
\end{cases}\end{split}
\end{equation*}
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{MARTINI}.png}
\caption{Graph of Van der Waals potential with and without the application of the \sphinxcode{\sphinxupquote{SWITCH}} function in \sphinxcode{\sphinxupquote{MARTINI}} force field. With the \sphinxcode{\sphinxupquote{SWITCH}} function active, the potential is smoothly reduced to 0.0 at the \sphinxcode{\sphinxupquote{Rcut}} distance.}\label{\detokenize{vdw_energy:id4}}\end{figure}

\end{description}


\chapter{Intermolecular Energy and Virial Function (Electrostatic)}
\label{\detokenize{electrostatic:intermolecular-energy-and-virial-function-electrostatic}}\label{\detokenize{electrostatic::doc}}
\sphinxAtStartPar
In this section, the virial and energy equation of electrostatic interaction for different potential function are discussed in details.


\section{Ewald}
\label{\detokenize{electrostatic:ewald}}
\sphinxAtStartPar
This option calculate electrostatic energy using standard \sphinxstyleemphasis{Ewald Summation Method}.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Once this option is activated, it would override the the electrostatic calculation using \sphinxcode{\sphinxupquote{VDW}}, \sphinxcode{\sphinxupquote{EXP6}}, \sphinxcode{\sphinxupquote{SHIFT}}, and \sphinxcode{\sphinxupquote{SWITCH}} functions.
\end{sphinxadmonition}
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Potential Calculation}}}] \leavevmode
\sphinxAtStartPar
Coulomb interactions between atoms can be modeled as
\begin{equation*}
\begin{split}E(\texttt{Ewald}) = E_{real} + E_{reciprocal} + E_{self} + E_{correction}\end{split}
\end{equation*}
\sphinxAtStartPar
\(E_{real}\): Defines the short range electrostatic energy according to
\begin{equation*}
\begin{split}E_{real} = \frac{1}{4\pi \epsilon_0} \frac{1}{2} \sum_{i =1}^{N} \sum_{j = 1}^{N} q_i q_j  \frac{erfc(\alpha r_{ij})}{r_{ij}}\end{split}
\end{equation*}
\sphinxAtStartPar
, where \(\alpha\) is \sphinxcode{\sphinxupquote{Ewald}} separation parameter according to
\begin{equation*}
\begin{split}\alpha = \frac {\sqrt{-\log (Tolerance)}}{r_{cut}}\end{split}
\end{equation*}
\sphinxAtStartPar
, where \sphinxcode{\sphinxupquote{Tolerance}} is a parameter, controlling the desired accuracy.

\sphinxAtStartPar
\(E_{reciprocal}\): Defines the long range electrostatic energy according to,
\begin{equation*}
\begin{split}E_{reciprocal} = \frac{1}{\epsilon_0 V} \frac {1}{2} \sum_{\overrightarrow{k} \ne 0}^{} \frac {1}{\overrightarrow{k}^2}\exp\bigg(\frac {-\overrightarrow{k}^2}{4 \alpha^2}\bigg) \Bigg[ {\Big| R_{sum} \Big|}^2 + {\Big| I_{sum} \Big|}^2 \bigg]\end{split}
\end{equation*}
\sphinxAtStartPar
, where \(\overrightarrow{k}\) is reciprocal vector, \(R_{sum}\) and \(I_{sum}\) are,
\begin{equation*}
\begin{split}R_{sum} = \sum_{i=1}^{N} q_i \cos \big(\overrightarrow{k}.\overrightarrow{x_i}\big)\end{split}
\end{equation*}\begin{equation*}
\begin{split}I_{sum} = \sum_{i=1}^{N} q_i \sin \big(\overrightarrow{k}.\overrightarrow{x_i}\big)\end{split}
\end{equation*}
\sphinxAtStartPar
\(E_{self}\): Defines the self energy according to,
\begin{equation*}
\begin{split}E_{self} = -\frac{\alpha}{4\pi \epsilon_0 \sqrt{\pi}} \sum_{i=1}^{N} {q_i}^2\end{split}
\end{equation*}
\sphinxAtStartPar
\(E_{correction}\): Defines intra\sphinxhyphen{}molecule nonbonded energy,
\begin{equation*}
\begin{split}E_{correction} = -\frac{1}{4\pi \epsilon_0} \frac{1}{2} \sum_{j=1}^{N }\sum_{l =1}^{N_j} \sum_{m = 1}^{N_j} q_{j_l} q_{j_m}  \frac{erf(\alpha r_{j_l j_m})}{r_{j_l j_m}}\end{split}
\end{equation*}
\item[{\sphinxcode{\sphinxupquote{Virial Calculation}}}] \leavevmode
\sphinxAtStartPar
Virial is basically the negative derivative of energy with respect to distance, multiplied by distance, Eq. 4. Coulomb force between atoms can be modeled as,
\begin{equation*}
\begin{split}W_{Ewald} = W_{real} + W_{reciprocal}\end{split}
\end{equation*}
\sphinxAtStartPar
\(W_{real}\) defines the short range electrostatic and \(W_{reciprocal}\) defines the long range electrostatic force according to,
\begin{equation*}
\begin{split}W_{real} = \frac{1}{4\pi \epsilon_0} \frac{1}{2} \sum_{i =1}^{N} \sum_{j = 1}^{N} q_i q_j  \bigg[ \frac{erfc(\alpha r_{ij})}{r_{ij}} + \frac{2\alpha}{ \sqrt{\pi}} \exp(-\alpha^2 {r_{ij}}^2) \bigg] \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}^2}\end{split}
\end{equation*}\begin{equation*}
\begin{split}\begin{split}
              W_{reciprocal} = \frac{1}{\epsilon_0 V} \frac {1}{2} \sum_{\overrightarrow{k} \ne 0}^{} \Bigg[\frac {1}{\overrightarrow{k}^2}\exp\bigg(\frac {-\overrightarrow{k}^2}{4 \alpha^2}\bigg) \bigg( {\Big| R_{sum} \Big|}^2 + {\Big| I_{sum} \Big|}^2 \bigg) \bigg(  1 - \frac{\overrightarrow{k}^2}{2\alpha^2} \bigg) \Bigg] +\\
  \sum_{i=1}^{N} \frac{1}{\epsilon_0 V}  \sum_{\overrightarrow{k} \ne 0}^{} \Bigg[ \frac {q_i}{\overrightarrow{k}^2}\exp\bigg(\frac {-\overrightarrow{k}^2}{4 \alpha^2}\bigg) \bigg[ I_{sum} \times\cos(\overrightarrow{k}.\overrightarrow{x_i})  - R_{sum} \times \sin(\overrightarrow{k}.\overrightarrow{x_i}) \bigg] \Bigg] \times \big( \overrightarrow{k}.\overrightarrow{r_{ic}} \big)
      \end{split}\end{split}
\end{equation*}
\sphinxAtStartPar
, where \(\overrightarrow{r_{ic}}\) is the vector between atom and the center of the mass of the molecule.

\end{description}


\section{VDW}
\label{\detokenize{electrostatic:vdw}}
\sphinxAtStartPar
Using \sphinxcode{\sphinxupquote{VDW}} potential type without \sphinxcode{\sphinxupquote{Ewald}} method, simply uses coulomb energy to calculate the electrostatic potential.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Potential Calculation}}}] \leavevmode
\sphinxAtStartPar
Coulomb interactions between atoms can be modeled as
\begin{equation*}
\begin{split}E_{\texttt{Elect}}(r_{ij}) = \frac{q_i q_j}{4\pi \epsilon_0 r_{ij}}\end{split}
\end{equation*}
\item[{\sphinxcode{\sphinxupquote{Virial Calculation}}}] \leavevmode
\sphinxAtStartPar
Virial is basically the negative derivative of energy with respect to distance, multiplied by distance.
\begin{equation*}
\begin{split}W_{\texttt{Elect}}(r_{ij}) = -\frac{dE_{\texttt{Elect}}(r_{ij})}{r_{ij}}\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}} = F_{\texttt{Elect}}(r_{ij}) \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}}\end{split}
\end{equation*}\begin{equation*}
\begin{split}F_{\texttt{Elect}}(r_{ij}) = \frac{q_i q_j}{4\pi \epsilon_0} \Big( \frac{1}{{r_{ij}}^2} \Big)\end{split}
\end{equation*}
\end{description}


\section{EXP6}
\label{\detokenize{electrostatic:exp6}}
\sphinxAtStartPar
Using \sphinxcode{\sphinxupquote{EXP6}} potential type without \sphinxcode{\sphinxupquote{Ewald}} method, simply uses coulomb energy to calculate the electrostatic potential.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Potential Calculation}}}] \leavevmode
\sphinxAtStartPar
Coulomb interactions between atoms can be modeled as
\begin{equation*}
\begin{split}E_{\texttt{Elect}}(r_{ij}) = \frac{q_i q_j}{4\pi \epsilon_0 r_{ij}}\end{split}
\end{equation*}
\item[{\sphinxcode{\sphinxupquote{Virial Calculation}}}] \leavevmode
\sphinxAtStartPar
Virial is basically the negative derivative of energy with respect to distance, multiplied by distance.
\begin{equation*}
\begin{split}W_{\texttt{Elect}}(r_{ij}) = -\frac{dE_{\texttt{Elect}}(r_{ij})}{r_{ij}}\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}} = F_{\texttt{Elect}}(r_{ij}) \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}}\end{split}
\end{equation*}\begin{equation*}
\begin{split}F_{\texttt{Elect}}(r_{ij}) = \frac{q_i q_j}{4\pi \epsilon_0} \Big( \frac{1}{{r_{ij}}^2} \Big)\end{split}
\end{equation*}
\end{description}


\section{SHIFT}
\label{\detokenize{electrostatic:shift}}
\sphinxAtStartPar
This option forces the electrostatic energy to be zero at \sphinxcode{\sphinxupquote{Rcut}} distance.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Potential Calculation}}}] \leavevmode
\sphinxAtStartPar
Coulomb interactions between atoms can be modeled as
\begin{equation*}
\begin{split}E_{\texttt{Elect}}(r_{ij}) = \frac{q_i q_j}{4\pi \epsilon_0} \Big( \frac{1}{r_{ij}} - \frac{1}{r_{cut}} \Big)\end{split}
\end{equation*}
\item[{\sphinxcode{\sphinxupquote{Virial Calculation}}}] \leavevmode
\sphinxAtStartPar
Virial is basically the negative derivative of energy with respect to distance, multiplied by distance.
\begin{equation*}
\begin{split}W_{\texttt{Elect}}(r_{ij}) = -\frac{dE_{\texttt{Elect}}(r_{ij})}{r_{ij}}\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}} = F_{\texttt{Elect}}(r_{ij}) \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}}\end{split}
\end{equation*}\begin{equation*}
\begin{split}F_{\texttt{Elect}}(r_{ij}) = \frac{q_i q_j}{4\pi \epsilon_0} \Big( \frac{1}{{r_{ij}}^2} \Big)\end{split}
\end{equation*}
\end{description}


\section{SWITCH}
\label{\detokenize{electrostatic:switch}}
\sphinxAtStartPar
This option in \sphinxcode{\sphinxupquote{CHARMM}} or \sphinxcode{\sphinxupquote{EXOTIC}} force field forces the electrostatic energy to be zero at \sphinxcode{\sphinxupquote{Rcut}} distance.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Potential Calculation}}}] \leavevmode
\sphinxAtStartPar
Coulomb interactions between atoms can be modeled as,
\begin{equation*}
\begin{split}E_{\texttt{Elect}}(r_{ij}) = \frac{q_i q_j}{4\pi \epsilon_0} \bigg( \Big(\frac{r_{ij}}{r_{cut}} \Big)^2 - 1.0\bigg)^2 \frac{1}{r_{ij}}\end{split}
\end{equation*}
\item[{\sphinxcode{\sphinxupquote{Virial Calculation}}}] \leavevmode
\sphinxAtStartPar
Virial is basically the negative derivative of energy with respect to distance, multiplied by distance.
\begin{equation*}
\begin{split}W_{\texttt{Elect}}(r_{ij}) = -\frac{dE_{\texttt{Elect}}(r_{ij})}{r_{ij}}\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}} = F_{\texttt{Elect}}(r_{ij}) \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}}\end{split}
\end{equation*}\begin{equation*}
\begin{split}F_{\texttt{Elect}}(r_{ij}) = \frac{q_i q_j}{4\pi \epsilon_0} \Bigg[ \bigg( \Big(\frac{r_{ij}}{r_{cut}} \Big)^2 - 1.0\bigg)^2 \frac{1}{{r_{ij}}^2} - \bigg( \frac{4}{{r_{cut}}^2} \bigg) \bigg( \Big(\frac{r_{ij}}{r_{cut}} \Big)^2 - 1.0\bigg) \Bigg]\end{split}
\end{equation*}
\end{description}


\section{SWITCH (MARTINI)}
\label{\detokenize{electrostatic:switch-martini}}
\sphinxAtStartPar
This option in \sphinxcode{\sphinxupquote{MARTINI}} force field smoothly forces the potential energy to be zero at \sphinxcode{\sphinxupquote{Rcut}} distance and starts modifying the potential at \sphinxcode{\sphinxupquote{Rswitch = 0.0}} distance.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Potential Calculation}}}] \leavevmode
\sphinxAtStartPar
Coulomb interactions between atoms can be modeled as,
\begin{equation*}
\begin{split}E_{\texttt{Elect}}(r_{ij})=\frac{q_i q_j}{4\pi\epsilon_0\epsilon_1}\bigg(\frac{1}{r_{ij}}+\varphi_{E, 1}(r_{ij})\bigg)\end{split}
\end{equation*}
\sphinxAtStartPar
, where \(\epsilon_1\) is the dielectric constant, which in \sphinxcode{\sphinxupquote{MARTINI}} force field is equal to 15.0 and \(\varphi_{E, \alpha = 1}(r_{ij})\) is defined as:
\begin{equation*}
\begin{split}\varphi_{E, \alpha}(r_{ij}) =
\begin{cases}
  -C_{\alpha} & r_{ij} \leq r_{switch} \\
  -\frac{A_{\alpha}}{3} (r_{ij} - r_{switch})^3 -\frac{B_{\alpha}}{4} (r_{ij} - r_{switch})^4 - C_{\alpha} & r_{switch} < r_{ij} < r_{cut} \\
  0 & r_{ij} \geq r_{cut}
\end{cases}\end{split}
\end{equation*}\begin{equation*}
\begin{split}A_{\alpha} = \alpha \frac{(\alpha + 1) r_{switch} - (\alpha +4) r_{cut}} {{r_{cut}}^{(\alpha + 2)} {(r_{cut} - r_{switch})}^2}\end{split}
\end{equation*}\begin{equation*}
\begin{split}B_{\alpha} = \alpha \frac{(\alpha + 1) r_{switch} - (\alpha +3) r_{cut}} {{r_{cut}}^{(\alpha + 2)} {(r_{cut} - r_{switch})}^3}\end{split}
\end{equation*}\begin{equation*}
\begin{split}C_{\alpha} =  \frac{1}{{r_{cut}}^{\alpha}} -\frac{A_{\alpha}}{3} (r_{cut} - r_{switch})^3 -\frac{B_{\alpha}}{4} (r_{cut} - r_{switch})^4\end{split}
\end{equation*}
\item[{\sphinxcode{\sphinxupquote{Virial Calculation}}}] \leavevmode
\sphinxAtStartPar
Virial is basically the negative derivative of energy with respect to distance, multiplied by distance.
\begin{equation*}
\begin{split}W_{\texttt{Elect}}(r_{ij}) = -\frac{dE_{\texttt{Elect}}(r_{ij})}{r_{ij}}\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}} = F_{\texttt{Elect}}(r_{ij}) \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}}\end{split}
\end{equation*}\begin{equation*}
\begin{split}F_{\texttt{Elect}}(r_{ij})=\frac{q_iq_j}{4\pi\epsilon_0\epsilon_1}\bigg(\frac{1}{{r_{ij}}^2}+\varphi_{F, 1}(r_{ij})\bigg)\end{split}
\end{equation*}
\sphinxAtStartPar
, where \(\varphi_{F, \alpha = 1} (r_{ij})\) is defined as:
\begin{equation*}
\begin{split}\varphi_{F, \alpha}(r_{ij}) =
\begin{cases}
  0 & r_{ij} \leq r_{switch} \\
  A_{\alpha} (r_{ij} - r_{switch})^2 + B_{\alpha} (r_{ij} - r_{switch})^3 & r_{switch} < r_{ij} < r_{cut} \\
  0 & r_{ij} \geq r_{cut}
\end{cases}\end{split}
\end{equation*}
\end{description}


\chapter{Long\sphinxhyphen{}range Correction (Energy and Virial)}
\label{\detokenize{long_range_correction:long-range-correction-energy-and-virial}}\label{\detokenize{long_range_correction::doc}}
\sphinxAtStartPar
To accelerate the simulation performance, the nonbonded potential is usually truncated at specific cut\sphinxhyphen{}off (\sphinxcode{\sphinxupquote{Rcut}}) distance.
To compensate the missing potential energy and force, beyond the \sphinxcode{\sphinxupquote{Rcut}} distance, the long\sphinxhyphen{}range correction (\sphinxcode{\sphinxupquote{LRC}}) or tail correction to energy and virial must be
calculated and added to total energy and virial of the system, to account for infinite cutoff distance.

\sphinxAtStartPar
The \sphinxcode{\sphinxupquote{VDW}} and \sphinxcode{\sphinxupquote{EXP6}} energy functions, evaluates the energy up to specified \sphinxcode{\sphinxupquote{Rcut}} distance. In this section, the \sphinxcode{\sphinxupquote{LRC}} equations for virial and energy term
for Van der Waals interaction are discussed in details.


\section{VDW}
\label{\detokenize{long_range_correction:vdw}}
\sphinxAtStartPar
This option calculates potential energy using standard Lennard Jones (12\sphinxhyphen{}6) or Mie (n\sphinxhyphen{}6) potentials, up to specific \sphinxcode{\sphinxupquote{Rcut}} distance.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Energy}}}] \leavevmode
\sphinxAtStartPar
For homogeneous system, the long\sphinxhyphen{}range correction energy can be analytically calculated:
\begin{equation*}
\begin{split}E_{\texttt{LRC(VDW)}} = \frac{2\pi N^2}{V} \int_{r=r_{cut}}^{\infty} r^2 E_{\texttt{VDW}}(r) dr\end{split}
\end{equation*}\begin{equation*}
\begin{split}E_{\texttt{VDW}}(r) = C_{n} \epsilon \bigg[\bigg(\frac{\sigma}{r}\bigg)^{n} - \bigg(\frac{\sigma}{r}\bigg)^6\bigg]\end{split}
\end{equation*}
\sphinxAtStartPar
where \(N\), \(V\), \(r\), \(\epsilon\), and \(\sigma\) are the number of molecule, volume of the system, separation, minimum potential, and collision diameter, respectively.
The constant \(C_n\) is a normalization factor such that the minimum of the potential remains at \(-\epsilon\) for all \(n\). In the 12\sphinxhyphen{}6 potential, \(C_n\) reduces to the familiar value of 4.
\begin{equation*}
\begin{split}C_{n} = \bigg(\frac{n}{n - 6} \bigg)\bigg(\frac{n}{6} \bigg)^{6/(n - 6)}\end{split}
\end{equation*}
\sphinxAtStartPar
Substituting the general Lennard Jones energy equation into the integral, the long\sphinxhyphen{}range correction energy term is defined by:
\begin{equation*}
\begin{split}E_{\texttt{LRC(VDW)}} = \frac{2\pi N^2}{V} C_{n} \epsilon {\sigma}^3 \bigg[\frac{1}{n-3}\bigg(\frac{\sigma}{r_{cut}}\bigg)^{(n-3)} - \frac{1}{3} \bigg(\frac{\sigma}{r_{cut}}\bigg)^3\bigg]\end{split}
\end{equation*}
\item[{\sphinxcode{\sphinxupquote{Virial}}}] \leavevmode
\sphinxAtStartPar
For homogeneous system, the long\sphinxhyphen{}range correction virial can be analytically calculated:
\begin{equation*}
\begin{split}W_{\texttt{LRC(VDW)}} = \frac{2\pi N^2}{V} \int_{r=r_{cut}}^{\infty} r^3 F_{\texttt{VDW}}(r) dr\end{split}
\end{equation*}\begin{equation*}
\begin{split}F_{\texttt{VDW}}(r) = \frac{6C_{n} \epsilon}{r} \bigg[\frac{n}{6} \times \bigg(\frac{\sigma}{r}\bigg)^{n} - \bigg(\frac{\sigma}{r}\bigg)^6\bigg]\end{split}
\end{equation*}
\sphinxAtStartPar
Substituting the general Lennard Jones force equation into the integral, the long\sphinxhyphen{}range correction virial term is defined by:
\begin{equation*}
\begin{split}W_{\texttt{LRC(VDW)}} = \frac{2\pi N^2}{V} C_{n} \epsilon {\sigma}^3 \bigg[\frac{n}{n-3}\bigg(\frac{\sigma}{r_{cut}}\bigg)^{(n-3)} - 2 \bigg(\frac{\sigma}{r_{cut}}\bigg)^3\bigg]\end{split}
\end{equation*}
\end{description}


\section{EXP6}
\label{\detokenize{long_range_correction:exp6}}
\sphinxAtStartPar
This option calculates potential energy using Buckingham potentials, up to specific \sphinxcode{\sphinxupquote{Rcut}} distance.
\begin{description}
\item[{\sphinxcode{\sphinxupquote{Energy}}}] \leavevmode
\sphinxAtStartPar
For homogeneous system, the long\sphinxhyphen{}range correction energy can be analytically calculated:
\begin{equation*}
\begin{split}E_{\texttt{LRC(VDW)}} = \frac{2\pi N^2}{V} \int_{r=r_{cut}}^{\infty} r^2 E_{\texttt{VDW}}(r) dr\end{split}
\end{equation*}\begin{equation*}
\begin{split}E_{\texttt{VDW}}(r) =
\begin{cases}
  \frac{\alpha\epsilon}{\alpha-6} \bigg[\frac{6}{\alpha} \exp\bigg(\alpha \bigg[1-\frac{r}{R_{min}} \bigg]\bigg) - {\bigg(\frac{R_{min}}{r}\bigg)}^6 \bigg] &  r \geq R_{max} \\
  \infty & r < R_{max}
\end{cases}\end{split}
\end{equation*}
\sphinxAtStartPar
where \(r\), \(\epsilon\), and \(R_{min}\) are, respectively, the separation, minimum potential, and minimum potential distance.
The constant \(\alpha\) is an  exponential\sphinxhyphen{}6 parameter. The cutoff distance \(R_{max}\) is the smallest positive value for which \(\frac{dE_{\texttt{VDW}}(r)}{dr}=0\).

\sphinxAtStartPar
Substituting the Buckingham potential into the integral, the long\sphinxhyphen{}range correction energy term is defined by:
\begin{equation*}
\begin{split}E_{\texttt{LRC(VDW)}} = \frac{2\pi N^2}{V} \bigg[AB \exp\big(\frac{-r_{cut}}{B}\big) \bigg(2 B^2 + 2 B r_{cut} + {r_{cut}}^2 \bigg) - \frac{C}{3 {r_{cut}}^3}   \bigg]\end{split}
\end{equation*}\begin{equation*}
\begin{split}A = \frac{6 \epsilon \exp(\alpha)}{\alpha - 6}\end{split}
\end{equation*}\begin{equation*}
\begin{split}B = \frac{R_{min}}{\alpha}\end{split}
\end{equation*}\begin{equation*}
\begin{split}C = \frac{\epsilon \alpha {R_{min}}^6}{\alpha - 6}\end{split}
\end{equation*}
\item[{\sphinxcode{\sphinxupquote{Virial}}}] \leavevmode
\sphinxAtStartPar
For homogeneous system, the long\sphinxhyphen{}range correction virial can be analytically calculated:
\begin{equation*}
\begin{split}W_{\texttt{LRC(VDW)}} = \frac{2\pi N^2}{V} \int_{r=r_{cut}}^{\infty} r^3 F_{\texttt{VDW}}(r) dr\end{split}
\end{equation*}\begin{equation*}
\begin{split}F_{\texttt{VDW}}(r) =
\begin{cases}
  \frac{6 \alpha\epsilon}{r\big(\alpha-6\big)} \bigg[\frac{r}{R{min}} \exp\bigg(\alpha \bigg[1-\frac{r}{R_{min}} \bigg]\bigg) - {\bigg(\frac{R_{min}}{r}\bigg)}^6 \bigg] &  r \geq R_{max} \\
  \infty & r < R_{max}
\end{cases}\end{split}
\end{equation*}
\sphinxAtStartPar
Substituting the Buckingham potential into the integral, the long\sphinxhyphen{}range correction virial term is defined by:
\begin{equation*}
\begin{split}W_{\texttt{LRC(VDW)}} = \frac{2\pi N^2}{V} \bigg[A \exp\big(\frac{-r_{cut}}{B}\big) \bigg(6 B^3 + 6 B^2 r_{cut} + 3 B {r_{cut}}^2 + {r_{cut}}^3 \bigg) - \frac{2C}{3 {r_{cut}}^3}   \bigg]\end{split}
\end{equation*}
\end{description}


\chapter{Coupling Interaction with \protect\(\lambda\protect\)}
\label{\detokenize{free_energy:coupling-interaction-with-lambda}}\label{\detokenize{free_energy::doc}}
\sphinxAtStartPar
In this section, the scaling nonbonded and long\sphinxhyphen{}range correction energies with \(\lambda\) is discussed in detailed.
\begin{equation*}
\begin{split}E_{\lambda} = E_{\lambda}(\texttt{VDW}) + E_{\lambda}(\texttt{Elect}) + E_{\lambda}(\texttt{LRC-VDW}) + E_{\lambda}(\texttt{LRC-Elect})\end{split}
\end{equation*}

\section{VDW}
\label{\detokenize{free_energy:vdw}}

\subsection{Soft\sphinxhyphen{}core}
\label{\detokenize{free_energy:soft-core}}
\sphinxAtStartPar
In free energy calculation, the \sphinxcode{\sphinxupquote{VDW}} interaction between solute and solvent is scaled with \(\lambda\), non\sphinxhyphen{}linearly (soft\sphinxhyphen{}core scheme), to avoid
end\sphinxhyphen{}point catastrophe and numerical issue
\begin{equation*}
\begin{split}E_{\lambda}(\texttt{VDW}) = \lambda_{\texttt{VDW}} E_{\texttt{VDW}}(r_{sc})\end{split}
\end{equation*}
\sphinxAtStartPar
the scaled solute\sphinxhyphen{}solvent distance, \(r_{sc}\) is defined as:
\begin{equation*}
\begin{split}r_{sc} = \bigg[\alpha {\big(1 - \lambda_{\texttt{VDW}} \big)}^{p}{\sigma}^6 + {r}^6 \bigg]^{\frac{1}{6}}\end{split}
\end{equation*}
\sphinxAtStartPar
where, \(\alpha\) and \(p\) are the soft\sphinxhyphen{}core parameters defined by user (\sphinxcode{\sphinxupquote{ScaleAlpha}}, \sphinxcode{\sphinxupquote{ScalePower}}) and \(\sigma\) is the diameter of atom.
To improve numerical convergence of the calculation, a minimum interaction diameter \(\sigma_{min}\) should be defined by user (\sphinxcode{\sphinxupquote{MinSigma}}) for any atom with a diameter
less than \(\sigma_{min}\), e.g. hydrogen atoms attached to oxygen in water or alcohols.

\sphinxAtStartPar
To calculate the solvation free energy with thermodynamic integration (TI) method, the derivative of energy with
respect to lambda (\(\frac{dE_{\lambda}(\texttt{VDW})}{d\lambda_{\texttt{VDW}}}\)) is required:
\begin{equation*}
\begin{split}\frac{dE_{\lambda}(\texttt{VDW})}{d\lambda_{\texttt{VDW}}} = E_{\texttt{VDW}}(r_{sc}) + \frac{p \alpha \lambda_{\texttt{VDW}}}{6} \bigg(1 - \lambda_{\texttt{VDW}}\bigg)^{p-1} \bigg(\frac{{\sigma}^6}{{r_{sc}}^5} \bigg) F_{\texttt{VDW}}(r_{sc})\end{split}
\end{equation*}

\section{Electrostatic}
\label{\detokenize{free_energy:electrostatic}}

\subsection{Hard\sphinxhyphen{}core}
\label{\detokenize{free_energy:hard-core}}
\sphinxAtStartPar
In free energy calculation, the \sphinxcode{\sphinxupquote{Coulombic}} interaction between solute and solvent can be scaled with \(\lambda\), \sphinxstylestrong{linearly} (hard\sphinxhyphen{}core scheme),
by setting the \sphinxcode{\sphinxupquote{ScaleCoulomb}} to false.
\begin{equation*}
\begin{split}E_{\lambda}(\texttt{Elect}) = \lambda_{\texttt{Elect}} E_{\texttt{Elect}}(r)\end{split}
\end{equation*}
\sphinxAtStartPar
where, \(r\) is the distance between solute and solvent, without any modification.

\sphinxAtStartPar
To calculate the solvation free energy with thermodynamic integration (TI) method, the derivative of energy with
respect to lambda (\(\frac{dE_{\lambda}(\texttt{Elect})}{d\lambda_{\texttt{Elect}}}\)) is required:
\begin{equation*}
\begin{split}\frac{dE_{\lambda}(\texttt{Elect})}{d\lambda_{\texttt{Elect}}} = E_{\texttt{Elect}}(r)\end{split}
\end{equation*}
\begin{sphinxadmonition}{warning}{Warning:}
\sphinxAtStartPar
To avoid end\sphinxhyphen{}point catastrophe and numerical issue, it’s suggested to turn on the \sphinxcode{\sphinxupquote{VDW}} interaction completely, before turning
on the \sphinxcode{\sphinxupquote{Coulombic}} interaction.
\end{sphinxadmonition}


\subsection{Soft\sphinxhyphen{}core}
\label{\detokenize{free_energy:id1}}
\sphinxAtStartPar
In free energy calculation, the \sphinxcode{\sphinxupquote{Coulombic}} interaction between solute and solvent can be scaled with \(\lambda\), \sphinxstylestrong{non\sphinxhyphen{}linearly} (soft\sphinxhyphen{}core scheme), to avoid
end\sphinxhyphen{}point catastrophe and numerical issue. This option can be activated by setting the \sphinxcode{\sphinxupquote{ScaleCoulomb}} to true.
\begin{equation*}
\begin{split}E_{\lambda}(\texttt{Elect}) = \lambda_{\texttt{Elect}} E_{\texttt{Elect}}(r_{sc})\end{split}
\end{equation*}
\sphinxAtStartPar
the scaled solute\sphinxhyphen{}solvent distance, \(r_{sc}\) is defined as:
\begin{equation*}
\begin{split}r_{sc} = \bigg[\alpha {\big(1 - \lambda_{\texttt{Elect}} \big)}^{p}{\sigma}^6 + {r}^6 \bigg]^{\frac{1}{6}}\end{split}
\end{equation*}
\sphinxAtStartPar
where, \(\alpha\) and \(p\) are the soft\sphinxhyphen{}core parameters defined by user (\sphinxcode{\sphinxupquote{ScaleAlpha}}, \sphinxcode{\sphinxupquote{ScalePower}}) and \(\sigma\) is the diameter of atom.
To improve numerical convergence of the calculation, a minimum interaction diameter \(\sigma_{min}\) should be defined by user (\sphinxcode{\sphinxupquote{MinSigma}}) for any atom with a diameter
less than \(\sigma_{min}\), e.g. hydrogen atoms attached to oxygen in water or alcohols.

\sphinxAtStartPar
To calculate the solvation free energy with thermodynamic integration (TI) method, the derivative of energy with
respect to lambda (\(\frac{dE_{\lambda}(\texttt{Elect})}{d\lambda_{\texttt{Elect}}}\)) is required:
\begin{equation*}
\begin{split}\frac{dE_{\lambda}(\texttt{Elect})}{d\lambda_{\texttt{Elect}}} = E_{\texttt{Elect}}(r_{sc}) + \frac{p \alpha \lambda_{\texttt{Elect}}}{6} \bigg(1 - \lambda_{\texttt{Elect}}\bigg)^{p-1} \bigg(\frac{{\sigma}^6}{{r_{sc}}^5} \bigg) F_{\texttt{Elect}}(r_{sc})\end{split}
\end{equation*}
\begin{sphinxadmonition}{warning}{Warning:}
\sphinxAtStartPar
Using soft\sphinxhyphen{}core scheme to scale the coulombic interaction non\sphinxhyphen{}linearly, would result in \sphinxstylestrong{inaccurate} results if \sphinxcode{\sphinxupquote{Ewald}} method is activated.

\sphinxAtStartPar
Using \sphinxstyleemphasis{Ewald Summation Method}, we suggest to use hard\sphinxhyphen{}core scheme, to scale the coulombic interaction linearly with \(\lambda\).
\end{sphinxadmonition}


\section{Long\sphinxhyphen{}range Correction (VDW)}
\label{\detokenize{free_energy:long-range-correction-vdw}}
\sphinxAtStartPar
The effect of long\sphinxhyphen{}range corrections on predicted free energies were determined for \sphinxcode{\sphinxupquote{VDW}} interactions via a linear coupling with \(\lambda\).
\begin{equation*}
\begin{split}E_{\lambda}(\texttt{LRC-VDW}) = \lambda_{\texttt{VDW}} \Delta E_{\texttt{LRC(VDW)}}\end{split}
\end{equation*}
\sphinxAtStartPar
where, \(\Delta E_{\texttt{LRC(VDW)}}\) is the the change in the long\sphinxhyphen{}range correction energy, due to adding a fully interacting solute
to the solvent for \sphinxcode{\sphinxupquote{VDW}} interaction.

\sphinxAtStartPar
To calculate the solvation free energy with thermodynamic integration (TI) method, the derivative of energy with
respect to lambda (\(\frac{dE_{\lambda}(\texttt{LRC-VDW})}{\lambda_{\texttt{VDW}}}\)) is required:
\begin{equation*}
\begin{split}\frac{dE_{\lambda}(\texttt{LRC-VDW})}{d\lambda_{\texttt{VDW}}} = \Delta E_{\texttt{LRC(VDW)}}\end{split}
\end{equation*}

\section{Long\sphinxhyphen{}range Correction (Electrostatic)}
\label{\detokenize{free_energy:long-range-correction-electrostatic}}
\sphinxAtStartPar
Using \sphinxstyleemphasis{Ewald Summation Method}, the effect of long\sphinxhyphen{}range corrections on predicted free energies were determined for \sphinxcode{\sphinxupquote{Coulombic}} interactions
via a linear coupling with \(\lambda\).
\begin{equation*}
\begin{split}E_{\lambda}(\texttt{LRC-Elect}) = \lambda_{\texttt{Elect}} \bigg[\Delta E_{reciprocal} + \Delta E_{self} + \Delta E_{correction} \bigg]\end{split}
\end{equation*}
\sphinxAtStartPar
where, \(\Delta E_{reciprocal}\), \(\Delta E_{self}\), and \(\Delta E_{correction}\) are the the change in the reciprocal, self,
and correction energy term in \sphinxcode{\sphinxupquote{Ewald}} method, due to adding a fully interacting solute to the solvent.

\sphinxAtStartPar
To calculate the solvation free energy with thermodynamic integration (TI) method, the derivative of energy with
respect to lambda (\(\frac{dE_{\lambda}(\texttt{LRC-Elect})}{\lambda_{\texttt{Elect}}}\)) is required:
\begin{equation*}
\begin{split}\frac{dE_{\lambda}(\texttt{LRC-Elect})}{d\lambda_{\texttt{Elect}}} = \Delta E_{reciprocal} + \Delta E_{self} + \Delta E_{correction}\end{split}
\end{equation*}

\chapter{Hybrid Monte Carlo\sphinxhyphen{}Molecular Dynamics (MCMD)}
\label{\detokenize{hybrid_MC_MD:hybrid-monte-carlo-molecular-dynamics-mcmd}}\label{\detokenize{hybrid_MC_MD::doc}}
\sphinxAtStartPar
In this section, the tips and tricks to get a hybrid MCMD simumlation with GOMC and NAMD running are discussed.
Most of these issues will be handled by the scripts provided with py\sphinxhyphen{}MCMD, but the concerns are raised here for users interested in setting up custom systems.  Careful attention should be made to ensure the system is centered in the first octant of 3D space, originates at {[}boxlength/2, boxlength/2, boxlength/2{]}, and the box length excedes the radius of gyration of all molecules.

\sphinxAtStartPar
Link to documentation: \sphinxurl{https://py-mcmd.readthedocs.io/en/latest/}

\sphinxAtStartPar
Link to Github Repository: \sphinxurl{https://github.com/GOMC-WSU/py-MCMD}


\section{GOMC Requirements}
\label{\detokenize{hybrid_MC_MD:gomc-requirements}}
\sphinxAtStartPar
GOMC currently requires that Box length / 2 excede the radius of gyration of all molecules in the system.

\sphinxAtStartPar
Grand\sphinxhyphen{}Canonical Molecular Dynamics (GCMD) or Gibbs Ensemble with Molecular Dynamics changes the number of molecules in each box.  This will alter the ordering of the molecules, posing a challenge when the user tries to concatenate the trajectories or follow one atom through a trajectory.

\sphinxAtStartPar
The GOMC checkpoint file will reload the molecules in the original order to ensure the GOMC trajectories (PDB/DCD) have a consistent ordering for analysis.  Atoms that are not currently in a specific box are given the coordinate (0.00, 0.00, 0.00). The occupancy value corresponds to the box a molecule is currently in (e.g. 0.00 for box 0; 1.00 for box 1).

\sphinxAtStartPar
The restart binary coordinates, velocities, and box dimensions (xsc) from NAMD need to be loaded along with the checkpoint file, restart PDB, and restart PSF from the previous GOMC cycle.

\sphinxAtStartPar
The python script from the py\sphinxhyphen{}MCMD git repository, combine\_data\_NAMD\_GOMC.py, requires the GOMC step reset to 0 every cycle

\begin{sphinxVerbatim}[commandchars=\\\{\}]
InitStep          0
\end{sphinxVerbatim}


\section{NAMD Requirements}
\label{\detokenize{hybrid_MC_MD:namd-requirements}}
\sphinxAtStartPar
GOMC outputs all the files needed to continue a simulation box in Molecule Dynamics (pdb, psf, xsc, coor, vel, xsc).  These files should all be used.

\sphinxAtStartPar
There are certain flexibilities that NAMD allows for that GOMC doesn’t support.  To ensure the two systems are compatible the following settings in the NAMD configuration file are required:

\sphinxAtStartPar
Rigid bonds, since GOMC doesn’t support bond length sampling.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
rigidBonds          all
\end{sphinxVerbatim}

\sphinxAtStartPar
Fixed volume, since GOMC maintains the origin of the box at {[}box length/2, box length/2, box length/2{]}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{} Constant Pressure Control (variable volume)
langevinPiston        off

useGroupPressure      yes

useFlexibleCell       no

useConstantArea       no
\end{sphinxVerbatim}

\sphinxAtStartPar
Box origin must be centered at {[}box length/2, box length/2, box length/2{]}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
cellOrigin        x\PYGZus{}box length/2      y\PYGZus{}box length/2          z\PYGZus{}box length/2
\end{sphinxVerbatim}


\section{Dynamic Subvolumes for Dual Control Volume Molecular Dynamics}
\label{\detokenize{hybrid_MC_MD:dynamic-subvolumes-for-dual-control-volume-molecular-dynamics}}
\sphinxAtStartPar
To define a subvolume in the simulation, use the subvolume keywords to choose an subvolume id, center, either the geometric center of a list of atoms (dynamic) or absolute cartesian coordinate (static), and dimensions.  The residues that can be inserted/deleted in the subvolume, custom chemical potential, and periodicity of the subvolume may also be specified.  Fugacity can be replaced for chemical potential.  A chemical gradient can be established in the simulation by defining two or more subvolume with different chemical potentials of a given residue.  After the molecule is inserted/deleted within one subvolume, it can diffuse to the low concentration subvolume where it is deleted maintaining the concentration gradient via two (2) difference chemical potentials.

\sphinxAtStartPar
To define two control volumes forming a gradient from the left to the right of the box

\begin{sphinxVerbatim}[commandchars=\\\{\}]
SubVolumeBox                0       0

SubVolumeDim                0       left\PYGZus{}one\PYGZus{}fifth y\PYGZus{}dim\PYGZus{}box\PYGZus{}0 z\PYGZus{}dim\PYGZus{}box\PYGZus{}0

SubVolumeResidueKind        0       DIOX

SubVolumeRigidSwap          0       true

SubVolumeCenter             0       left\PYGZus{}center y\PYGZus{}origin\PYGZus{}box z\PYGZus{}origin\PYGZus{}box

SubVolumePBC                0       XYZ

SubVolumeChemPot            0       DIOX    \PYGZhy{}2000


SubVolumeBox                1       0

SubVolumeDim                1       right\PYGZus{}one\PYGZus{}fifth  y\PYGZus{}dim\PYGZus{}box\PYGZus{}0 z\PYGZus{}dim\PYGZus{}box\PYGZus{}0

SubVolumeResidueKind        1       DIOX

SubVolumeRigidSwap          1       true

SubVolumeCenter             1       right\PYGZus{}center y\PYGZus{}origin\PYGZus{}box z\PYGZus{}origin\PYGZus{}box

SubVolumePBC                1       XYZ

SubVolumeChemPot            1       DIOX    \PYGZhy{}4000
\end{sphinxVerbatim}


\section{Run a Hybrid Monte Carlo\sphinxhyphen{}Molecular Dynamics Sim}
\label{\detokenize{hybrid_MC_MD:run-a-hybrid-monte-carlo-molecular-dynamics-sim}}
\sphinxAtStartPar
GOMC and NAMD produce compatible input/output files, which allow the system to alternate between Monte Carlo and Molecular Dynamics.
The py\sphinxhyphen{}MCMD script automates the directory generation, running of GOMC and NAMD, and concatenation of the short alternating runs.
Simulating the Grand Canonical ensemble in GOMC with only molecule transfers, allows the MD simulations to continue where they left off, with a varying number of molecules.

\sphinxAtStartPar
Refer to the section on Hybrid Monte Carlo\sphinxhyphen{}Molecular Dynamics in the manual and attached links.

\sphinxAtStartPar
Link to documentation: \sphinxurl{https://py-mcmd.readthedocs.io/en/latest/}

\sphinxAtStartPar
Link to Github Repository: \sphinxurl{https://github.com/GOMC-WSU/py-MCMD}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git clone https://github.com/GOMC\PYGZhy{}WSU/py\PYGZhy{}MCMD.git
\PYGZdl{} \PYG{n+nb}{cd} py\PYGZhy{}MCMD
\PYG{c+c1}{\PYGZsh{}\PYGZsh{}\PYGZsh{} Run hybrid simulation}
\PYGZdl{} python run\PYGZus{}NAMD\PYGZus{}GOMC.py \PYGZhy{}f user\PYGZus{}input\PYGZus{}NAMD\PYGZus{}GOMC.json
\PYG{c+c1}{\PYGZsh{}\PYGZsh{}\PYGZsh{} Combine alternating GOMC/NAMD cycles into two single GOMC and NAMD data and trajectories.}
\PYGZdl{} python combine\PYGZus{}data\PYGZus{}NAMD\PYGZus{}GOMC.py \PYGZhy{}f user\PYGZus{}input\PYGZus{}combine\PYGZus{}data\PYGZus{}NAMD\PYGZus{}GOMC.json
\end{sphinxVerbatim}


\chapter{How to?}
\label{\detokenize{howto:how-to}}\label{\detokenize{howto::doc}}
\sphinxAtStartPar
In this section, we are providing a summary of what actions or modification need to be done in order to answer your simulation problem.


\section{Visualizing Simulation}
\label{\detokenize{howto:visualizing-simulation}}
\sphinxAtStartPar
If \sphinxcode{\sphinxupquote{CoordinatesFreq}} is enabled in configuration file, GOMC will output the molecule coordinates every
specified stpes. The PDB and PSF output (merging of atom entries) has already been mentioned/explained in
previous sections. To recap: The PDB file’s ATOM entries’ occupancy is used to represent the box the molecule
is in for the current frame. All molecules are listed in order in which they were read (i.e. if box 0 has
\(1, 2, ..., N1\) molecules and box 1 has \(1, 2, ..., N2\) molecules, then all of the molecules in
box 0 are listed first and all the molecules in box 1, i.e. \(1, 2 ,... ,N1\), \(N1 + 1, ..., N1 + N2\)).
PDB frames are written as standard PDBs to consecutive file frames.

\sphinxAtStartPar
To visualize, open the output PDB and PSF files by GOMC using VMD, type this command in the terminal:

\sphinxAtStartPar
For all simulation except Gibbs ensemble that has one simulation box:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} vmd   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}merged.psf  ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}0.pdb
\end{sphinxVerbatim}

\sphinxAtStartPar
For Gibbs ensemble, visualizing the first box:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} vmd   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}merged.psf  ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}0.pdb
\end{sphinxVerbatim}

\sphinxAtStartPar
For Gibbs ensemble, visualizing the second box:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} vmd   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}merged.psf  ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}1.pdb
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Restart coordinate file (OutputName\_BOX\_0\_restart.pdb) cannot be visualize using merged psf file, because atom number does not match. However, you can still open it in vmd using following command and vmd will automatically find the bonds of the molecule based on the coordinates.
\end{sphinxadmonition}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} vmd   ISB\PYGZus{}T\PYGZus{}270\PYGZus{}k\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.pdb
\end{sphinxVerbatim}


\section{Build molecule and topology file}
\label{\detokenize{howto:build-molecule-and-topology-file}}
\sphinxAtStartPar
There are many open\sphinxhyphen{}source software that can build a molecule for you, such as \sphinxhref{https://avogadro.cc/docs/getting-started/drawing-molecules/}{Avagadro} ,
\sphinxhref{http://www.ks.uiuc.edu/Research/vmd/plugins/molefacture/}{molefacture} in VMD and more. Here we use molefacture features to not only build a molecule,
but also creating the topology file.


\subsection{Regular molecule}
\label{\detokenize{howto:regular-molecule}}
\sphinxAtStartPar
First, make sure that VMD is installed on your computer. Then, to learn how to build a single PDB file and topology file for united atom butane molecule,
please refer to this \sphinxhref{https://github.com/GOMC-WSU/Workshop/blob/master/NVT/butane/build/Molefacture.pdf}{document} .

\sphinxAtStartPar
We encourage to try to go through our workshop materials:
\begin{itemize}
\item {} 
\sphinxAtStartPar
To try two days workshop, execute the following command in your terminal to clone the workshop:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git  clone    https://github.com/GOMC\PYGZhy{}WSU/Workshop.git \PYGZhy{}\PYGZhy{}branch master \PYGZhy{}\PYGZhy{}single\PYGZhy{}branch
\PYGZdl{} \PYG{n+nb}{cd}   Workshop
\end{sphinxVerbatim}

\sphinxAtStartPar
or simply download it from \sphinxhref{https://github.com/GOMC-WSU/Workshop/tree/master}{GitHub} .

\item {} 
\sphinxAtStartPar
To try two hours workshop, execute the following command in your terminal to clone the workshop:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git  clone    https://github.com/GOMC\PYGZhy{}WSU/Workshop.git \PYGZhy{}\PYGZhy{}branch AIChE \PYGZhy{}\PYGZhy{}single\PYGZhy{}branch
\PYGZdl{} \PYG{n+nb}{cd}   Workshop
\end{sphinxVerbatim}

\sphinxAtStartPar
or simply download it from \sphinxhref{https://github.com/GOMC-WSU/Workshop/tree/AIChE}{GitHub} .

\end{itemize}


\subsection{Molecule with dummy atoms}
\label{\detokenize{howto:molecule-with-dummy-atoms}}
\sphinxAtStartPar
To simulate a molecule that includes one or more atoms with electrostatic interaction only and no LJ interaction (i.e. dummy atom near of the oxygen along
the bisector of the HOH angle in \sphinxhref{http://dx.doi.org/10.1063/1.2121687}{TIP4P water model}), we must perform the following steps
to define the dummy atom/atoms:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Create a PDB file for single water molecule atoms (H1, O, H2) and a dummy atom (M, in this example), where dummy atom located at 0.150 Å of oxygen and along
the bisector of the H1\sphinxhyphen{}O\sphinxhyphen{}H2 angle.

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
CRYST1    0.000    0.000    0.000  90.00  90.00  90.00 P 1           1
ATOM      1  O   TIP4    1      \PYGZhy{}0.189   1.073   0.000  0.00  0.00           O
ATOM      2  H1  TIP4    1       0.768   1.114   0.000  0.00  0.00           H
ATOM      3  H2  TIP4    1      \PYGZhy{}0.469   1.988   0.000  0.00  0.00           H
ATOM      4  M   TIP4    1      \PYGZhy{}0.102   1.195   0.000  0.00  0.00           D
END
\end{sphinxVerbatim}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{1}
\item {} 
\sphinxAtStartPar
Pack your desire number of TIP4 water molecule in a box using packmol, as explained before.

\item {} 
\sphinxAtStartPar
Include the dummy atom (M) and its charge in your topology file. Define a bond between oxygen and dummy atom.
Use vmd and build script to generate your PSF files.

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
* Custom top file \PYGZhy{}\PYGZhy{} TIP4P water

MASS   1  OH    15.9994  O !
MASS   2  HO     1.0080  H !
MASS   3  MO     0.0000  D ! Dummy atom for TIP4P model

DEFA FIRS none LAST none
AUTOGENERATE ANGLES DIHEDRALS

RESI TIP4           0.0000 ! TIP4P water
GROUP
ATOM O      OH      0.0000 !        O
ATOM H1     HO      0.5564 !     /  |  \PYGZbs{}
ATOM H2     HO      0.5564 !    /   M   \PYGZbs{}
ATOM M      MO     \PYGZhy{}1.1128 !  H1        H2
BOND   O  H1   O  H2   O  M
PATCHING FIRS NONE LAST NONE

END
\end{sphinxVerbatim}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{3}
\item {} 
\sphinxAtStartPar
Define all bonded parameters (bond, angles, and dihedral) and nonbonded parameters in your parameter file.

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
*parameteres for TIP4P

BONDS
!
!V(bond) = Kb(b \PYGZhy{} b0)**2
!
!atom type          Kb          b0
OH   HO    99999999999       0.9572 ! TIP4P O\PYGZhy{}H bond length
OH   MO    99999999999       0.1500 ! TIP4P M\PYGZhy{}O bond length


ANGLES
!
!V(angle) = Ktheta(Theta \PYGZhy{} Theta0)**2
!
!atom types         Ktheta       Theta0
HO   OH   HO    9999999999999    104.52  ! H\PYGZhy{}O\PYGZhy{}H Fix Angle
HO   OH   MO    9999999999999     52.26  ! H\PYGZhy{}O\PYGZhy{}M Fix Angle


DIHEDRALS
!
!V(dihedral) = Kchi(1 + cos(n(chi) \PYGZhy{} delta))
!
!atom types             Kchi    n   delta


NONBONDED
!
!V(Lennard\PYGZhy{}Jones) = Eps,i,j[(Rmin,i,j/ri,j)**12 \PYGZhy{} 2(Rmin,i,j/ri,j)**6]
!
!atom  ignored      epsilon      Rmin/2   ignored   eps,1\PYGZhy{}4    Rmin/2,1\PYGZhy{}4
HO      0.000000     0.00000    0.000000    0.0     0.0         0.0
MO      0.000000     0.00000    0.000000    0.0     0.0         0.0
OH      0.000000    \PYGZhy{}0.18521    1.772873    0.0     0.0         0.0
\end{sphinxVerbatim}


\section{Simulate rigid molecule}
\label{\detokenize{howto:simulate-rigid-molecule}}
\sphinxAtStartPar
Currently, GOMC can simulate rigid molecules for any molecular topology in NVT and NPT ensemble, if none of the Monte Carlo moves that lead to change in
molecular configuration (e.g. \sphinxcode{\sphinxupquote{Regrowth}}, \sphinxcode{\sphinxupquote{Crankshaft}}, \sphinxcode{\sphinxupquote{IntraSwap}}, and etc.) was used.

\sphinxAtStartPar
In general, GOMC can simulate rigid molecules in all ensembles for the following molecular topology:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Linear and branched molecules with no dihedrals. For instance, carbon dioxide, dimethyl ether, and all water models (SPC, SPC/E, TIP3P, TIP4P, etc).

\item {} 
\sphinxAtStartPar
Cyclic molecules, where at least two atoms in all defined angles, belong to the body of the ring. For instance, benzene, toluene, Xylene, and more.

\end{enumerate}

\begin{sphinxadmonition}{important}{Important:}\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
For linear and branched molecule, the molecule’s bonds and angles  will be adjusted according to the equilibrium values, defined in parameter file.

\item {} 
\sphinxAtStartPar
For cyclic molecules, the molecule’s bonds and angles would not change! It is very important to create the initial molecule with correct bonds and angles.

\end{enumerate}
\end{sphinxadmonition}


\subsection{Setup rigid  molecule}
\label{\detokenize{howto:setup-rigid-molecule}}
\sphinxAtStartPar
To simulate the rigid molecules in GOMC, we need to perform the following steps:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Define all bonds in topology file and use \sphinxstylestrong{AUTOGENERATE ANGLES DIHEDRALS} in topology file to specify all angles and dihedral in PSF files.

\item {} 
\sphinxAtStartPar
Define all bond parameters in the parameter file. If you wish to not to include the bond energy in your simulation, set the
the \(K_b\) to a large value i.e. “999999999999”.

\item {} 
\sphinxAtStartPar
Define all angle parameters in the parameter file. If you wish to not to include the bend energy in your simulation, set the
the \(K_{\theta}\) to a large value i.e. “999999999999”.

\item {} 
\sphinxAtStartPar
Define all dihedral parameters in parameter file. If you wish to not to include the dihedral energy in your simulation, set the all
the \(C_n\) to zero. \sphinxstylestrong{For cyclic molecules only}

\end{enumerate}


\section{Restart the simulation}
\label{\detokenize{howto:restart-the-simulation}}

\subsection{Restart the simulation with \sphinxstyleliteralintitle{\sphinxupquote{Restart}}}
\label{\detokenize{howto:restart-the-simulation-with-restart}}
\sphinxAtStartPar
If you intend to start a new simulation from previous simulation state, you can use this option. Restarting the simulation with \sphinxcode{\sphinxupquote{Restart}} \sphinxstylestrong{would not} result in
an identitcal outcome, as if previous simulation was continued.
Make sure that in the previous simulation config file, the flag \sphinxcode{\sphinxupquote{RestartFreq}} was activated and the restart PDB file/files (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_0\_restart.pdb)
and merged PSF file (\sphinxcode{\sphinxupquote{OutputName}}\_merged.psf) were printed.

\sphinxAtStartPar
In order to restart the simulation from previous simulation we need to perform the following steps to modify the config file:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Set the \sphinxcode{\sphinxupquote{Restart}} to True.

\item {} 
\sphinxAtStartPar
Use the dumped restart PDB file to set the \sphinxcode{\sphinxupquote{Coordinates}} for each box.

\item {} 
\sphinxAtStartPar
Use the dumped merged PSF file to set the \sphinxcode{\sphinxupquote{Structure}} for both boxes.

\item {} 
\sphinxAtStartPar
It is a good practice to comment out the \sphinxcode{\sphinxupquote{CellBasisVector}} by adding ‘\#’ at the beginning of each cell basis vector. However, GOMC will override
the cell basis information with the cell basis data from restart PDB file/files.

\item {} 
\sphinxAtStartPar
Use the different \sphinxcode{\sphinxupquote{OutputName}} to avoid overwriting the output files.

\end{enumerate}

\sphinxAtStartPar
Here is the example of starting the NPT simulation of dimethyl ether, from equilibrated NVT simulation:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} Parameters need to be modified
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Restart         true

Coordinates     0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.pdb

Structure       0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.psf

\PYGZsh{}CellBasisVector1   0       45.00   0.00    0.00
\PYGZsh{}CellBasisVector2   0       0.00    55.00   0.00
\PYGZsh{}CellBasisVector3   0       0.00    0.00    45.00

OutputName          dimethylether\PYGZus{}NPT
\end{sphinxVerbatim}

\sphinxAtStartPar
Here is the example of starting the NPT\sphinxhyphen{}GEMC simulation of dimethyl ether, from equilibrated NVT simulation:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} Parameters need to be modified
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Restart         true

Coordinates     0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.pdb
Coordinates     1   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}1\PYGZus{}restart.pdb

Structure       0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.psf
Structure       1   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}1\PYGZus{}restart.psf

\PYGZsh{}CellBasisVector1   0       45.00   0.00    0.00
\PYGZsh{}CellBasisVector2   0       0.00    55.00   0.00
\PYGZsh{}CellBasisVector3   0       0.00    0.00    45.00

\PYGZsh{}CellBasisVector1   1       45.00   0.00    0.00
\PYGZsh{}CellBasisVector2   1       0.00    55.00   0.00
\PYGZsh{}CellBasisVector3   1       0.00    0.00    45.00

OutputName          dimethylether\PYGZus{}NPT\PYGZus{}GEMC
\end{sphinxVerbatim}


\subsection{Restart the simulation with \sphinxstyleliteralintitle{\sphinxupquote{Checkpoint}}}
\label{\detokenize{howto:restart-the-simulation-with-checkpoint}}
\sphinxAtStartPar
If you intend to continue your simulation from previous simulation, you can use this option. Restarting the simulation with \sphinxcode{\sphinxupquote{Checkpoint}} would result in an
identitcal outcome, as if previous simulation was continued.
Make sure that in the previous simulation config file, the flag \sphinxcode{\sphinxupquote{RestartFreq}} was activated and the restart PDB file/files (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_N\_restart.pdb)
, restart PSF file/files (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_N\_restart.psf), binary coodinate file/files (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_N\_restart.coor), XSC file/files (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_N\_restart.xsc) , and checkpoint file (\sphinxcode{\sphinxupquote{OutputName}}\_restart.chk) were printed.

\sphinxAtStartPar
In order to restart the simulation from previous simulation we need to perform the following steps to modify the config file:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Set the \sphinxcode{\sphinxupquote{Checkpoint}} to True and provide the Checkpoint file.

\item {} 
\sphinxAtStartPar
Use the dumped restart PDB files to set the \sphinxcode{\sphinxupquote{Coordinates}} for each box.

\item {} 
\sphinxAtStartPar
Use the dumped restart PSF files to set the \sphinxcode{\sphinxupquote{Structure}} for both boxes.

\item {} 
\sphinxAtStartPar
Use the dumped restart xsc files to set the \sphinxcode{\sphinxupquote{extendedSystem}} for both boxes.

\item {} 
\sphinxAtStartPar
Use the dumped restart coor files to set the \sphinxcode{\sphinxupquote{binCoordinates}} for both boxes.

\item {} 
\sphinxAtStartPar
It is a good practice to comment out the \sphinxcode{\sphinxupquote{CellBasisVector}} by adding ‘\#’ at the beginning of each cell basis vector. However, GOMC will override
the cell basis information with the cell basis data from XSC file/files.

\item {} 
\sphinxAtStartPar
Use the different \sphinxcode{\sphinxupquote{OutputName}} to avoid overwriting the output files.

\end{enumerate}

\sphinxAtStartPar
Here is the example of restarting the NPT simulation of dimethyl ether, from equilibrated NVT simulation:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} Parameters need to be modified
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Checkpoint   true   dimethylether\PYGZus{}NVT\PYGZus{}restart.chk

Coordinates     0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.pdb

Structure       0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.psf

extendedSystem   0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.xsc

binCoordinates   0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.coor

\PYGZsh{}CellBasisVector1   0       45.00   0.00    0.00
\PYGZsh{}CellBasisVector2   0       0.00    55.00   0.00
\PYGZsh{}CellBasisVector3   0       0.00    0.00    45.00

OutputName          dimethylether\PYGZus{}NPT
\end{sphinxVerbatim}

\sphinxAtStartPar
Here is the example of restarting the NPT\sphinxhyphen{}GEMC simulation of dimethyl ether, from equilibrated NVT simulation:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} Parameters need to be modified
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Checkpoint   true   dimethylether\PYGZus{}NVT\PYGZus{}restart.chk

Coordinates     0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.pdb
Coordinates     1   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}1\PYGZus{}restart.pdb

Structure     0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.psf
Structure     1   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}1\PYGZus{}restart.psf

extendedSystem   0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.xsc
extendedSystem   1   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}1\PYGZus{}restart.xsc

binCoordinates   0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0\PYGZus{}restart.coor
binCoordinates   1   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}1\PYGZus{}restart.coor

\PYGZsh{}CellBasisVector1   0       45.00   0.00    0.00
\PYGZsh{}CellBasisVector2   0       0.00    55.00   0.00
\PYGZsh{}CellBasisVector3   0       0.00    0.00    45.00

\PYGZsh{}CellBasisVector1   1       45.00   0.00    0.00
\PYGZsh{}CellBasisVector2   1       0.00    55.00   0.00
\PYGZsh{}CellBasisVector3   1       0.00    0.00    45.00

OutputName          dimethylether\PYGZus{}NPT\PYGZus{}GEMC
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
As of right now, restarting is not supported for Multi\sphinxhyphen{}Sim.
\end{sphinxadmonition}


\section{Recalculate the energy}
\label{\detokenize{howto:recalculate-the-energy}}
\sphinxAtStartPar
GOMC is capable of recalculate the energy of previous simulation snapshot, with same or different force field. Simulation snapshot is the printed molecule’s
coordinates at specific steps, which controls by \sphinxcode{\sphinxupquote{CoordinatesFreq}}. First, we need to make sure that in the previous simulation config file, the flag \sphinxcode{\sphinxupquote{CoordinatesFreq}}
was activated and the coordinates PDB file/files (\sphinxcode{\sphinxupquote{OutputName}}\_BOX\_0.pdb) and merged PSF file (\sphinxcode{\sphinxupquote{OutputName}}\_merged.psf) were printed.

\sphinxAtStartPar
In order to recalculate the energy from previous simulation we need to perform the following steps to modify the config file:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Set the \sphinxcode{\sphinxupquote{Restart}} to True.

\item {} 
\sphinxAtStartPar
Use the dumped coordinates PDB file to set the \sphinxcode{\sphinxupquote{Coordinates}} for each box.

\item {} 
\sphinxAtStartPar
Use the dumped merged PSF file to set the \sphinxcode{\sphinxupquote{Structure}} for both boxes.

\item {} 
\sphinxAtStartPar
Set the \sphinxcode{\sphinxupquote{RunSteps}} to zero to activare the energy recalculation.

\item {} 
\sphinxAtStartPar
Use the different \sphinxcode{\sphinxupquote{OutputName}} to avoid overwriting the merged PSF files.

\end{enumerate}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
GOMC only recalculated the energy terms and does not recalulate the thermodynamic properties. Hence, no output file, except merged PSF file, will be
generated.
\end{sphinxadmonition}

\sphinxAtStartPar
Here is the example of recalculating energy from previous NVT simulation snapshot:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} Parameters need to be modified
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Restart         true

Coordinates     0   dimethylether\PYGZus{}NVT\PYGZus{}BOX\PYGZus{}0.pdb

Structure       0   dimethylether\PYGZus{}NVT\PYGZus{}merged.psf

RunSteps        0

OutputName          Recalculate
\end{sphinxVerbatim}


\section{Simulate adsorption}
\label{\detokenize{howto:simulate-adsorption}}
\sphinxAtStartPar
GOMC is capable of simulating gas adsorption in rigid framework using GCMC and NPT\sphinxhyphen{}GEMC simulation. In this section, we discuss how to generate PDB and PSF file,
how to modify the configuration file to simulate adsorption.


\subsection{Build PDB and PSF file}
\label{\detokenize{howto:build-pdb-and-psf-file}}
\sphinxAtStartPar
Generating PDB and PSF file for reservoir is similar to generating PDB and PSF file for isobutane, explained before. Here, we are focusing on how to generate
PDB and PSF file for adsorbent.
As mensioned before, GOMC can only read PDB and PSF file as input file. If you are using “*.cif” file for your adsorbent, you need to perform few steps
to extend the unit cell and export it as PDB file. There are two ways that you can prepare your adsorption simulation:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Using High Throughput Screening (HTS)}

\sphinxAtStartPar
GOMC development group created a python code combined with Tcl scripting to automatically generate GOMC input files for adsorption simulation.
In this code, we use CoRE\sphinxhyphen{}MOF repository created by \sphinxhref{https://pubs.acs.org/doi/abs/10.1021/cm502594j}{Snurr et al.} to prepare the simulation input file.

\sphinxAtStartPar
To try this code, execute the following command in your terminal to clone the HTS repository:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git  clone    https://github.com/GOMC\PYGZhy{}WSU/Workshop.git \PYGZhy{}\PYGZhy{}branch HTS \PYGZhy{}\PYGZhy{}single\PYGZhy{}branch
\PYGZdl{} \PYG{n+nb}{cd}   Workshop
\end{sphinxVerbatim}

\sphinxAtStartPar
or simply download it from \sphinxhref{https://github.com/GOMC-WSU/Workshop/tree/HTS}{GitHub} .

\sphinxAtStartPar
Make sure that you installed all \sphinxhref{https://github.com/GOMC-WSU/Workshop/blob/HTS/GOMC\_Software\_Requirements.pdf}{GOMC software requirement}. Follow the
“Readme.md” for more information.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Manual Preparation}

\sphinxAtStartPar
To illustrate the steps that need to be taken to prepare the PDB and PSF file, we will use an example provided in one of our workshop. Make sure that you
installed all \sphinxhref{https://github.com/GOMC-WSU/Workshop/blob/master/GOMC\_Requirements.pdf}{GOMC software requirement}.

\sphinxAtStartPar
To clone the workshop, execute the following command in your terminal to clone the workshop:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git  clone    https://github.com/GOMC\PYGZhy{}WSU/Workshop.git \PYGZhy{}\PYGZhy{}branch master \PYGZhy{}\PYGZhy{}single\PYGZhy{}branch
\end{sphinxVerbatim}

\sphinxAtStartPar
or simply download it from \sphinxhref{https://github.com/GOMC-WSU/Workshop/tree/master}{GitHub} .

\sphinxAtStartPar
To show how to extend the unit cell of IRMOF\sphinxhyphen{}1 and build the PDB and PSF files, change your directory to:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd}   Workshop/adsorption/GCMC/argon\PYGZus{}IRMOF\PYGZus{}1/build/base/.
\end{sphinxVerbatim}

\sphinxAtStartPar
In this directory, there is a “README.txt” file, which provides detailed information of steps need to be taken. Here we just provide a summary of these steps:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Extend the unit cell of “EDUSIF\_clean\_min.cif” file using \sphinxhref{https://jp-minerals.org/vesta/en/download.html}{VESTA}. To learn how to extend the
unit cell, removing bonds, and export it as PDB file, please refere to this \sphinxhref{https://github.com/GOMC-WSU/Workshop/blob/master/adsorption/GCMC/argon\_IRMOF\_1/build/base/VESTA.pdf}{documente} to generate “EDUSIF\_clean\_min.pdb” file.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Generated PDB file does not provide all necessary information. Further modification must be made.
\end{sphinxadmonition}

\item {} 
\sphinxAtStartPar
The easy way to generate PSF file is to treat each atom as a separate molecule kind to avoid defining bonds, angles, and dihedrals. To modify the “EDUSIF\_clean\_min.pdb” file (set the residue ID, resname, …), execute the following command to generate the
“EDUSIF\_clean\_min\_modified.pdb” file.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
vmd \PYGZhy{}dispdev text \PYGZlt{} convert\PYGZus{}VESTA\PYGZus{}PDB.tcl
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
Treating each atom as separate molecule kind will make it easy to generate topology file. Here is an example of topology file where each atom is treated
as a separate residue kind:

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
* Topology file for IRMOF\PYGZhy{}1 (Zn4O(BDC)3)
!
MASS   1  O     15.999      O  !
MASS   2  C     12.011      C  !
MASS   3  H      1.008      H  !
MASS   4  ZN    65.380      ZN !

DEFA FIRS none LAST none
AUTOGENERATE ANGLES DIHEDRALS

RESI    C         0.000
GROUP
ATOM    C   C     0.000
PATCHING FIRS NONE LAST NONE

RESI    H         0.000
GROUP
ATOM    H   H     0.000
PATCHING FIRS NONE LAST NONE

RESI    O          0.000
GROUP
ATOM    O   O      0.000
PATCHING FIRS NONE LAST NONE

RESI    Zn         0.000
GROUP
ATOM    Zn  ZN     0.000
PATCHING FIRS NONE LAST NONE

END
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
To generate the PSF file, each molecule kind must be separated and stored in separate pdb file. Then we use VMD to generate the PSF file.
All these process are scripted in “build\_EDUSIF\_auto.tcl” and we just need to execute the following command to generate the “IRMOF\_1\_BOX\_0.pdb” and
“IRMOF\_1\_BOX\_0.psf” files.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
vmd \PYGZhy{}dispdev text \PYGZlt{} build\PYGZus{}EDUSIF\PYGZus{}auto.tcl
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
Last steps to fix the adsorbent atoms in their position. As mensioned in PDB section, setting the \sphinxcode{\sphinxupquote{Beta = 1.00}} value of a molecule in PDB file, will
fix that molecule position. This can be done by a text editor but here we use another Tcl scrip to do that. Execute the following command in your terminal
to set the \sphinxcode{\sphinxupquote{Beta}} value of all atoms in “IRMOF\_1\_BOX\_0.pdb” to 1.00.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
vmd \PYGZhy{}dispdev text \PYGZlt{} setBeta.tcl
\end{sphinxVerbatim}

\end{enumerate}


\subsection{Adsorption in GCMC}
\label{\detokenize{howto:adsorption-in-gcmc}}
\sphinxAtStartPar
To simulate adsorption using GCMC ensemble, we need to perform the following steps to modify the config file:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Use the generated PDB files for adsorbent and adsorbate to set the \sphinxcode{\sphinxupquote{Coordinates}}.

\item {} 
\sphinxAtStartPar
Use the generated PSF files for adsorbent and adsorbate to set the \sphinxcode{\sphinxupquote{Structure}}.

\item {} 
\sphinxAtStartPar
Calculate the cell basis vectors for each box and set the \sphinxcode{\sphinxupquote{CellBasisVector1,2,3}} for each box.

\end{enumerate}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
To calculate the cell basis vector with cell length \(\boldsymbol{a} , \boldsymbol{b}, \boldsymbol{c}\) and cell angle
\(\alpha, \beta. \gamma\) we use the following equations:

\sphinxAtStartPar
\(a_x = \boldsymbol{a}\)

\sphinxAtStartPar
\(a_y = 0.0\)

\sphinxAtStartPar
\(a_z = 0.0\)

\sphinxAtStartPar
\(b_x = \boldsymbol{b} \times cos(\gamma)\)

\sphinxAtStartPar
\(b_y = \boldsymbol{b} \times sin(\gamma)\)

\sphinxAtStartPar
\(c_x = \boldsymbol{c} \times cos(\beta)\)

\sphinxAtStartPar
\(c_y = \boldsymbol{c} \times \frac{ cos(\alpha) - cos(\beta) \times cos(\gamma) } { sin(\gamma) }\)

\sphinxAtStartPar
\(c_z = \boldsymbol{c} \times \sqrt {{sin(\beta)}^2 - { \bigg(\frac{ cos(\alpha) - cos(\beta) \times cos(\gamma) } { sin(\gamma) }} \bigg)^2}\)

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{CellBasisVector1}} = \((a_x , a_y, a_z)\)

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{CellBasisVector2}} = \((b_x , b_y, b_z)\)

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{CellBasisVector3}} = \((c_x , c_y, c_z)\)
\end{sphinxadmonition}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{3}
\item {} 
\sphinxAtStartPar
Set the \sphinxcode{\sphinxupquote{Fugacity}} for adsorbate and include \sphinxcode{\sphinxupquote{Fugacity}} for adsorbent with arbitrary value (e.g. 0.00).

\end{enumerate}

\sphinxAtStartPar
Here is the example of argon (AR) adsorption at 5 bar in IRMOF\sphinxhyphen{}1 using GCMC ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} Parameters need to be modified
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Coordinates     0   ../build/base/IRMOF\PYGZus{}1\PYGZus{}BOX\PYGZus{}0.pdb
Coordinates     1   ../build/reservoir/START\PYGZus{}BOX\PYGZus{}1.pdb

Structure       0   ../build/base/IRMOF\PYGZus{}1\PYGZus{}BOX\PYGZus{}0.psf
Structure       1   ../build/reservoir/START\PYGZus{}BOX\PYGZus{}1.psf

CellBasisVector1    0   36.8140   0.00     0.00
CellBasisVector2    0   18.2583  31.9880   0.00
CellBasisVector3    0   18.2712  10.5596  30.1748

CellBasisVector1    1   40.00     0.00    0.00
CellBasisVector2    1    0.00    40.00    0.00
CellBasisVector3    1    0.00    00.00   40.00

Fugacity    AR      5.0
Fugacity    C       0.0
Fugacity    H       0.0
Fugacity    O       0.0
Fugacity    ZN      0.0
\end{sphinxVerbatim}


\subsection{Adsorption in NPT\sphinxhyphen{}GEMC}
\label{\detokenize{howto:adsorption-in-npt-gemc}}
\sphinxAtStartPar
To simulate adsorption using NPT\sphinxhyphen{}GEMC ensemble, simulaiton box 0 is used for adsorbent with fixed volume and simulaiton box 1 is used for adsorbate, where
volume of this box is fluctuating at imposed pressure. To simulation adsorption in NPT\sphinxhyphen{}GEMC ensemble we need to perform the following steps to modify the
config file:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Use the generated PDB file for adsorbent to set the \sphinxcode{\sphinxupquote{Coordinates}} for box 0.

\item {} 
\sphinxAtStartPar
Use the generated PDB file for adsorbate to set the \sphinxcode{\sphinxupquote{Coordinates}} for box 1.

\item {} 
\sphinxAtStartPar
Use the generated PSF file for adsorbent to set the \sphinxcode{\sphinxupquote{Structure}} for box 0.

\item {} 
\sphinxAtStartPar
Use the generated PSF file for adsorbate to set the \sphinxcode{\sphinxupquote{Structure}} for box 1.

\item {} 
\sphinxAtStartPar
Calculate the cell basis vectors for each box and set the \sphinxcode{\sphinxupquote{CellBasisVector1,2,3}} for each box.

\item {} 
\sphinxAtStartPar
Set the \sphinxcode{\sphinxupquote{GEMC}} simulaiton type to “NPT”.

\item {} 
\sphinxAtStartPar
Set the imposed \sphinxcode{\sphinxupquote{Pressure}} (bar) for fluid phase.

\item {} 
\sphinxAtStartPar
Keep the volume of box 0 constant by activating the \sphinxcode{\sphinxupquote{FixVolBox0}}.

\end{enumerate}

\sphinxAtStartPar
Here is the example of argon (AR) adsorption at 5 bar in IRMOF\sphinxhyphen{}1 using NPT\sphinxhyphen{}GEMC ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} Parameters need to be modified
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Coordinates     0   ../build/base/IRMOF\PYGZus{}1\PYGZus{}BOX\PYGZus{}0.pdb
Coordinates     1   ../build/reservoir/START\PYGZus{}BOX\PYGZus{}1.pdb

Structure       0   ../build/base/IRMOF\PYGZus{}1\PYGZus{}BOX\PYGZus{}0.psf
Structure       1   ../build/reservoir/START\PYGZus{}BOX\PYGZus{}1.psf

CellBasisVector1    0   36.8140   0.00     0.00
CellBasisVector2    0   18.2583  31.9880   0.00
CellBasisVector3    0   18.2712  10.5596  30.1748

CellBasisVector1    1   40.00     0.00    0.00
CellBasisVector2    1    0.00    40.00    0.00
CellBasisVector3    1    0.00    00.00   40.00

GEMC        NPT

Pressure    5.0

FixVolBox0  true
\end{sphinxVerbatim}


\section{Calculate Solvation Free Energy}
\label{\detokenize{howto:calculate-solvation-free-energy}}
\sphinxAtStartPar
GOMC is capable of calcutating absolute solvation free energy in NVT or NPT ensemble. Here
we are focusing how to setup the GOMC simulation files to calculate absolute solvation free energy.

\sphinxAtStartPar
GOMC outputs the required informations (\(\frac{dE_{\lambda}}{d\lambda}\), \(\Delta E_{\lambda}\))
to calculate solvation free energy with various estimators, such as TI, BAR, MBAR, and more.


\subsection{Setup Simulation Files}
\label{\detokenize{howto:setup-simulation-files}}\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Using FreeEnergy BASH Script}

\sphinxAtStartPar
GOMC development group created a BASH script combined with Tcl scripting to automatically
generate GOMC input files for free energy simulations in NVT (master branch) or NPT (NPT branch) ensemble.

\sphinxAtStartPar
To try this script, execute the following command in your terminal to clone the FreeEnergy repository:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git  clone    https://github.com/msoroush/FreeEnergy.git
\PYGZdl{} \PYG{n+nb}{cd}   FreeEnergy
\end{sphinxVerbatim}

\sphinxAtStartPar
or simply download it from \sphinxhref{https://github.com/msoroush/FreeEnergy.git}{GitHub} .

\sphinxAtStartPar
Make sure that you installed all \sphinxhref{https://github.com/GOMC-WSU/Workshop/blob/AIChE2019/GOMC\_Requirements.pdf}{GOMC software requirement}. Follow the
\sphinxhref{https://github.com/msoroush/FreeEnergy/blob/master/README.md}{README} for more information.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Manual Preparation}

\sphinxAtStartPar
To simulate solvation free energy, we need to perform the following steps:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Generate the PDB and PSF files for a system containes 1 solulte + \sphinxstyleemphasis{N} solvent molecules.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Number of solvent molecules (\sphinxstyleemphasis{N}) must be determined by user, based on the system size.
\end{sphinxadmonition}

\item {} 
\sphinxAtStartPar
Equilibrate your system in NVT ensemble at specified \sphinxcode{\sphinxupquote{Temperature}}.

\item {} 
\sphinxAtStartPar
Equilibrate your system in NPT ensemble at specified \sphinxcode{\sphinxupquote{Temperature}} and \sphinxcode{\sphinxupquote{Pressure}}, using
PDB and PSF \sphinxcode{\sphinxupquote{restart}} files, generated from previous equilibration simulation.

\item {} 
\sphinxAtStartPar
Determine the number of intermediate states that lead to adequate overlaps between
neighboring states.

\item {} 
\sphinxAtStartPar
For each intermediate state (\(\lambda_i\)), create an unique directory and perform the following steps:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumii}{enumiii}{}{.}%
\item {} 
\sphinxAtStartPar
Use the \sphinxcode{\sphinxupquote{restart}} PDB file, generated from NPT equilibration simulation, to set the \sphinxcode{\sphinxupquote{Coordinates}}.

\item {} 
\sphinxAtStartPar
Use the \sphinxcode{\sphinxupquote{merged}} PSF files, generated from NPT equilibration simulation, to set the \sphinxcode{\sphinxupquote{Structure}}.

\item {} 
\sphinxAtStartPar
Define the free energy parameters in \sphinxcode{\sphinxupquote{config}} file:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Set the frequency of free energy calculation

\item {} 
\sphinxAtStartPar
Set the solute molecule kind name (resname) and number (resid)

\item {} 
\sphinxAtStartPar
Set the soft\sphinxhyphen{}core parameters

\item {} 
\sphinxAtStartPar
Define the lambda vecotrs for \sphinxcode{\sphinxupquote{VDW}} and \sphinxcode{\sphinxupquote{Coulomb}} interaction

\item {} 
\sphinxAtStartPar
Set the index (\(i\)) of the lambda vetor (\(\lambda\)), at which solute\sphinxhyphen{}solvent interaction
will be coupled with \(\lambda_i\), using \sphinxcode{\sphinxupquote{InitialState}} keyword.

\end{itemize}

\sphinxAtStartPar
Here is the example of free energy parameters for CO2 (resid 1) solvation,
with 9 intermediate states, where the solute\sphinxhyphen{}solvent interaction will be
coupled with \(\lambda_{\texttt{VDW}}(6)= 1.0\) , \(\lambda_{\texttt{Elect}}(6)= 0.50\).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} FREE ENERGY PARAMETERS
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
FreeEnergyCalc true   1000
MoleculeType   CO2   1
InitialState   6
ScalePower     2
ScaleAlpha     0.5
MinSigma       3.0
ScaleCoulomb   false
\PYGZsh{}states        0    1    2    3    4    5    6    7    8
LambdaVDW      0.00 0.25 0.50 0.75 1.00 1.00 1.00 1.00 1.00
LambdaCoulomb  0.00 0.00 0.00 0.00 0.00 0.25 0.50 0.75 1.00
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
Equilibrate your system in NVT or NPT ensemble.

\item {} 
\sphinxAtStartPar
Perform the production simulation in NVT or NPT ensemble.

\end{enumerate}

\end{itemize}

\end{enumerate}


\subsection{Process GOMC Free Energy Outputs}
\label{\detokenize{howto:process-gomc-free-energy-outputs}}
\sphinxAtStartPar
I free energy perturbation method, the free energy difference between two states A
(\(\lambda = 0.0\)) and B (\(\lambda = 1.0\)), with N \sphinxhyphen{} 2 intermediate states is given by:
\begin{equation*}
\begin{split}\Delta G(A \rightarrow B) = -\frac{1}{\beta} \sum_{i=0}^{N-1} \ln \big< \exp \big(- \beta \Delta E_{i, i+1} \big) \big>_i\end{split}
\end{equation*}
\sphinxAtStartPar
where \(\Delta E_{i, i+1} = E_{i+1} - E_{i}\) is the energy difference of the system between states \sphinxstyleemphasis{i} and \sphinxstyleemphasis{i+1},
and \(\big< \big>_i\) is the ensemble average for simulation performed in intermediate state \sphinxstyleemphasis{i}.

\sphinxAtStartPar
In thermodynamic integration, the free energy change is calculated from
\begin{equation*}
\begin{split}\Delta G(A \rightarrow B) = \int_{\lambda = 0}^{\lambda = 1} \big< \frac{dU_{\lambda}}{d\lambda} \big>_{\lambda} d\lambda\end{split}
\end{equation*}
\sphinxAtStartPar
where \(\frac{dU_{\lambda}}{d\lambda}\) is the derivative of energy with respect
to \(\lambda\), and \(\big< \big>_{\lambda}\) is the ensemble average for a
simulation run at intermediate state \(\lambda\).

\sphinxAtStartPar
GOMC outputs the raw informations, such as the lambda intermediate states,
the derivative of energy with respective to current lambda (\(\frac{dE_{\lambda}}{d\lambda}\)),
the energy different between current lambda state and all other neighboring lambda states
(\(\Delta E_{\lambda}\)), which is essential to calculate solvation free energy
with various estimators, such as TI, BAR, MBAR, and more.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{FE-snapshot}.png}
\caption{Snapshot of GOMC free energy output file (Free\_Energy\_BOX\_0\_ \sphinxcode{\sphinxupquote{OutputName}}.dat).}\label{\detokenize{howto:id1}}\end{figure}

\sphinxAtStartPar
There are variety of tools developed to caclulate free energy difference, including
\sphinxhref{https://github.com/alchemistry/alchemlyb}{alchemlyb} and
\sphinxhref{https://github.com/MobleyLab/alchemical-analysis}{alchemical\sphinxhyphen{}analysis} .
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Alchemlyb}

\sphinxAtStartPar
In \sphinxhref{https://alchemlyb.readthedocs.io/en/latest}{alchemlyb} , a variety of methods can be
used to estimate the free energy, including thermodynamic integration (TI),
Bennett acceptance ratio (BAR), and multistate Bennett acceptance ratio (MBAR).
\sphinxhref{https://alchemlyb.readthedocs.io/en/latest}{alchemlyb}  is also capable of loading GOMC
free energy output files (Free\_Energy\_BOX\_0\_ \sphinxcode{\sphinxupquote{OutputName}}.dat).

\sphinxAtStartPar
To learn more about alchemlybe, please refere to \sphinxhref{https://alchemlyb.readthedocs.io/en/latest}{alchemlyb documentation}
or \sphinxhref{https://github.com/alchemistry/alchemlyb}{alchemlyb GitHub} page.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Currently, alchemlyb does not support the free energy plots, overlap analysis,
and free energy convergance analysis.
\end{sphinxadmonition}

\sphinxAtStartPar
To use this tool, you must install python 3 and then execute the following command in
your terminal to install alchemlyb:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} pip install alchemlyb
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Alchemical Analysis}

\sphinxAtStartPar
The alchemical\sphinxhyphen{}analysis tools is developed by Mobley group at MIT, to Analyze alchemical free energy
calculations conducted in GROMACS, AMBER or SIRE. Alchemical Analysis is still available but deprecated and
in the process of migrating all functionality to \sphinxhref{https://github.com/alchemistry/alchemlyb}{alchemlyb} tool.

\sphinxAtStartPar
Alchemical Analysis tool handles analysis via a slate of free energy methods, including BAR,
MBAR, TI, and the Zwanzig relationship (exponential averaging) among others, and provides a good deal
of analysis of computed free energies and convergence in order to help you assess the quality of your results.

\sphinxAtStartPar
Since alchemical\sphinxhyphen{}analysis is no longer supported by its developers, the GOMC parser for this tool
was implemented and stored in a separate \sphinxhref{https://github.com/msoroush/alchemical-analysis}{repository}.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
We encourage user to use \sphinxhref{https://github.com/alchemistry/alchemlyb}{alchemlyb GitHub} tools for plotting,
once all the plotting features and free energy analysis was migrated.
\end{sphinxadmonition}

\sphinxAtStartPar
To use this tool, you must install python 2 and then execute the following command in
your terminal to clone the alchemical\sphinxhyphen{}analysis repository:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git  clone    https://github.com/msoroush/alchemical\PYGZhy{}analysis.git
\PYGZdl{} \PYG{n+nb}{cd}   alchemical\PYGZhy{}analysis
\PYGZdl{} sudo python setup.py install
\end{sphinxVerbatim}

\end{enumerate}


\section{Run a Multi\sphinxhyphen{}Sim}
\label{\detokenize{howto:run-a-multi-sim}}
\sphinxAtStartPar
GOMC can automatically generate independent simulations with varying temperatures from one input file.
This allows the user to sample a wider seach space.  To do so GOMC must be compiled in MPI mode,
and a couple of parameters must be added to the conf file.

\sphinxAtStartPar
To compile in MPI mode, navigate to the GOMC/ directory and issue the following commands:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} chmod u+x metamakeMPI.sh
\PYGZdl{} ./metamakeMPI.sh
\end{sphinxVerbatim}

\sphinxAtStartPar
Then once the compilation is complete, set up the conf file as you would for a standard GOMC simulation.

\sphinxAtStartPar
Finally, enter more than one value for \sphinxcode{\sphinxupquote{Temperature}} separated by a tab or space.
\begin{quote}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
\PYGZsh{} SIMULATION CONDITION
\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}\PYGZsh{}
Temperature   270.00    280.00    290.00    300.00
\end{sphinxVerbatim}
\end{quote}

\sphinxAtStartPar
A folder will be created for the output of each simulation, and the name will be generated from the temperatures you choose.
A parent folder containing all the child folders will be created so as to not overpopulate the initial directory.
You may elect to choose the name of the folder in which all the sub\sphinxhyphen{}folders for each replica are contained.
Enter this name as a string following the \sphinxcode{\sphinxupquote{MultiSimFolderName}} parameter.  If you don’t provide this parameter, the default “MultiSimFolderName” will be used.
\begin{quote}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
MultiSimFolderName  outputFolderName
\end{sphinxVerbatim}
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
To perform a multisim, GOMC must be compiled in MPI mode.  Also, if GOMC is compiled in MPI mode, a multisim must be performed.  To perform a standard simulation, use standard GOMC.
\end{sphinxadmonition}

\sphinxAtStartPar
The rest of the conf file should be similar to how you would set up a standard GOMC simulation.

\sphinxAtStartPar
To initiate the multi\sphinxhyphen{}sim, first decide how many MPI processes and openMP threads you want to use and call GOMC with the following format.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mpiexec \PYGZhy{}n \PYG{c+c1}{\PYGZsh{}ofsimulations GOMC\PYGZus{}xxx\PYGZus{}yyyy +p\PYGZlt{}\PYGZsh{}ofthreads\PYGZgt{}(optional) conffile}
\end{sphinxVerbatim}

\sphinxAtStartPar
The number of MPI processes must equal the number of simulations you wish to run.  Each will by default be assigned one openMP thread; however, if you have leftover processors, you may assign them as openMP threads.
There must be an equal amount of openMP threads assigned to each process.

\sphinxAtStartPar
A formula to determine how many threads to use is as follows:
\begin{equation*}
\begin{split}OpenMP Threads = floor[(Number Of Processors Available - Number Of MPI Processes) / Number Of MPI Processes]\end{split}
\end{equation*}
\sphinxAtStartPar
Floor{[} {]} \sphinxhyphen{} Rounds down a real number to the nearest integer.

\sphinxAtStartPar
For example, if I have 7 processors and I wanted to run 2 simulations in my multi\sphinxhyphen{}sim.
\begin{equation*}
\begin{split}OpenMP Threads= floor[(7 - 2) / 2] = floor[2.5] = 2\end{split}
\end{equation*}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mpiexec \PYGZhy{}n \PYG{l+m}{2} ./GOMC\PYGZus{}CPU\PYGZus{}GEMC +p2 in.conf
\end{sphinxVerbatim}


\chapter{Get Help or Technical Support}
\label{\detokenize{help:get-help-or-technical-support}}\label{\detokenize{help::doc}}
\sphinxAtStartPar
For get any help or technical support, please send message to GOMC gitter:

\sphinxAtStartPar
\sphinxurl{https://gitter.im/GOMC\_WSU/Lobby}

\sphinxAtStartPar
or send email to:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Jeffrey Potoff: \sphinxhref{mailto:jpotoff@wayne.edu}{jpotoff@wayne.edu}

\item {} 
\sphinxAtStartPar
Loren Schwiebert: \sphinxhref{mailto:loren@wayne.edu}{loren@wayne.edu}

\end{itemize}



\renewcommand{\indexname}{Index}
\printindex
\end{document}